<?xml version="1.0" encoding="UTF-8" ?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.1//EN" 
   "http://www.w3.org/TR/xhtml11/DTD/xhtml11.dtd">

<html xmlns="http://www.w3.org/1999/xhtml">
<head>
  <meta http-equiv="content-type" content="text/html; charset=utf-8" />
  <meta name="KeyWords" content=
  "current topics in biostatistics, Korbinian Strimmer" />
  <meta name="Author" content="Korbinian Strimmer" />
  <meta name="Description" content="Course Description" />
  <meta name="viewport" content="width=device-width, initial-scale=1" />

  <title>Current Topics in Biostatistics</title>
  <link rel="stylesheet" type="text/css" href="../../../styles.css" />
</head>

<body>
  <script type="text/javascript">
//<![CDATA[
  <!--  
   if (top.location != self.location) {top.location = self.location}
  //-->
//]]>
</script>

  <div id="page">

  <div id="header">Current Topics in Biostatistics</div>


 <div id="main">

 
  <span class="sidenote">
  Back to <a href="https://strimmerlab.github.io/courses.html">Teaching</a> page


    </span>

  <p><strong>Contact and Information:</strong></p>


    <p><a href="https://strimmerlab.github.io/korbinian.html">Korbinian Strimmer</a> and <a href="http://www.brunel.ac.uk/people/alex-lewin">Alex Lewin</a><br />
       
 Imperial College London <br />
 Academic Year 2016-17</p>

  <span class="sidenote">
<b>Begin:</b> 15 September 2016 <br />
<b>End:</b> July 2017 <br />
<b>Time:</b> 2pm-3pm<br />
<b>Place:</b> EBS Meeting Room (162), 1st Floor, St. Mary's<br />

    </span>


 
   <p><strong>Synopsis:</strong></p>


<p>
In this journal club we aim to discuss in an informal way statistical and machine learning papers presenting methods
relevant to biostatistical analysis.   We read across books, journals and preprints, as we see fit.
</p>


 

    <p><strong>Schedule:</strong></p>


<p>We restart the session on <strong>15 September 2016</strong> and 
then meet regulary about twice a month. For precise dates and links to all the papers discussed see below!</p>

<p>Announcements concerning this journal club are also sent via the
<a href="https://mailman.ic.ac.uk/mailman/listinfo/statistics-stmarys">statistics-stmarys</a>
mailing list. Please join this list if you would like to receive regular updates by email.</p>


<p>Sessions:</p>
<table border="1"  summary="dates and contents of each session">
  <tr>
    <th>Session </th>
    <th>Date</th>
    <th>Topic</th>
  </tr>


<tr>
    <td>46</td>
    <td>14 September 2017</td>
    <td>D. Ahfock, W. J. Astle, and S. Richardson. 2017.
<a href="https://arxiv.org/abs/1706.03665">
Statistical properties of sketching algorithms</a>. arXiv:1706.03665.
    </td>
 </tr>

<tr>
    <td>&nbsp;</td>
    <td></td>
    <td>
    </td>
 </tr>

<tr>
    <td>45</td>
    <td>26 July 2017</td>
    <td>A. Terenin and D. Draper. 2017. <a href="https://arxiv.org/abs/1507.06597">Cox's Theorem and the Jaynesian Interpretation of Probability</a>. arXiv:1507.06597.
    </td>
 </tr>
<tr>
    <td>44</td>
    <td>12 July 2017</td>
    <td>Boyle et al. 2017. <a href="http://dx.doi.org/10.1016/j.cell.2017.05.038">An expanded view of complex traits:
From polygenic to omnigenic.</a> Cell 169: 1177-1186.
    </td>
 </tr>

<tr>
    <td>&nbsp;</td>
    <td></td>
    <td>
    </td>
 </tr>

<tr>
    <td>43</td>
    <td>28 June 2017</td>
    <td>
From Goodfellow et al. (2016):
     <a href="http://www.deeplearningbook.org/contents/convnets.html">Chapter 9: Convolutional Networks</a> and <a href="http://www.deeplearningbook.org/contents/rnn.html">Chapter 10: Sequence Modeling: Recurrent and Recursive Nets</a>.

    </td>
 </tr>
<tr>
    <td>42</td>
    <td>14 June 2017</td>

    <td>
    I. Goodfellow, Y. Bengio, and A. Courville. 2016. Deep Learning. MIT Press.
    (<a href="http://www.deeplearningbook.org/">official book web page</a>, 
     <a href="https://github.com/HFTrader/DeepLearningBook/blob/master/DeepLearningBook.pdf">inofficial pdf</a>). <br />
     We read: <a href="http://www.deeplearningbook.org/contents/ml.html">Chapter 5: Machine Learning Basics</a>  (only very briefly, essentially to catch up on notation),
     <a href="http://www.deeplearningbook.org/contents/mlp.html">Chapter 6: Deep Forward Networks</a>, and <a href="http://www.deeplearningbook.org/contents/regularization.html">Chapter 7: Regularization for Deep Learning</a>.
    </td>
 </tr>

<tr>
    <td>&nbsp;</td>
    <td></td>
    <td>
    </td>
 </tr>

<tr>
    <td>41</td>
    <td>31 May 2017</td>
    <td>Keziou and  Regnault. 2017. <a href="http://www.dx.doi.org/10.1109/TIT.2016.2620163">Semiparametric estimation of
        mutual information and related criteria: optimal test of independence.</a> IEEE Trans. Inf. Theo. 63: 57-71
    </td>
 </tr>
<tr>
    <td>40</td>
    <td>17 May 2017</td>
    <td>Zhang et al 2017. <a href="https://arxiv.org/abs/1611.03530">Understanding deep learning requires rethinking generalization.</a> To appear in Proceedings of <a href="http://www.iclr.cc/doku.php?id=iclr2017:main">ICLR 2017</a>.  
    </td>
 </tr>
<tr>
    <td>39</td>
    <td>3 May 2017</td>
    <td>Wang and Raj. 2017. <a href="https://arxiv.org/abs/1702.07800">On the 
         origin of deep learning.</a>  arXiv:1702.07800.
    </td>
 </tr>

<tr>
    <td>&nbsp;</td>
    <td></td>
    <td>
    </td>
 </tr>

<tr>
    <td>38</td>
    <td>5 April 2017</td>
    <td> Leek and Jager. 2017. <a href="https://dx.doi.org/10.1146/annurev-statistics-060116-054104"> Is most published research really false?</a> Ann. Rev. Statist. Appl. 4:109-122.
    </td>
 </tr>

<tr>
    <td>&nbsp;</td>
    <td></td>
    <td>
    </td>
 </tr>

<tr>
    <td>37</td>
    <td>22 March 2017</td>
    <td>
    Schafer 1999. <a href="https://dx.doi.org//10.1177/096228029900800102">Multiple imputation: a primer</a>. Statist. Meth. Medical Res. 8: 3-15;<br />
    Stekhoven  and B&uuml;hlmann. 2012. 
    <a href="https://dx.doi.org/10.1093/bioinformatics/btr597">MissForest - non-parametric missing value 
    imputation for mixed-type data</a>. Bioinformatics 28: 112-118.
    </td>
 </tr>



<tr>
    <td>36</td>
    <td>8 March 2017</td>
    <td>
    Deng et al 2016.  <a href="http:dx.doi.org//10.1038/srep21689">Multiple imputation for general missing data patterns in the presence of high-dimensional data</a>. Scientific Reports 6: 21689.
    </td>
 </tr>

<tr>
    <td>&nbsp;</td>
    <td></td>
    <td>
    </td>
 </tr>

<tr>
    <td>35</td>
    <td>22 February 2017</td>
    <td>
    Cannings and Samworth. 2015. <a href="https://arxiv.org/abs/1504.04595">Random projection ensemble classification</a>. JRSS B to appear.  See also the info on the <a href="http://www.rss.org.uk/RSS/Publications/Journals/Discussion_Papers/RSS/Publications/Journals_sub/discussion_papers.aspx#15March17">RSS Discussion Meeting, Wednesday 15 March 2017</a>.
    </td>
 </tr>

<tr>
    <td>34</td>
    <td>8 February 2017</td>
    <td>
Schulam and Saria. 2016. <a href="http://jmlr.org/papers/v17/15-436.html">Integrative analysis using coupled latent variable models for individualizing prognoses</a>. Journal of Machine Learning Research 17: 1-35.
    </td>
 </tr>


<tr>
    <td>&nbsp;</td>
    <td></td>
    <td>
    </td>
 </tr>

<tr>
    <td>33</td>
    <td>25 January 2017</td>
    <td>Buzdugan et al. 2016. <a href="http://dx.doi.org/10.1093/bioinformatics/btw128">Assessing statistical significance in multivariable genome wide association analysis</a>.
        Bioinformatics 32:1990-2000. <br />
       Mandozzi and B&uuml;hlmann. 2016.  <a href="http://dx.doi.org/10.1080/01621459.2015.1007209">Hierarchical testing in the high-dimensional setting with correlated variables</a>. JASA 111:331-343. 


    </td>
 </tr>


<tr>
    <td>&nbsp;</td>
    <td></td>
    <td>
    </td>
 </tr>


<tr>
    <td>32</td>
    <td>8 December 2016</td>
    
<td>M. Papathomas et al. 2012. <a href="http://dx.doi.org/10.1002/gepi.21661">
        Exploring Data From Genetic Association Studies Using Bayesian
Variable Selection and the Dirichlet Process: Application to Searching
for Gene by Gene Patterns</a>. Genetic Epidemiology 36 : 663-674; <br />
R. Argiento et al. 2015. <a href="http://link.springer.com/chapter/10.1007/978-3-319-19518-6_6">Modeling the Association Between Clusters
of SNPs and Disease Responses.</a> Pages 115-134 (Chapter 6) in: R. Mitra
  and P. M&uuml;ller (eds.). Nonparametric Bayesian Inference in Biostatistics,
  Springer.
    </td>
</tr>
<tr>
    <td>&nbsp;</td>
    <td></td>
    <td>
    </td>
 </tr>



<tr>
    <td>31</td>
    <td>3 November 2016</td>
    <td>A. Klami et al. 2013. <a href="http://www.jmlr.org/papers/v14/klami13a.html">Bayesian canonical correlation analysis</a>.
        JMLR 14:965-1003.
    </td>
 </tr>

<tr>
    <td>&nbsp;</td>
    <td></td>
    <td>
    </td>
 </tr>


<tr>
    <td>30</td>
    <td>20 October 2016</td>
    <td>M. T. Ribeiro et al. 2016. <a href="http://dx.doi.org/10.1145/2939672.2939778">''Why should I trust you?'' Explaining the predictions of any classifier.</a> KDD 2016 - Proceedings of the 22nd ACM SIGKDD International Conference on Knowledge Discovery and Data Mining: 1135-1144.
 (<a href="https://arxiv.org/abs/1602.04938">arXiv:1602.04938</a>)
    </td>
 </tr>


<tr>
    <td>&nbsp;</td>
    <td></td>
    <td>
    </td>
 </tr>

<tr>
    <td>29</td>
    <td>15 September 2016</td>
    <td>
M. Drton and M. Plummer. 2016. <a href="http://arxiv.org/abs/1309.0911">A Bayesian information criterion for singular models</a>. JRSS B to appear.  This paper will be <a href="https://www.statslife.org.uk/events/events-calendar/eventdetail/719/-/ordinary-meeting-a-bayesian-information-criterion-for-singular-models">read on 5 October 2016 at the Royal Statistical Society</a>. <br /> For this session, we meet at <a href="http://www.brunel.ac.uk/">Brunel University</a>, John Crank Building, Room 128, at 12:30 - please send email to <a href="mailto:alex.lewin@brunel.ac.uk">Alex Lewin</a> to confirm attendance.
    </td>

 </tr>


<tr>
    <td> </td>
    <td> </td>
    <td>&nbsp;
    </td>
 </tr>

<tr>
    <td>&nbsp;</td>
    <td></td>
    <td>
    </td>
 </tr>



<tr>
    <td>28</td>
    <td>26 Jul 2016</td>
    <td>
J. W. Miller and M. T. Harrison. 2015. 
<a href="http://arxiv.org/abs/1502.06241">Mixture models with a prior on the number of
components</a>. arXiv:1502.06241
    </td>
 </tr>
<tr>
    <td>27</td>
    <td>12 Jul 2016</td>
    <td>
J. G. Scott et al. 2015. <a href="http://dx.doi.org/10.1080/01621459.2014.990973">False discovery rate regression: An application to neural synchrony detection  in primary visual cortex</a>. JASA 110: 459-471; <br />
W. Tansey, O. Koyejo, R. A. Poldrack, J. G. Scott.
2014. <a href="http://arxiv.org/abs/1411.6144">False discovery rate smoothing</a>. 
arXiv:1411.6144.
    </td>
 </tr>

<tr>
    <td>&nbsp;</td>
    <td></td>
    <td> 
    </td>
 </tr>


<tr>
    <td>26</td>
    <td>7 Jun 2016</td>
    <td>G. J. Szekely and M. L. Rizzo. 2009. 
<a href="http://dx.doi.org/10.1214/09-AOAS312">Brownian distance correlation</a>. Ann. Applied Statist. 3: 1236-1265; <br />
 D. N. Reshef et al. 2011. <a href="http://dx.doi.org/10.1126/science.1205438">Detecting novel associations in large data sets</a> 
Science 334: 1518-1524; <br />
N. Simon and R. Tibshirani. 2014.
<a href="http://arxiv.org/abs/1401.7645">Comment on "Detecting novel associations in large data sets"</a>.
arXiv:1401.7645; <br />
S. de Siqueira Santos et al. 2014.
<a href="http://dx.doi.org/10.1093/bib/bbt051">A comparative study of statistical methods used to identify dependencies between gene expression signals</a>. 
Brief. Bioinf. 15: 906-918. 
    </td>
 </tr>

<tr>
    <td>&nbsp;</td>
    <td></td>
    <td> 
    </td>
 </tr>


<tr>
    <td>25</td>
    <td>17 May 2016</td>
    <td>J. Wang, Q. Zhao, T. Hastie, and A. B. Owen. 2016. 
        <a href="http://arxiv.org/abs/1508.04178">Confounder adjustment in 
        multiple hypothesis testing.</a> arXiv:1508.04178.
    </td>
 </tr>



<tr>
    <td>24</td>
    <td>5 May 2016</td>
    <td>R. Guhaniyogi and D. B. Dunson 2015. <a href="http://dx.doi.org/10.1080/01621459.2014.969425">Bayesian compressed regression.</a>
         JASA 110: 1500-1514.
    </td>
 </tr>

<tr>
    <td>&nbsp;</td>
    <td></td>
    <td> 
    </td>
 </tr>



<tr>
    <td>23</td>
    <td>17 Mar 2016</td>
    <td>
 R. Zhang, C. Czado, and K. Sigloch. 2016.
<a href="http://dx.doi.org/10.1111/rssc.12118">Bayesian spatial modelling for high
dimensional seismic inverse problems</a>. JRSS C 65: 187-213.
    </td>
 </tr>

<tr>
    <td>22</td>
    <td>Thu 3 Mar 2016</td>
    <td>Q. Song and F. Liang. 2015.  <a href="http://dx.doi.org/10.1111/rssb.12095">A split-and-merge Bayesian variable selection approach for ultrahigh dimensional regression</a>. JRSS B 77: 947-972.   
    </td>
 </tr>

<tr>
    <td>&nbsp;</td>
    <td></td>
    <td> 
    </td>
 </tr>

<tr>
    <td>21</td>
    <td>Thu 18 Feb 2016</td>
    <td>  
       B. Efron. 2014. <a href="http://dx.doi.org/10.1214/13-STS455">Two modeling strategies for empirical Bayes estimation.</a>
        Statistical Science 29:285-301. B. Efron. 2015. <a href="http://statweb.stanford.edu/~ckirby/brad/papers/2015BayesDeconvolutionProblem.pdf">The Bayes deconvolution problem.</a> Preprint.
    </td>
 </tr>


<tr>
    <td>20</td>
    <td>Thu 4 Feb 2016</td>
    <td>N. Fusi et al. 2012.
<a href="http://dx.doi.org/10.1371/journal.pcbi.1002330">Joint modelling of confounding factors and prominent genetic regulators provides increased accuracy in genetical genomics studies.</a> PLOS Comp. Biol. 8: e1002330 and 
O. Stegle et al. 2010. <a href="http://dx.doi.org/10.1371/journal.pcbi.1000770">A Bayesian framework to account for complex non-genetic factors.</a> PLOS Comp. Biol. 6: e1000770
    </td>
 </tr>

<tr>
    <td>&nbsp;</td>
    <td></td>

    <td>  
    </td>
 </tr>


<tr>
    <td>19</td>
    <td>Thu 21 Jan 2016</td>
    <td> M. Taddy et al. 2015. 
       <a href="http://jmlr.org/proceedings/papers/v37/matthew15.pdf">Bayesian and empirical Bayesian forests.</a>  32nd International Conference on Machine Learning (ICML). JMLR Workshop Proceeedings Vol. 37.

    </td>
 </tr>

<tr>
    <td>&nbsp;</td>
    <td></td>
    <td>  
    </td>
 </tr>


<tr>
    <td>18</td>
    <td>Mon 14 Dec 2015</td>
    <td>J. Taylor and R. J. Tibshirani. 2015. <a href="http://dx.doi.org/10.1073/pnas.1507583112">Statistical learning and selective inference.</a>
        PNAS 25:7629-7634.


    </td>
 </tr>
 

<tr>
    <td>17</td>
    <td>Thu 3 Dec 2015</td>
    <td>A. P. Dawid. 2010. <a href="http://jmlr.csail.mit.edu/proceedings/papers/v6/dawid10a/dawid10a.pdf">Beware of the DAG!</a>
       JMLR Workshop and Conference Proceedings 6:59-86 (NIPS 2008 Workshop on Causality).
    </td>
 </tr>


<tr>
    <td>&nbsp;</td>
    <td></td>
    <td>  
    </td>
 </tr>


<tr>
    <td>16</td>
    <td>Thu 19 Nov 2015</td>
    <td>J. Piironen and A. Vehtari. 2015. 
        <a href="http://arxiv.org/abs/1503.08650">Comparison of Bayesian predictive 
        methods for model selection</a>. arXiv:1503.08650. <br />
J. Piironen and A. Vehtari. 2015. <a href="http://arxiv.org/abs/1508.02502">Projection predictive variable selection using
Stan+R</a>. arXiv:1508.02502.
        
    </td>
 </tr>



<tr>
    <td>&nbsp;</td>
    <td></td>
    <td>
    </td>
 </tr>



<tr>
    <td>15</td>
    <td>Thu 29 Oct 2015</td>
    <td> 
 D. P. Simpson, H. Rue, T. G. Martins, A. Riebler, and S. H. S&oslash;rbye. 2014. <a href="http://arxiv.org/abs/1403.4630">Penalising model component complexity: A principled, practical approach to constructing priors</a>. arXiv:1403.4630.
    </td>
 </tr>



<tr>
    <td>14</td>
    <td>Wed 14 Oct 2015</td>
    <td>
         B. Lakshminarayanan, D. M. Roy, Y. W. Teh. 
         2014. <a href="http://arxiv.org/abs/1406.2673">Mondrian forests: efficient online random forests</a>.
         Advances in Neural Information Processing Systems 27:3140-3148.<br />
B. Lakshminarayanan, D. M. Roy, Y. W. Teh. 
         2015. 
         <a href="http://arxiv.org/abs/1506.03805">Mondrian forests for 
         large-scale regression when uncertainty matters</a>. arXiv:1506.03805.
         


 
    </td>
 </tr>



<tr>
    <td>&nbsp;</td>
    <td></td>
    <td>  
    </td>
 </tr>



<tr>
    <td>13</td>
    <td>Wed 30 Sep 2015</td>
    <td>B. Letham, C. Rudin, T.H. McCormick and D. Madigan. 2015. <a href="http://www.e-publications.org/ims/submission/AOAS/user/submissionFile/17235?confirm=20d145e7">Interpretable classifiers using rules and

         Bayesian analysis: building a better stroke prediction model.</a> AOAS in press.<br />
         See also the <a href="https://www.aaai.org/ocs/index.php/WS/AAAIW13/paper/download/7088/6696">short note</a> presented at the 2013 AAAI Conference on Artificial Intelligence.
    </td>
 </tr>

<tr>
    <td> </td>
    <td> </td>
    <td>&nbsp;
    </td>
 </tr>

<tr>
    <td> </td>
    <td> </td>
    <td>&nbsp;
    </td>
 </tr>

<tr>
    <td>12</td>
    <td>Wednesday 29 July 2015</td>
    <td>  LeCun et al. 2015. <a href="http://dx.doi.org/10.1038/nature14539">Deep learning</a>. Nature 521: 436-444, and
          Salakhutdinov. 2014. <a href="http://dx.doi.org/10.1146/annurev-statistics-010814-020120">Learning deep generative models</a>.
          Annu. Rev. Stat. Appl. 2: 361-385.
    </td>
 </tr>



<tr>
    <td>11</td>
    <td>Wednesday 15 July 2015</td>
    <td>Lange et al. 2013. <a href="http://aje.oxfordjournals.org/content/179/4/513.abstract">Assessing Natural Direct and Indirect Effects Through Multiple Pathways</a>. AJE  179:513-518, and Boca et al. 2014. 
<a href="http://bioinformatics.oxfordjournals.org/content/30/2/214.abstract">Testing multiple biological mediators simultaneously</a>. Bioinformatics 30:214-220.
    </td>
 </tr>



 <tr>
    <td> </td>
    <td> </td>
    <td>&nbsp;
    </td>
 </tr>


<tr>
    <td>10</td>
    <td>Thursday 11 June 2015</td>
<td>Kamary, Mengersen, Robert and Rousseau. 2014. <a href="http://arxiv.org/abs/1412.2044">Testing hypotheses via a mixture
estimation model</a>. arXiv:1412.2044.
    </td>
 </tr>

<tr>
    <td>9</td>
    <td>Thursday 4 June 2015</td>
    <td>
    Hao et al. 2015. <a href="http://onlinelibrary.wiley.com/doi/10.1111/rssb.12092/abstract">Sparsifying the Fisher linear discriminant by rotation</a>.
    JRSS B  in press.
    </td>    
 </tr>

 <tr>
    <td> </td>
    <td> </td>
    <td>&nbsp;
    </td>
 </tr>

<tr>
    <td>8</td>
    <td>Thursday 14 May 2015</td>
    <td>Kirk et al. 2012. <a href="http://bioinformatics.oxfordjournals.org/content/28/24/3290.abstract">Bayesian correlated clustering to integrate multiple datasets.</a> Bioinformatics 28:3290-3297,
        and Savage et al. 2010. <a href="http://bioinformatics.oxfordjournals.org/content/26/12/i158.abstract">Discovering transcriptional modules by Bayesian data integration</a>.  Bioinformatics 26:i158-i167.
    </td>
 </tr>

 <tr>
    <td> </td>
    <td> </td>
    <td>&nbsp;
    </td>
 </tr>

<tr>
    <td>7</td>
    <td>Thursday 30 April 2015</td>
    <td>Varin, Cattelan and Firth. 2015.  
        <a href="http://www.rss.org.uk/Images/PDF/publications/rss-preprint-varin-may-2015.pdf">Statistical Modelling of Citation Exchange Among Statistics Journals. JRSS A to appear.</a> (<a href="http://www.rss.org.uk/Images/PDF/publications/rss-preprint-varin-supplement-may-2015.zip">supplementary material</a>, <a href="http://www.statslife.org.uk/events/events-calendar/eventdetail/426/-/rss-ordinary-meeting-statistical-modelling-of-citation-exchange-among-statistics-journals">RSS ordinary meeting</a>)
    </td>
 </tr>

<tr>
    <td>6</td>
    <td>Wednesday 15 April 2015</td>
    <td>Donoho and Jin. 2015. <a href="http://projecteuclid.org/euclid.ss/1425492437">Higher Criticism for Large-Scale Inference, Especially for Rare and Weak Effects.</a>
        Statist. Sci.  30:1-25. <br />
Klaus and Strimmer. 2013.
<a href="http://dx.doi.org/10.1093/biostatistics/kxs030">Signal identification for rare and weak features: higher criticism or false discovery rates?</a>
 Biostatistics 14: 129-143.
    </td>
 </tr>

  <tr>
    <td> </td>
    <td> </td>
    <td>&nbsp;
    </td>
 </tr>

  <tr>
    <td>5</td>
    <td>Wednesday 25 March 2015</td>
    <td>Jara et al 2011. 
<a href="http://www.jstatsoft.org/v40/i05/">DPpackage: Bayesian Semi- and Nonparametric
Modeling in R. </a> J. Statist. Software 40:5.

    </td>
 </tr>
 <tr>
    <td>4</td>
    <td>Wednesday 18 March 2015</td>
    <td> Chapter 7 ("Applications in Biostatistics") of <a href="http://www.cambridge.org/us/academic/subjects/statistics-probability/statistical-theory-and-methods/bayesian-nonparametrics">"Bayesian Nonparametrics"</a> book.
    </td>
 </tr>
 <tr>
    <td>3</td>
    <td>Thursday 12 March 2015</td>
    <td> 
Yau and Holmes. 2011. 
<a href="http://projecteuclid.org/euclid.ba/1339612049">Hierarchical Bayesian nonparametric mixture models for clustering with variable relevance determination.</a>
Bayesian Anal. 6:329-351

 <br />Liverani et al. 2014. 
    <a href="http://arxiv.org/abs/1303.2836">PReMiuM: An R Package for Profile Regression Mixture Models using Dirichlet Processes</a>  J. Statist. Software, to appear.

    </td>
 </tr>


  <tr>
    <td>2</td>
    <td>Wednesday 4 March 2015</td>
    <td>Gershman and Blei. 2012. <a href="http://dx.doi.org/10.1016/j.jmp.2011.08.004">A tutorial on Bayesian nonparametric models.</a> J. Math. Psychol. 56: 1-12.
    </td>
 </tr>

<tr>
    <td> </td>
    <td> </td>
    <td>&nbsp;
    </td>
 </tr>


  <tr>
    <td>1</td>
    <td>Wednesday 25 February 2015</td>
    <td>
Overview ("An Invitation to Bayesian Nonparametrics") and Chapter 1 ("Motivation and Ideas") of <a href="http://www.cambridge.org/us/academic/subjects/statistics-probability/statistical-theory-and-methods/bayesian-nonparametrics">"Bayesian Nonparametrics"</a> by Nils Lid Hjort, Chris Holmes, Peter M&uuml;ller, and Stephen G. Walker.
2010. Cambridge University Press.
    </td>
 </tr>

<tr>
    <td></td>
    <td>2011-2014</td>
<td>
       For related book discussion groups run 2014 and 2011/12 at the University of Leipzig see 
        <a href="https://strimmerlab.github.io/courses/2013-14/current-topics/index.html">here</a> (Lindley "Understanding Uncertainty")
        and <a href="https://strimmerlab.github.io/courses/2010-11/current-topics/index.html">here</a> (Jaynes "Probability: The Logic of Science"). 
    </td>
 </tr>

</table>  


</div>


<div id="footer">
<em>Last modified:</em>
        September 12, 2017<br />
        <a href="http://validator.w3.org/check?uri=referer">
           <img style="float: right;" src="../../../images/valid-xhtml11.png" alt="Valid XHTML 1.1"
           height="31" width="88" /></a>

  </div>

  </div>
</body>
</html>
