<?xml version="1.0" encoding="UTF-8" ?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.1//EN" 
   "http://www.w3.org/TR/xhtml11/DTD/xhtml11.dtd">

<html xmlns="http://www.w3.org/1999/xhtml">

<head>
  <meta http-equiv="content-type" content="text/html; charset=utf-8" />
  <meta name="KeyWords" content=
  "Korbinian Strimmer, biostatistics, biomathematics, statistical learning, bionformatics, genetic networks, complex networks, reverse engineering, information theory, transcriptome analysis" />
  <meta name="Author" content="Korbinian Strimmer" />
  <meta name="Description" content="Publications" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />

  <title>Korbinian Strimmer - Publications</title>

  <link rel="icon" href="images/favicon.ico" />
  <link rel="shortcut icon" href="images/favicon.ico" />
  <link rel="stylesheet" type="text/css" href="styles.css" />
</head>

<body>
  <script type="text/javascript">
//<![CDATA[
  <!--  
   if (top.location != self.location) {top.location = self.location}
  //-->
  
//]]>
</script>


<div id="page">

  <div id="header"> 

      <a href="https://www.manchester.ac.uk/">
       <img style="border-width:0px;  margin-top: 4px; margin-left: 3px;"  
       src="images/logo-university-of-manchester.png" 
       alt="logo of The University of Manchester"/>
      </a>

<a href="images/people/korbinian.jpg"><img class="iconsize" 
        style="border-width:0px; float: right; margin-right: 1px; margin-top: 2px;"
        src="images/people/korbinian-small.jpg" alt="Picture of Korbinian" /></a> 
      <br />
      Korbinian Strimmer

  </div>


    <div id="nav">
       <a href="korbinian.html">Korbinian Strimmer</a> -
       <a href="https://www.maths.manchester.ac.uk/our-research/">Department of
          Mathematics</a>
       <br />
       <a href="index.html">Home</a> | 
       <a href="people.html">People</a> |
       <a href="courses.html">Teaching</a> |
       <a href="supervision.html">Supervision</a> |
       <a href="research.html">Research</a> | 
       Publications  
    </div>


  <div id="main">

 <span class="sidenote">

   Click on my portrait above for a larger view.
  
 </span>

<p><strong>Publications:</strong></p>

<p>Please find below a list of my publications.
You may also be interested in
my <a href="https://arxiv.org/search/?searchtype=author&amp;query=Strimmer%2C+K">preprints on arXiv</a>,
my <a href="https://scholar.google.com/citations?user=Vezv6t0AAAAJ">Google Scholar Page</a>
and my <a href="https://publons.com/researcher/2863203/korbinian-strimmer/">Publons profile</a>.
</p>

<p><strong>Lecture notes:</strong></p>

<ul>
<li><a href="publications/lecture-notes/MATH38161/index.html">
Multivariate Statistics and Machine Learning (MATH38161)</a>. 2018-2021.<br />  181 pages (+ support material: worksheets, lecture videos).</li>

<li><a href="publications/lecture-notes/MATH20802/index.html">
Statistical Methods: Likelihood, Bayes and Regression (MATH20802)</a>. 2019-2022.<br />  197 pages (+ support material: worksheets, lecture videos).</li>
</ul>

<p><strong>Refereed publications:</strong></p>

<ul >

<li>
<strong>Quantification of protein abundance and interaction defines a mechanism for operation of the circadian clock</strong>.<br />
A. A. Koch and others. 2022.
<a href="https://doi.org/10.7554/eLife.73976"> eLife <strong>11</strong>: e73976</a>.<br />
(<a href="https://www.biorxiv.org/content/10.1101/2021.08.27.456017v1">biorXiv:2021.08.27.456017</a>)
</li>



<li>
 <span class="sidenote">See <a href="software/whitening/index.html">whitening</a> homepage for supporting R code.</span>

<strong>A whitening approach to probabilistic canonical correlation analysis for omics data integration</strong>.<br />
T. Jendoubi and K. Strimmer. 2019.
<a href="https://doi.org/10.1186/s12859-018-2572-9">BMC Bioinformatics <strong>20</strong>: 15</a>.<br />
(<a href="https://arxiv.org/abs/1802.03490">arXiv:1802.03490</a>)
</li>

<li>

 <span class="sidenote">All methods implemented in <a href="software/whitening/index.html">whitening</a>. Run <a href="software/whitening/download/optimal-whitening-table2.R">optimal-whitening-table2.R</a> to reproduce Table 2.</span>

<strong>Optimal whitening and decorrelation</strong>.<br />
A. Kessy, A. Lewin, and K. Strimmer. 2018. <br />
<a href="https://doi.org/10.1080/00031305.2016.1277159">The American Statistician <strong>72</strong>: 309-314</a>. <br />
(<a href="https://arxiv.org/abs/1512.00809">arXiv:1512.00809</a>)
</li>


<li>
 <span class="sidenote">See also the <a href="software/maldiquant/index.html">MALDIquant</a> homepage.</span>

<strong>Mass spectrometry analysis using MALDIquant</strong>.<br />
S. Gibb and  K. Strimmer. 2016. <br />
<a href="http://link.springer.com/chapter/10.1007/978-3-319-45809-0_6">Chapter 6 (pp. 101-124)</a> in: Datta, S., and Mertens, B. (eds). 
<a href="http://link.springer.com/book/10.1007/978-3-319-45809-0">Statistical Analysis of Proteomics, Metabolomics, and Lipidomics Data Using Mass Spectrometry</a>. Springer, New
York.<br />
(<a href="https://arxiv.org/abs/1607.03180">arXiv:1607.03180</a>)

</li>


<li>
<span class="sidenote">This method is available in the <a href="software/binda/index.html">binda</a> R package.</span>
<strong>Differential protein expression and peak selection in mass
spectrometry data by binary discriminant analysis</strong>.<br />
S. Gibb and K. Strimmer. 2015. <br />
<a href="https://doi.org/10.1093/bioinformatics/btv334">Bioinformatics <strong>31</strong>: 3156-3162</a>.<br />
(<a href="https://arxiv.org/abs/1502.07959">arXiv:1502.07959</a>)
</li>


<li>
  <span class="sidenote">
<img src="images/highlyaccessed.gif" alt=
        "BMC highly accessed" style=
        "height: 12px;" />
</span>
<strong>A simple data-adaptive probabilistic variant calling model</strong>.<br />
S. Hoffmann, P. F. Stadler, and K. Strimmer. 2015. <br />
<a href="https://doi.org/10.1186/s13015-015-0037-5">Algorithms for Molecular Biology <strong>10</strong>: 10.</a><br />
(<a href="https://arxiv.org/abs/1405.5251">arXiv:1405.5251</a>)
</li>

<li>
<strong>Signal identification for rare and weak features: higher criticism 
or false discovery rates?</strong><br />
B. Klaus and K. Strimmer. 2013. <br />
<a href="https://doi.org/10.1093/biostatistics/kxs030">Biostatistics <strong>14</strong>: 129-143</a>.<br />
(<a href="https://arxiv.org/abs/1112.2615">arXiv:1112.2615</a>)
</li>

<li>
  <span class="sidenote">
<img src="images/highlyaccessed.gif" alt=
        "BMC highly accessed" style=
        "height: 12px;" />
<br />
See the accompanying <a href="software/care/index.html#GAW17">R code to reproduce the complete analysis</a>.</span>

<strong>A novel algorithm for simultaneous SNP selection in
high-dimensional genome-wide association studies</strong>.<br />
V. Zuber, A. P. Duarte Silva, and K. Strimmer. 2012. <br />
<a href="https://doi.org/10.1186/1471-2105-13-284">BMC Bioinformatics <strong>13</strong>: 284</a><br />
(<a href="https://arxiv.org/abs/1203.3082">arXiv:1203.3082</a>)
</li>


<li>
  <span class="sidenote">See also the <a href="software/maldiquant/index.html">MALDIquant</a> homepage.</span>

<strong>MALDIquant: a versatile R package for the analysis of
            mass spectrometry data</strong>.<br />
S. Gibb and K. Strimmer. 2012. <br />
<a href="https://doi.org/10.1093/bioinformatics/bts447">Bioinformatics <strong>28</strong>: 2270-2271</a>.<br />
(<a href="https://arxiv.org/abs/1203.5885">arXiv:1203.5885</a>)
</li>


<li>
<strong>Introduction to graphical modelling</strong>.<br />
M. Scutari and K. Strimmer. 2011. <br />
Chapter 11 in: M. P. H. Stumpf, D. J. Balding, and M. Girolami (eds.).
      <a href="http://eu.wiley.com/WileyCDA/WileyTitle/productCd-0470710861.html">Handbook of Statistical Systems Biology</a>.  Wiley, Chichester, UK, <a href="https://doi.org/10.1002/9781119970606.ch11">pp. 237-254</a>.<br />(<a href="https://arxiv.org/abs/1005.1036">arXiv:1005.1036</a>)
</li>


<li>
<strong>Learning false discovery rates by fitting sigmoidal threshold functions</strong>.<br />
B. Klaus and K. Strimmer. 2011. <br />
<a href="http://journal-sfds.fr/article/view/62/53">Journal de la Soci&eacute;t&eacute;  Fran&ccedil;aise de Statistique <strong>152</strong>: 39-50</a>.<br />
(<a href="https://arxiv.org/abs/1104.5414">arXiv:1104.5414</a>)
</li>


<li>
  <span class="sidenote">This regression approach is available in the <a href="software/care/index.html">care</a> R package.</span>


<strong>High-dimensional regression and variable selection using CAR scores</strong>.<br />
V. Zuber and K. Strimmer. 2011. <br />
<a href="https://doi.org/10.2202/1544-6115.1730">Statist. Appl. 
      Genet. Mol. Biol. <strong>10</strong>: 34</a>. <br />
      (<a href="https://arxiv.org/abs/1007.5516">arXiv:1007.5516</a>,
       <a href="publications/journals/carscore2011.pdf">original bepress version</a>)
</li>


 <li> 
  <span class="sidenote">See the <a href="software/maldiquant/index.html">MALDIquant</a> R package.</span>

    <strong>Analysis of proteomic data using MALDIquant</strong>. <br />
    S. Gibb and K. Strimmer. 2011. <br />
    Proceedings of the <a href="http://www.wcsb2011.ethz.ch/">8th International Workshop on Computational Systems 
    Biology, WCSB 2011</a> (June 6-8, 2011, Zurich, Switzerland), <a href="publications/conferences/maldiquant-wcsb2011.pdf">pp. 49-52.</a>
</li>


<li>
<strong>Over-optimism in bioinformatics: an illustration</strong>.<br />
M. Jelizarow, V. Guillemot, A. Tenenhaus, K. Strimmer, and A.-L. Boulesteix. 2010. <br />
<a href="https://doi.org/10.1093/bioinformatics/btq323">Bioinformatics <strong>26</strong>: 1990-1998</a>.
</li>

 

<li> 
    <strong>Thresholding methods for feature selection in genomics: higher criticism versus
     false non-discovery rates</strong>. <br />
    B. Klaus and K. Strimmer. 2010. <br />
    Proceedings of the <a href="http://www.cs.tut.fi/wcsb10/">7th International Workshop on Computational Systems 
    Biology, WCSB 2010</a> (June 16-18, 2010, Luxembourg, Luxembourg),
    <a href="publications/conferences/hc-comparison2010.pdf">pp. 59-62.</a> 
</li>


<li>
     <span class="sidenote">See the <a href="software/sda/index.html">sda</a> R package.</span>

     <strong>Feature selection in omics prediction problems using cat scores and false non-discovery rate control</strong>.<br />
        M. Ahdesm&auml;ki and K. Strimmer. 2010. <br />
<a href="https://doi.org/10.1214/09-AOAS277">Ann. Appl. Stat. <strong>4</strong>: 503-519</a>.<br />
       (<a href="https://arxiv.org/abs/0903.2003">arXiv:0903.2003</a>)
</li>  

<li>
     <span class="sidenote">The cat score is implemented in the <a href="software/sda/index.html">sda</a> and <a href="software/st/index.html">st</a> R packages.</span>

     <strong>Gene ranking and biomarker discovery under correlation</strong>.<br />
        V. Zuber and K. Strimmer. 2009. <br />
       <a href="https://doi.org/10.1093/bioinformatics/btp460">Bioinformatics <strong>25</strong>: 2700-2707</a>.<br />
       (<a href="https://arxiv.org/abs/0902.0751">arXiv:0902.0751</a>)
</li>  



<li>
     <span class="sidenote">This method is implemented in the <a href="software/entropy/index.html">entropy</a> R package.</span>

     <strong>Entropy inference and the James-Stein estimator, with application
         to nonlinear gene association networks</strong>.<br />
        J. Hausser and K. Strimmer. 2009. <br />
       <a href="https://jmlr.csail.mit.edu/papers/v10/hausser09a.html">J. Mach. Learn. Res. <strong>10</strong>: 1469-1484</a>.<br />
       (<a href="https://arxiv.org/abs/0811.3579">arXiv:0811.3579</a>)
</li>  


<li>
    <span class="sidenote">
        <img src="images/highlyaccessed.gif" alt=
        "BMC highly accessed" style=
        "height: 12px;" />
    </span>

     <strong>A general modular framework for gene set enrichment analysis</strong>.<br />
        M. Ackermann and K. Strimmer. 2009. <br />
 <a href="https://doi.org/10.1186/1471-2105-10-47">BMC Bioinformatics <strong>10</strong>: 47</a>. 
</li>


<li> 
    <strong>Comparison of the Gaussian process classification approach with classical
            penalized classification methods in high-dimensional omics data</strong>. <br />
    M. Ahdesm&auml;ki and K. Strimmer. 2009. <br />
    Proceedings of the <a href="http://www.cs.tut.fi/wcsb09/">6th International Workshop on Computational Systems 
    Biology, WCSB 2009</a> (June 10-12, 2009, Aarhus, Denmark), <a href="publications/conferences/gp-wcsb2009.pdf">pp. 19-22.</a>
</li>

<li> 
    <strong>Correlation-adjusted t-scores in application to functional magnetic resonance imaging data</strong>. <br />
    V. Zuber and K. Strimmer. 2009. <br />
    Proceedings of the <a href="http://www.cs.tut.fi/wcsb09/">6th International Workshop on Computational Systems 
    Biology, WCSB 2009</a> (June 10-12, 2009, Aarhus, Denmark), <a href="publications/conferences/cat-fmri-wcsb2009.pdf">pp. 163-166.</a>
</li>

<li><strong>Genetic distances and nucleotide
      substitution models: Theory</strong>. <br />
      K. Strimmer and A. von Haeseler. 2009.<br />
      Chapter 4 in: P. Lemey, M. Salemi, A.M. Vandamme (eds.).
      <a href="http://www.cambridge.org/uk/catalogue/catalogue.asp?isbn=9780521877107">The 
      Phylogenetic Handbook 2nd edition.</a> Cambridge University
      Press, Cambridge, UK, <a href="https://doi.org/10.1017/CBO9780511819049.006">pp. 111-125</a><a href="publications/misc/handbook-chap4-2006.pdf">.</a> <br />
      (<a href="publications/misc/handbook-chap4-2009-typos.txt">typographic corrections</a>)
</li>


<li>
    <span class="sidenote">
        <img src="images/highlyaccessed.gif" alt=
        "BMC highly accessed" style=
        "height: 12px;" />
    </span>

    <strong>A unified approach to false discovery rate estimation</strong>.<br />
    K. Strimmer. 2008.  <br />
     <a href="https://doi.org/10.1186/1471-2105-9-303">BMC Bioinformatics <strong>9</strong>: 303</a>. <br />
    (<a href="publications/journals/fdr-estimation2008-typos.txt">typographic corrections</a>)
</li>


<li> 
     <span class="sidenote">Visit the <a href="software/fdrtool/index.html">fdrtool</a> homepage.</span>

       <strong>fdrtool: a versatile R package
       for estimating local and tail area- based false discovery rates</strong>. <br />
       K. Strimmer. 2008. <br />
     <a href="https://doi.org/10.1093/bioinformatics/btn209">Bioinformatics
    <strong>24</strong>: 1461-1462</a>.
</li>

 <li>  
    <strong>Connections between the augmented bootstrap
    and the shrinkage covariance estimator. Comment on: Augmenting the  bootstrap
    to analyze high- dimensional data</strong>. <br />
    K. Strimmer. 2008.<br /> <a href="https://doi.org/10.1007/s11749-008-0101-2">TEST 
    <strong>17</strong>: 25-27</a>.
</li>


<li>
 <span class="sidenote">
  
     <img src="images/wcsb08-cover.png" alt=
        "WCSB 2008 proceedings cover" style=
        "height: 102px;" /><br />
      Download the <a href="publications/conferences/wcsb2008.pdf">WCSB 2008 proceedings</a>.
    
   </span>

     <strong>Proceedings of the <a href="http://www.cs.tut.fi/wcsb08/">5th International 
    Workshop on Computational Systems 
    Biology</a>, WCSB 2008, June 11-13, 2008, Leipzig, Germany.</strong> <br />
    M. Ahdesm&auml;ki, K. Strimmer, N. Radde, J. Rahnenf&uuml;hrer,
    K. Klemm, H. L&auml;hdesm&auml;ki, and O. Yli-Harja (eds). 2008. <a href="publications/conferences/wcsb2008.pdf">201 pages.</a>
</li> 

<li>
     <strong>Therapeutic vaccination reduces HIV sequence variability</strong>. <br />
     D. Hoffmann, J. Seebach, A. Cosma, F. D. Goebel, K.
     Strimmer, H. M. Sch&auml;tzl, and V. Erfle. 2008.<br />
     <a href="https://doi.org/10.1096/fj.06-7975com">FASEB J. <strong>22</strong>:437-444</a>.
</li>

<li>
     <span class="sidenote">
        <img src="images/highlyaccessed.gif" alt=
        "BMC highly accessed" style=
        "height: 12px;" /><br />
      This method is available in the <a href="software/genenet/index.html">GeneNet</a> R package.
      </span>
       
      <strong>From correlation to causation networks: a simple approximate
      learning algorithm and its application to high-dimensional plant
      gene expression data</strong>. <br />
      R. Opgen-Rhein and K. Strimmer. 2007.  <br /> <a href=
      "https://doi.org/10.1186/1752-0509-1-37">BMC Syst. Biol.
      <strong>1</strong>: 37</a>. 
</li>


<li>
      
      <strong>Learning causal networks from systems biology time course
      data: an effective model selection procedure for the 
      vector autoregressive process</strong>. <br />
      R. Opgen-Rhein and K. Strimmer. 2007. <br />
      <a href="https://doi.org/10.1186/1471-2105-8-S2-S3">BMC Bioinformatics <strong>8</strong> 
      Suppl. 2: S3</a>.
</li>
   
 
<li>
      <span class="sidenote">The "shrinkage t" statistic is implemented in the <a href="software/st/index.html">st</a> R package.</span>
 
      
      <strong>Accurate ranking of differentially 
      expressed genes by a distribution- free shrinkage
      approach</strong>. <br /> R. Opgen-Rhein  and K. Strimmer. 2007. <br />
      <a href="https://doi.org/10.2202/1544-6115.1252">Statist. Appl. 
      Genet. Mol. Biol. <strong>6</strong>: 9</a>. <br />
      (<a href="publications/journals/shrinkt2007.pdf">original bepress version</a>, <a href="publications/journals/shrinkt2007-typos.txt">typographic corrections</a>)
</li>
 
<li>
   <span class="sidenote">The PLS analyzes were conducted using the R package
    <a href="https://cran.r-project.org/web/packages/plsgenomics/">plsgenomics</a>.</span>
    
      <strong>Partial least squares: a versatile tool for the analysis of
      high- dimensional genomic data</strong>. <br />
      A.-L. Boulesteix   and K. Strimmer. 2007. <br />
      <a href="https://doi.org/10.1093/bib/bbl016">Briefings 
      in Bioinformatics. 
      <strong>8</strong>: 32-44</a>. 
</li>    



   <li> 
      <strong>Inferring gene dependency networks from genomic longitudinal data:
      a functional data approach</strong>. <br />
      R. Opgen-Rhein and K. Strimmer. 2006.<br />
      <a href="http://www.ine.pt/revstat/pdf/rs060103.pdf">REVSTAT
      <strong>4</strong>:53-65</a>. 
    </li>    

 <li> 
    <strong>Using regularized dynamic correlation to infer gene dependency
    networks from time-series microarray data</strong>. <br />
    R. Opgen-Rhein and K. Strimmer. 2006. <br />
    Proceedings of the <a href="http://www.cs.tut.fi/wcsb06/">4th International Workshop on Computational Systems 
    Biology, WCSB 2006</a> (June 12-13, 2006, Tampere, Finland), <a href="publications/conferences/dyncorshrink2006.pdf">pp. 73-76.</a>  
      </li>

    <li>
    <strong>Reverse engineering genetic networks using the GeneNet package</strong>.<br />
    J. Sch&auml;fer, R. Opgen-Rhein,  and K. Strimmer. 2006.  <br />
    <a href="publications/misc/GeneNet-Rnews2006.pdf">R News
    <strong>6/5</strong>:50-53</a>. 
      </li>
  
 
     <li><strong>Book review of 
     "Statistical Methods in Bioinformatics: An Introduction", 
     2nd edition, Springer 2005, by W. J. Ewens and G. R. Grant.</strong> <br />
     K. Strimmer. 2006. <br />
     <a href="https://doi.org/10.1002/bimj.200510194">Biometr. J.
     <strong>48</strong>:178-179</a>. 
</li>


   <li>
      <span class="sidenote">See the <a href="software/corpcor/index.html">corpcor</a> R package
      for efficient estimation of large-scale covariance matrices and their inverses.</span>
      
      
      <strong>A shrinkage approach to large-scale covariance matrix
      estimation and implications for functional genomics</strong>. <br />
      J. Sch&auml;fer and K. Strimmer. 2005. <br /> 
      <a href="https://doi.org/10.2202/1544-6115.1175">Statist. Appl. 
      Genet. Mol. Biol. <strong>4</strong>: 32</a>.<br /> 
      (<a href="publications/journals/shrinkcov2005.pdf">original bepress version</a>)
    </li>    

    <li>
      
      <strong>Predicting transcription factor activities from combined analysis
      of microarray and ChIP data: a partial least squares approach</strong>. <br />
      A.-L. Boulesteix and K. Strimmer. 2005. <br />
      <a href="https://doi.org/10.1186/1742-4682-2-23">Theor. Biol.
      Med. Model. <strong>2</strong>: 23</a>.
    </li>

    
      <li>
      <span class="sidenote">See the <a href="software/genenet/index.html">GeneNet</a> R package.</span>

      <strong>An empirical Bayes approach to inferring large-scale gene association
      networks</strong>. <br /> 
      J. Sch&auml;fer and K. Strimmer. 2005. <br />
      <a href=
      "https://doi.org/10.1093/bioinformatics/bti062">Bioinformatics 
      <strong>21</strong>: 754-764</a>.
     </li>
 
      <li>
     <span class="sidenote">For rjMCMC inferences from DNA sequences
     use the <a href=
      "https://cran.r-project.org/web/packages/ape/">ape</a> software.</span>
  
      
      <strong>Inference of demographic history from genealogical trees using reversible
      jump Markov chain Monte Carlo</strong>. <br />
      R. Opgen-Rhein, L. Fahrmeir, and K. Strimmer. 2005. <br />
      <a href=
      "https://doi.org/10.1186/1471-2148-5-6">BMC Evol. Biol.
      <strong>5</strong>: 6</a>.
      </li>

 <li>
<span class="sidenote">This paper reviews procedures for GGM network inference.</span>
       
       <strong>Learning 
       large-scale graphical Gaussian models from genomic data</strong>.<br />
       J. Sch&auml;fer and K. Strimmer. 2005. <br /> 

       <a href="https://doi.org/10.1063/1.1985393">AIP Conf. Proc. <strong>776</strong>: 263-276</a>.
     </li>

      <li>
      <strong>APE: Analyses of phylogenetics and evolution in R language</strong>. <br />
      E. Paradis, J. Claude, and K. Strimmer. 2004. <br />
      <a href=
      "https://doi.org/10.1093/bioinformatics/btg412">Bioinformatics
      <strong>20</strong>: 289-290</a>.
     </li>

     <li>
     <span class="sidenote">
      Periodicity testing is available
      in the <a href="https://cran.r-project.org/web/packages/GeneCycle/">GeneCycle</a> R package.</span>
      
      <strong>Identifying periodically expressed transcripts in microarray time series
      data</strong>. <br />
      S. Wichert, K. Fokianos, and K. Strimmer. 2004. <br />
      <a href="https://doi.org/10.1093/bioinformatics/btg364">Bioinformatics
      <strong>20</strong>: 5-20</a>.<br /> 
      Corrigendum. 2008. <a href="https://doi.org/10.1093/bioinformatics/btn430">Bioinformatics <strong>24</strong>: 2274</a>. 
     </li>

  <li><strong>A brief introduction to the phylogenetic analysis
      library, version 1.5</strong>.  <br />
      M. Goode, K. Strimmer, A. Drummond, E. Buckler, and
      A. Rodrigo. 2004. <br />
      <a href="publications/conferences/pal2004.pdf">Pp. 175-179</a> in: Y.-P. Phoebe Chen (Ed.). 
      <a href="http://www.crpit.com/Vol29.html">CRPIT Volume 29</a>:
       Proceedings of the Second
      Asia-Pacific Bioinformatics Conference
      (APBC2004), Dunedin, NZ,
      January 2004.
  </li>

   <li><strong>Inference
      and applications of molecular phylogenies: an introductory guide</strong>. <br />
      K. Strimmer and D. L. Robertson. 2004. <br />
      <a href=
      "publications/misc/phylogeny2001.pdf">Chapter 5</a> 
      in: C. E. Sansom and R. M. Horton (eds.), <a href=
      "http://www.oup.com/uk/booksites/content/019963887X/">The Internet for
      Molecular Biologists</a> (Practical Approach Series). <a href=
      "http://www.oup.co.uk">Oxford University Press</a>, Oxford, UK. 
</li>

   
     <li>
      <strong>A CART-based approach to discover emerging patterns in microarray data</strong>. <br />
      A.-L. Boulesteix, G. Tutz, and K. Strimmer. 2003. <br />
      <a href="https://doi.org/10.1093/bioinformatics/btg361">Bioinformatics
      <strong>19</strong>: 2465-2472</a>. 
     </li>

      <li><strong>A novel exploratory method for visual recombination
      detection</strong>. <br />
      K. Strimmer, K. Forslund, B. Holland, and V.
      Moulton. 2003. <br />
      <a href="https://doi.org/10.1186/gb-2003-4-5-r33">Genome Biology
      <strong>4</strong>: R33</a>.
</li>

      <li><strong>Modeling gene expression
      measurement error: a quasi-likelihood approach</strong>.<br />
      K. Strimmer. 2003. <br /> <a href=
      "https://doi.org/10.1186/1471-2105-4-10">BMC Bioinformatics
      <strong>4</strong>: 10</a>.
      </li>

    <li><strong>Nucleotide
      substitution models</strong>. <br />
      K. Strimmer and A. von Haeseler. 2003.<br />
      <a href=
      "publications/misc/handbook-chap4-2001.pdf">Chapter 4</a> in: M. Salemi, A.M. Vandamme (eds.).
      <a href="http://www.isbnsearch.org/isbn/052180390X">The 
      Phylogenetic Handbook.</a> <a href="http://www.cup.org">Cambridge University
      Press</a>, Cambridge, UK. 
</li>

  <li><strong>Phylogeny
      inference based on maximum-likelihood methods with TREE-PUZZLE</strong>. <br />
      A. von Haeseler and K. Strimmer. 2003. <br /><a href=
      "publications/misc/handbook-chap6-2001.pdf">Chapter 6</a>
      in: M. Salemi, A.M. Vandamme (eds.). <a href=
      "http://www.isbnsearch.org/isbn/052180390X">The
      Phylogenetic Handbook.</a> <a href="http://www.cup.org">Cambridge University
      Press</a>, Cambridge, UK. 
</li>


 <li><strong>Comment on: Bayesian phylogenetic
      inference from animal mitochondrial genome rearrangements</strong>. <br />
      K. Strimmer. 2002.  <br /> <a href=
      "https://doi.org/10.1111/1467-9868.00359">J.
      R. Statist. Soc. B <strong>64</strong>: 745</a>.
</li>


     <li ><strong>TREE-PUZZLE: Maximum likelihood phylogenetic analysis
      using quartets and parallel computing</strong>. <br />
     H. A. Schmidt, K. Strimmer, M. Vingron, and A. von
      Haeseler. 2002. <br /> <a href=
      "https://doi.org/10.1093/bioinformatics/18.3.502">Bioinformatics
      <strong>18</strong>: 502-504</a>.
</li>

      <li><strong>Inferring
      confidence sets of possibly misspecified gene trees</strong>. <br />
      K. Strimmer and A. Rambaut. 2002. <br /> <a href=
      "https://doi.org/10.1098/rspb.2001.1862">Proc. R. Soc.
      Lond. B <strong>269</strong>: 137-142</a>.
</li>


     <li>
<span class="sidenote">The generalized skyline plot is implemented in <a href=
      "https://cran.r-project.org/web/packages/ape/">ape</a>.</span>
      <strong>Exploring the
      demographic history of DNA sequences using the generalized skyline plot</strong>.<br />
      K. Strimmer and O. G. Pybus. 2001. <br />
      <a href="http://mbe.oxfordjournals.org/cgi/content/abstract/18/12/2298">Mol. Biol. Evol.
      <strong>18</strong>: 2298-2305</a>. 
 </li>

      <li><strong>PAL: An
      object-oriented programming library for molecular evolution and
      phylogenetics</strong>. <br />
      A. Drummond and K. Strimmer. 2001.  <br /><a href=
      "https://doi.org/10.1093/bioinformatics/17.7.662">Bioinformatics
      <strong>17</strong>: 662-663</a>.
      </li>

      <li>
      <strong>Recombination analysis using directed graphical models</strong>.<br />
      K. Strimmer, C. Wiuf, and V. Moulton. 2001. <br /> <a href=
      "http://mbe.oxfordjournals.org/cgi/content/extract/18/1/97">Mol. Biol. Evol. <strong>18</strong>:
      97-99</a>.
</li>


  <li>
      <strong>Dating the
      common ancestor of SIVcpz and HIV-1 group M and the origin of HIV-1
      subtypes using a new method to uncover clock-like molecular evolution</strong>.<br />
      M. Salemi, K. Strimmer, W. W. Hall, M. Duffy, E. 
      Delaporte, S. Mboup, M. Peeters, and A.-M. Vandamme. 2001. <br />
      <a href="http://www.fasebj.org/cgi/content/full/15/2/276">FASEB J. <strong>15</strong>:276-278</a>.<br />
</li>


      <li><strong>Likelihood
      analysis of phylogenetic networks using directed graphical models</strong>.<br />
      K. Strimmer and V. Moulton. 2000. <br />
      <a href="http://mbe.oxfordjournals.org/cgi/content/abstract/17/6/875">Mol. Biol. Evol.
      <strong>17</strong>: 875-881</a>.
</li>

           



      <li>
      
      <strong>Likelihood mapping: a simple method to visualize phylogenetic content of a sequence
      alignment</strong>. <br />
      K. Strimmer  and A. von Haeseler. 1997. <br />
      <a href="http://www.pnas.org/cgi/content/abstract/94/13/6815">Proc. Natl. Acad. Sci. USA
      <strong>94</strong>: 6815-6819</a>.
 </li>

      <li>
      <strong>Bayesian probabilities and quartet puzzling</strong>. <br />
      K. Strimmer, N. Goldman, and A. von Haeseler. 1997. <br /> <a href=
      "http://mbe.oxfordjournals.org/content/14/2/210.full.pdf+html">Mol. Biol. Evol. <strong>14</strong>:
      210-211</a>.
</li>

     <li>

      <span class="sidenote">The <a href="http://tree-puzzle.de">TREE-PUZZLE</a> program
      implements "likelihood mapping" and "quartet puzzling" (among other things).</span>

      <strong>Quartet puzzling: a quartet maximum likelihood method for reconstructing 
      tree topologies</strong>. <br />
      K. Strimmer and A. von Haeseler. 1996. <br />
      <a href="http://mbe.oxfordjournals.org/content/13/7/964.full.pdf+html">Mol. Biol. Evol.
      <strong>13</strong>: 964-969</a>.<br /> (publisher PDF is defective - instead use this <a href=
      "publications/journals/qp1996.pdf">scan</a>)
</li>

      <li> <strong>Accuracy of
      neighbor joining for n-taxon trees</strong>. <br />
       K. Strimmer and A. von Haeseler. 1996. <br /><a href=
      "http://www.jstor.org/stable/2413528">Sys. Biol.
      <strong>45</strong>: 514-521</a>.
</li>


   </ul>

<p><strong>Theses:</strong></p>


 <ul>

<li>
<span class="sidenote">
  
     <img src="images/thesis1997-cover.png" alt=
        "KS thesis cover" style=
        "height: 114px;" />
    
   </span>
      <strong>Maximum likelihood methods in
      molecular phylogenetics</strong>. <br />
      K. Strimmer. 1997. <br />
      <a href=
      "publications/theses/phd-strimmer1997.pdf">Doctoral Thesis</a>. University of Munich. <a href=
      "http://www.utzverlag.de/shop.php?bn=31252">Herbert Utz Verlag</a>. 
</li>

     <li><strong>Modell-Theorie der Dynamik der
      Raum-Zeit auf der Basis unimodularer Drei-Metriken</strong>. <br />
      K. Strimmer. 1994. <br />
      <a href=
      "publications/theses/diplom-strimmer1994.pdf">Diploma Thesis</a>. University of Munich. 
</li>

   </ul>

    

    <p>&nbsp;</p>
    <p><em>Please note that all manuscript files are for private use only and may not
     be distributed without permission of the respective copyright owners.</em></p>
    
 </div>

<div id="footer">
<em>Last modified:</em>
        March 15, 2022<br />
        <a href="http://validator.w3.org/check?uri=https%3A%2F%2Fstrimmerlab.github.io%2Fkorbinian-publications.html">
           <img style="float: right;" src="images/valid-xhtml11.png" alt="Valid XHTML 1.1"
           height="31" width="88" /></a>


 
  </div>
</div>

</body>
</html>
