<?xml version="1.0" encoding="UTF-8" ?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.1//EN" 
   "http://www.w3.org/TR/xhtml11/DTD/xhtml11.dtd">

<html xmlns="http://www.w3.org/1999/xhtml">
<head>
  <meta http-equiv="content-type" content="text/html; charset=utf-8" />
  <meta name="KeyWords" content=
  "Korbinian Strimmer, biostatistics, biomathematics, statistical learning, bionformatics, genetic networks, big data" />
  <meta name="Author" content="Korbinian Strimmer" />
  <meta name="Description" content="Homepage" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />

  <title>Strimmer Lab - Research</title>

  <link rel="icon" href="images/favicon.ico" />
  <link rel="shortcut icon" href="images/favicon.ico" />
  <link rel="stylesheet" type="text/css" href="styles.css" />
</head>

<body>
  <script type="text/javascript">
//<![CDATA[
  <!--  
   if (top.location != self.location) {top.location = self.location}
  //-->
//]]>
</script>

  <div id="page">

   <div id="header"> 

      <a href="http://www.manchester.ac.uk/">
       <img style="border-width:0px;  margin-top: 4px; margin-left: 3px;"  
       src="images/logo-university-of-manchester.png" 
       alt="logo of The University of Manchester"/>
      </a>

       <img class="iconsize" 
        style="border-width:0px; float: right; margin-right: 1px; margin-top: 2px;"
        src="images/icon/microarray.png" alt="microarray logo" /> 
      <br />
      Statistical Learning and Data Science

  </div>


    <div id="nav">
       <a href="korbinian.html">Korbinian Strimmer</a> -
       <a href="http://www.maths.manchester.ac.uk/our-research/">Department of
          Mathematics</a>
       <br />
       <a href="index.html">Home</a> | 
       <a href="people.html">People</a> |
       <a href="courses.html">Teaching</a> |
       <a href="supervision.html">Supervision</a> |
       Research | 
       <a href="korbinian-publications.html">Publications</a>  
    </div>
 <div id="main">





<p><strong>Software:</strong></p>

    <p>On this page you find computer programs developed in our group.
       They are all free software (GNU GPL) and provide reference implementation
       for our methods.</p>

  
    <span class="sidenote">
    Most R packages have been deposited on <a href="http://cran.r-project.org">CRAN</a>, so that you
    can install them easily from within R.
    </span>
 



    <p><strong>R packages - statistical methods:</strong></p>
 
   <ul>
      <li><a href="software/binda/index.html">binda</a>: 
      Multi-class discriminant analysis using binary predictors.</li>

      <li><a href="software/care/index.html">care</a>: 
      High-dimensional regression and CAR score variable selection.</li>

      <li><a href="software/corpcor/index.html">corpcor</a>: 
      Efficient estimation of covariance and (partial) correlation.</li>

      <li><a href="http://cran.r-project.org/package=crossval">crossval</a>: 
      Generic functions for cross validation.</li>

      <li><a href="software/entropy/index.html">entropy</a>:  
      Estimation of entropy, mutual information and other related quantities.</li>

      <li><a href="software/fdrtool/index.html">fdrtool</a>:  
      Estimation of (local) false discovery rates and higher criticism.</li>

      <li><a href="software/longitudinal/index.html">longitudinal</a>: 
      Analysis of multiple time course data.</li>

      <li><a href="software/sda/index.html">sda</a>: 
      Shrinkage discriminant analysis and CAT score variable selection.</li>

     <li><a href="software/st/index.html">st</a>: Shrinkage
      t statistic and CAT score.</li>

     <li><a href="software/whitening/index.html">whitening</a>: Whitening and 
      high-dimensional canonical correlation analysis.</li>
 
   </ul>
 
   
    <p><strong>R packages - computational biology:</strong></p>

   <ul>

      <li><a href="http://bioconductor.org/packages/release/bioc/html/cleaver.html">cleaver</a>: Cleavage of polypeptide sequences.</li>

       <li><a href="http://cran.r-project.org/package=GeneCycle">GeneCycle</a>: 
      Identification of periodically expressed genes.</li>

      <li><a href="software/genenet/index.html">GeneNet</a>: 
      Modeling and inferring gene networks.</li>

      <li><a href="software/maldiquant/index.html">MALDIquant</a>: 
      Quantitative analysis of mass spectrometry data.</li>

      <li><a href="software/maldiquant/index.html">MALDIquantForeign</a>: 
      Import/export routines for MALDIquant.</li>

      <li><a href="http://cran.r-project.org/package=readBrukerFlexData">readBrukerFlexData</a>: 
      Read MS data in Bruker *flex format.</li>

      <li><a href="http://cran.r-project.org/package=readMzXmlData">readMzXmlData</a>: 
      Read MS data in mzXML format.</li>

 <li> Contributions to
      <a href= "http://cran.r-project.org/package=ape">ape</a> (phylogenetics in R) and
      <a href="http://cran.r-project.org/package=plsgenomics">plsgenomics</a>.
</li> 

 <li>R code for <a href="software/shrink-VAR-R-code.tar.gz">shrinkage estimation of  VAR gene network</a>.</li>
 
   </ul>


     <p><strong>Matlab/Octave code:</strong></p>
 
    <span class="sidenote">
    This code was kindly contributed by <a href="http://www.cs.ubc.ca/~murphyk/">Kevin P. Murphy</a>.
    </span>


     <ul>
     <li>Shrinkage covariance and correlation estimation 
        (<a href="software/m-files/covshrink-kpm/README">README</a>):<br />
 	 <a href="software/m-files/covshrink-kpm/covshrinkKPM.m">covshrinkKPM.m</a> (Matlab function),<br />
	 <a href="software/m-files/covshrink-kpm/testscript-matlab.txt">testscript-matlab.txt</a>,
	 <a href="software/m-files/covshrink-kpm/testscript-R.txt">testscript-R.txt</a> (test scripts),<br />
	 <a href="software/m-files/covshrink-kpm/smalldata.txt">smalldata.txt</a>,
	 <a href="software/m-files/covshrink-kpm/largedata.txt">largedata.txt</a> (example data files).<br />
     </li>
     </ul>
 
     <p><strong>Phylogenetics:</strong></p>
 
     <ul>
 
     <li><a href="http://www.tree-puzzle.de/">TREE-PUZZLE</a>:
      Maximum-likelihood tree reconstruction package.</li>

      <li><a href="https://web.archive.org/web/20041204185511/http://www.cebl.auckland.ac.nz/pal-project/download.html">PAL</a>: Java
      library for methods in molecular sequence analysis.</li>

      <li><a href="software/vanilla/index.html">Vanilla</a>: Java command
      line programs based on PAL.</li>

 
    </ul>
   


<p><strong>Data:</strong></p>

    <p>
    <span class="sidenote">Most of the data have been prepared for analysis using 
     <a href="http://www.r-project.org">R</a>. If part of an 
     R package they can be imported using the data() function otherwise using load().</span>
    This page compiles data sets that we have used
    in our papers.   We provide the data exactly in the form in which we have analyzed them ourselves
    (i.e. including our preprocessing, calibration etc.)
    in order to facilitate <a href="http://en.wikipedia.org/wiki/Reproducibility">reproducibility</a>.
    For further information concerning the raw data please consult
    the original authors and sources.
    </p>


<p><strong>From <a href="https://doi.org/10.1186/s12859-018-2572-9">Jendoubi and Strimmer, 2019</a> (CCA data integration):</strong></p>  
    <ul>
       <li>
<span class="sidenote">
  See the <a href="software/whitening/index.html">whitening</a> R package.
 </span>
       <a href="http://doi.org/10.1038/nature12634">The Cancer Genome Atlas</a> LUSC data:  see <a href="http://cran.r-project.org/web/packages/whitening/whitening.pdf">lusc</a> data.</li>
      <li> 

<a href="http://doi.org/10.1002/hep.21510">Martin et al. (2007)</a> nutrimouse study:  see <a href="http://cran.r-project.org/web/packages/whitening/whitening.pdf">nutrimouse</a> data.</li>
    </ul>


    <p><strong>From <a href="http://dx.doi.org/10.1093/bioinformatics/btv334">Gibb and Strimmer, 2015</a> (differential proteomics):</strong></p>  
    <ul>
       <li>
<span class="sidenote">
  See the <a href="software/binda/index.html">binda</a> and <a href="software/maldiquant/index.html">MALDIquant</a> R packages.
 </span>
       <a href="https://archive.ics.uci.edu/ml/datasets/Dorothea">Dorothea</a> NIPS 2003 drug discovery data set:   <a href="data/dorothea.rda">dorothea.rda</a>.</li>
      <li> 

<a href="http://dx.doi.org/10.1158/1078-0432.CCR-08-2701">Fiedler et al. (2009)</a> pancreas cancer proteomics data: <a href="https://github.com/sgibb/MALDIquantExamples/blob/master/inst/extdata/fiedler2009/spectra.tar.gz?raw=true">spectra.tar.gz</a> and <a href="https://github.com/sgibb/MALDIquantExamples/blob/master/inst/extdata/fiedler2009/spectra_info.csv?raw=true">spectra_info.csv</a></li>
    </ul>

 
    <p><strong>From <a href="http://dx.doi.org/10.2202/1544-6115.1730">Zuber and Strimmer, 2011</a> (CAR score):</strong></p>  
    <ul>
       <li>
<span class="sidenote">
  See the <a href="software/care/index.html">care</a> R package.
 </span>
       <a href="http://www.nature.com/nature/journal/v429/n6994/abs/nature02661.html">Lu et al. (2004)</a>
      brain aging gene expression data: see <a href="http://cran.r-project.org/web/packages/care/care.pdf">lu2004</a> data.</li>
    </ul>


    <p><strong>From 
       <a href="http://dx.doi.org/10.1093/bioinformatics/btp460">Zuber and Strimmer, 2009</a> (CAT score):</strong></p>  
    <ul>
       <li>
<span class="sidenote">
  See the <a href="software/sda/index.html">sda</a> R package.
 </span>
       <a href="http://www.nature.com/nature/journal/v457/n7231/abs/nature07762.html">Sreekumar et al. (2009)</a>
        Prostate cancer metabolomic data: <a href="data/sreekumar2009.rda.gz">sreekumar2009.rda.gz</a>.</li>
    </ul>



     <p><strong>From <a href="http://dx.doi.org/10.1186/1471-2105-8-S2-S3">Opgen-Rhein and Strimmer, 2007a</a> (VAR network) and 
             <a href="http://dx.doi.org/10.1186/1752-0509-1-37">2007b</a> (approximate causal network):</strong></p>  
    
    <ul>
       <li>
<span class="sidenote">
  See the <a href="software/genenet/index.html">GeneNet</a> R package.
 </span>

<a href="http://www.plantphysiol.org/cgi/content/abstract/136/1/2687/">Smith et al. (2004)</a>
      A. thaliana time series data: <a href="data/arth-smith2004.rda.gz">arth-smith2004.rda.gz</a>.</li>
      <li>See also the <a href="software/genenet/download/arabidopsis-net.R">R script for computing an approximate causal network</a> and the
<a href="software/shrink-VAR-R-code.tar.gz">R code for estimating 
the VAR network</a>.</li>
   
    </ul>
    
     <p><strong>From <a href="http://dx.doi.org/10.2202/1544-6115.1252">Opgen-Rhein and Strimmer, 2007</a> (shrinkage t statistic):</strong></p>  
   
    <ul>
      <li><a href="http://bioinformatics.oxfordjournals.org/cgi/content/abstract/20/3/323">Cope et al. (2004)</a>
      Affymetrix spike-in data: <a href="data/affyspike.rda.gz">affyspike.rda.gz</a>.</li>
    
      <li>
<span class="sidenote">
  See the <a href="software/st/index.html">st</a> R package.
    
   </span>


<a href="http://genomebiology.com/2005/6/2/r16/abstract/">Choe et al. (2005)</a>
      "golden spike" experiment: see <a href="http://cran.r-project.org/web/packages/st/st.pdf">choedata</a>.</li>
      
      <li><a href="http://jvi.asm.org/cgi/content/abstract/77/2/1392">van 't Wout et al. (2003)</a>
      HIV-1 data: <a href="data/vantwout-hiv1.rda.gz">vantwout-hiv1.rda.gz</a>.</li>
      
     </ul>
   
     
     <p><strong>From <a href="http://dx.doi.org/10.1186/1742-4682-2-23">Boulesteix and Strimmer, 2005</a> (PLS transcription factor prediction):</strong></p>  
   
    <ul>
    
       <li><a href="http://www.pnas.org/cgi/content/abstract/101/2/641">Kao et al. (2004)</a> Ecoli experiment:  
          <a href="data/kaoecoli.rda.gz">kaoecoli.rda.gz</a>.
      </li>
      <li><a href="http://www.pnas.org/cgi/content/abstract/100/26/15522">Liao et al. (2003)</a>
      hemoglobin data: <a href="data/hemoglobin.rda.gz">hemoglobin.rda.gz</a>.</li>
      
      <li><a href="http://www.molbiolcell.org/cgi/content/abstract/9/12/3273">Spellmann et al. (1998)</a> and 
      <a href="http://www.molbiolcell.org/cgi/content/abstract/11/12/4241">Gasch et al. (2000)</a> yeast data, with compatible 
      <a href="http://www.sciencemag.org/cgi/content/abstract/298/5594/799">Lee et al. (2002)</a>
      ChIP matrix: <a href="data/yeast-tfa.rda.gz">yeast-tfa.rda.gz</a>.
      </li>
      
      <li>Use this <a href="data/tfa-data.R">R script</a> to check that you retrieved the complete set of data.</li>
     </ul>
    
    <p><strong>From <a href="http://dx.doi.org/10.1093/bioinformatics/btg364">Wichert, Fokianos and Strimmer, 2004</a> (identification of cell cycle genes):</strong></p>  

    <ul>
      <li><a href="http://www.nature.com/ng/journal/v27/n1/abs/ng0101_48.html">Cho et al. (2001)</a>
      human fibrolast data: <a href="data/fibroblasts.rda.gz">fibroblasts.rda.gz</a>.</li>
     
      <li>
<span class="sidenote">
  See the <a href="http://cran.r-project.org/web/packages/GeneCycle/">GeneCycle</a> R package.
    
   </span>


<a href="http://www.sciencemag.org/cgi/content/abstract/290/5499/2144">Laub et al. (2000)</a>
      caulobacter data: <a href="http://cran.r-project.org/web/packages/GeneCycle/GeneCycle.pdf">caulobacter</a>.</li>
      <li><a href="http://www.molbiolcell.org/cgi/content/abstract/9/12/3273">Spellmann et al. (1998)</a>
      yeast data: <a href="data/spellmann-yeast.rda.gz">spellmann-yeast.rda.gz</a>.</li>
      <li><a href="http://www.molbiolcell.org/cgi/content/abstract/13/6/1977">Whitfield et al. (2002)</a>
       human HeLa data: <a href="data/humanhela.rda.gz">humanhela.rda.gz</a>.</li>
      <li>Use this <a href="data/cycle-data.R">R script</a> to check that you retrieved the complete set of data.</li>
      <li>Table 2 of Wichert et al. (2004) can be reproduced by using this <a href="data/wichert-table2.R">R script</a>.</li>
   
    </ul>
   
  
     <p><strong>From <a href="http://dx.doi.org/10.1186/1471-2105-4-10">Strimmer, 2003</a> (quasi-likelihood approach):</strong></p>  
    
    <ul>
       <li><a href="http://www.sciencemag.org/cgi/content/abstract/286/5439/531/">Golub et al. (1999)</a>
      leukemia data: <a href="data/golub.rda.gz">golub.rda.gz</a>.</li>   
    </ul>
    
    
    <p><strong>From diverse other papers:</strong></p>  
    <ul>
 <li>
<span class="sidenote">
 See the <a href="software/longitudinal/index.html">longitudinal</a> R package.
 </span>
<a href="http://bioinformatics.oxfordjournals.org/cgi/content/abstract/20/9/1361">Rangel et al. (2004)</a> 
      human T-cell data:  <a href="http://cran.r-project.org/web/packages/longitudinal/longitudinal.pdf">tcell</a>.</li>
   
     
      <li><a href="http://content.nejm.org/cgi/content/abstract/344/8/539">Hedenfalk et al. (2001)</a>
      breast cancer data: <a href="data/hedenfalk.rda.gz">hedenfalk.rda.gz</a>.</li>
      
      <li><a href="http://www.pnas.org/cgi/content/abstract/91/7/2621">Hedges (1994)</a>
      alignment: <a href="data/amniotes-seqs.txt">amniotes-seqs.txt</a> and 
      <a href="data/amniotes-info.txt">amniotes-info.txt</a>.</li>
          
    
      
      <li>
<span class="sidenote">
  See the <a href="software/genenet/index.html">GeneNet</a> R package.
 </span>

<a href="data/schmidt-heck2004.pdf">Schmidt-Heck et al. (2004)</a>
      E. coli data:  <a href="http://cran.r-project.org/web/packages/GeneNet/GeneNet.pdf">ecoli</a>.</li>


      <li><a href="http://www.springerlink.com/link.asp?id=rmfec4eceh4d3f4k">Tanaka et al. (2004)</a> HCV tree: 
      <a href="data/hcv.tre">hcv.tre</a> (Newick tree).</li>


      <li><a href="http://www.pnas.org/cgi/content/abstract/98/20/11462">West et al. (2001)</a>
      breast cancer data: <a href="data/westdataclean.rda.gz">westdataclean.rda.gz</a>.</li>

       <li>
<span class="sidenote">
 See the <a href="http://cran.r-project.org/web/packages/ape/">ape</a> R package.
 </span>
<a href="http://www.journals.royalsoc.ac.uk/link.asp?id=dx1w9pepux4dl5lv">Yusim et al. (2001)</a> HIV-1 tree: 
      <a href="http://cran.r-project.org/web/packages/ape/ape.pdf">hivtree</a>.</li>
      
    </ul>
    
    <p>
    If you miss a data set that we have analyzed in our papers please send email, and we will add it here.
    </p>
     


<p><strong>Methods:</strong></p>

<span class="sidenote">


<a href="images/research/methods.png"><img src="images/research/methods.png" width="140" alt="Overview of methods developed in the group" /></a><br />

Timeline of methods developed in our group
 (click image for a larger view)<br /> </span>

<p>Our research focuses on the development of state-of the art statistical and machine learning methods and tools
for the analysis of high-dimensional biomolecular data.  Correspondingly, over the last few years we contributed to 
development of methods for the following areas:</p>
<ul>
<li><a href="#highdimdata">High-dimensional data analysis</a></li>
<li><a href="#transprot">Methods for Transcriptomics and proteomics</a></li>
<li><a href="#ranking">Gene ranking and biomarker discovery</a></li>
<li><a href="#signal">Signal identification and FDR</a></li>
<li><a href="#networks">Graphical models and biological networks</a></li>
<li><a href="#molevol">Molecular evolution</a></li>
</ul>



<p>For reproducibility we make available our methods in <a href="software.html">free software</a>, mostly for the R platform.</p> 

<span class="sidenote">
   See also the <a href="http://www.researcherid.com/rid/C-1522-2009">ISI ResearcherID</a> and <a href="http://scholar.google.com/citations?user=Vezv6t0AAAAJ">Google Scholar</a> pages.
    </span>


<p>Please find below a representative selection of our publications.
For a complete list of publications and for current preprints see the <a href="korbinian-publications.html">publications of Korbinian Strimmer</a>
        and the web pages of the other <a href="people.html">group members</a>.</p>


<p><strong><a id="highdimdata">High-dimensional data analysis:</a></strong><br />
 <span class="sidenote">
   <a href="images/research/toolbox.png"><img src="images/research/toolbox.png" width="150" alt="big data toolbox" /></a><br />
Overview of our toolbox for analyzing high-dimensional data
 (click image for a larger view)<br /> 
    </span>

We are interested in methodology for high-dimensional 
inference, including <a href="courses/2010-11/current-topics/index.html">Bayesian learning</a>, regularization and <a href="software/corpcor/index.html">shrinkage methods</a>.  We study multiple testing 
approaches (such as <a href="software/fdrtool/index.html">false discovery rates</a> and higher criticism), are interested in model and variable selection for <a href="software/care/index.html">regression</a> and <a href="software/sda/index.html">classification</a>, and in computationally effective algorithms for
large-scale data.  Our methods are implemented in a toolbox for high-dimensional data analysis:</p>

<ul>
<li><a href="software/sda/index.html">sda</a>, <a href="software/care/index.html">care</a>, <a href="software/binda/index.html">binda</a>: high-dimensional regression and classification with variable selection, </li>
<li><a href="software/corpcor/index.html">corpcor</a>, <a href="software/entropy/index.html">entropy</a>: empirical Bayes and shrinkage estimation, and</li> 
<li><a href="software/fdrtool/index.html">fdrtool</a>: false discovery rate and higher criticism.</li>
</ul>

<p>
A common feature of our R packages is the avoidance of computationally expensive procedures and preference on scalable algorithms with low complexity.  Specifically, we rely on analytic approximation where possible (e.g. for optimizing tuning parameters) and also use effective analytic model selection strategies (higher criticism, CAR and CAT scores etc.).
This toolbox is used, e.g., in our software for gene network analysis
(<a href="software/genenet/index.html">GeneNet</a>), gene ranking (<a href="software/st/index.html">st</a>) and mass spectrometry (<a href="software/maldiquant/index.html">MALDIquant</a>).
</p>

<span class="sidenote">See the example R scripts <a href="software/r-scripts/whiten.R">whiten.R</a> and <a href="software/r-scripts/whiten-example.R">whiten-example.R</a>.</span>
<p>Recently, we have also discovered the unique <a href="http://arxiv.org/abs/1512.00809">optimal whitening procedure</a> for variable selection,
showing the optimality of the previously introduced <a href="software/sda/index.html">CAT</a> and <a href="software/care/index.html">CAR</a> scores.</p>

<p><strong><a id="transprot">Transcriptome and proteome analysis:</a></strong><br />
 <span class="sidenote">
   <a href="images/research/spectrum.png"><img src="images/research/spectrum.png" width="150" alt="example mass spectrum" /></a><br />
Example of a protein mass spectrum
 (click image for a larger view)<br /> 
    </span>

We are interested in statistical bioinformatics approaches
to analyze gene expression and proteomics data.
To this end we have developed a <a href="software/maldiquant/index.html">versatile platform for computational mass spectrometry</a>. Another current interest is the <a href="courses/2011-12/current-topics/index.html">analysis of RNA-Seq data</a>.
</p>
<ul>

<li>
<strong>Differential protein expression and peak selection in mass
spectrometry data by binary discriminant analysis</strong>.<br />
S. Gibb and K. Strimmer. 2015. <br />
<a href="http://dx.doi.org/10.1093/bioinformatics/btv334">Bioinformatics <strong>31</strong>: 3156-3162</a>.<br />
(<a href="http://arxiv.org/abs/1502.07959">arXiv:1502.07959</a>)

</li>

<li>
<strong>MALDIquant: a versatile R package for the analysis of
            mass spectrometry data</strong>.<br />
S. Gibb and K. Strimmer. 2012. <br />
<a href="http://dx.doi.org/10.1093/bioinformatics/bts447">Bioinformatics <strong>28</strong>: 2270-2271</a>.<br />
(<a href="http://arxiv.org/abs/1203.5885">arXiv:1203.5885</a>)
</li>
<li> 
 
    <strong>Analysis of proteomic data using MALDIquant</strong>. <br />
    S. Gibb and K. Strimmer. 2011. <br />
    Proceedings of the <a href="http://www.wcsb2011.ethz.ch/">8th International Workshop on Computational Systems 
    Biology, WCSB 2011</a> (June 6-8, 2011, Zurich, Switzerland), <a href="publications/conferences/maldiquant-wcsb2011.pdf">pp. 49-52.</a>
</li>
<li>
   

     <strong>A general modular framework for gene set enrichment analysis</strong>.<br />
        M. Ackermann and K. Strimmer. 2009. <br />
 <a href="http://dx.doi.org/10.1186/1471-2105-10-47">BMC Bioinformatics <strong>10</strong>: 47</a>. 
</li>

<li>    
      <strong>Partial least squares: a versatile tool for the analysis of
      high- dimensional genomic data</strong>. <br />
      A.-L. Boulesteix   and K. Strimmer. 2007. <br />
      <a href="http://dx.doi.org/10.1093/bib/bbl016">Briefings 
      in Bioinformatics. 
      <strong>8</strong>: 32-44</a>. 
</li>  


<li>
      
      <strong>Predicting transcription factor activities from combined analysis
      of microarray and ChIP data: a partial least squares approach</strong>. <br />
      A.-L. Boulesteix and K. Strimmer. 2005. <br />
      <a href="http://dx.doi.org/10.1186/1742-4682-2-23">Theor. Biol.
      Med. Model. <strong>2</strong>: 23</a>.
    </li>

<li>
     
      <strong>Identifying periodically expressed transcripts in microarray time series
      data</strong>. <br />
      S. Wichert, K. Fokianos, and K. Strimmer. 2004. <br />
      <a href="http://dx.doi.org/10.1093/bioinformatics/btg364">Bioinformatics
      <strong>20</strong>: 5-20</a>.<br /> 
      Corrigendum. 2008. <a href="http://dx.doi.org/10.1093/bioinformatics/btn430">Bioinformatics <strong>24</strong>: 2274</a>. 
     </li>
</ul>



<p><strong><a id="ranking">Gene ranking and biomarker discovery:</a></strong><br />
<span class="sidenote">
   <a href="images/research/car-models.png"><img src="images/research/car-models.png" width="150" alt="CAR regression models" /></a><br />
CAR regression models for diabetes data
 (click image for a larger view)<br /> 
    </span>

We are interested biomarker discovery and have recently proposed 
the CAT and CAR scores for ranking of correlated genes. In addition we introduced
the shrinkage t statistic, a regularized t-score useful in high-dimensional
data analysis with small samples:<br />
</p>
<ul>
<li>
<strong>A novel algorithm for simultaneous SNP selection in
high-dimensional genome-wide association studies</strong>.<br />
V. Zuber, A. P. Duarte Silva, and K. Strimmer. 2012. <br />
<a href="http://dx.doi.org/10.1186/1471-2105-13-284">BMC Bioinformatics <strong>13</strong>: 284</a>.<br />
(<a href="http://arxiv.org/abs/1203.3082">arXiv:1203.3082</a>)
</li>
<li>
<strong>High-dimensional regression and variable selection using CAR scores</strong>.<br />
V. Zuber and K. Strimmer. 2011. <br />
<a href="http://dx.doi.org/10.2202/1544-6115.1730">Statist. Appl. 
      Genet. Mol. Biol. <strong>10</strong>: 34</a>. <br />
      (<a href="http://arxiv.org/abs/1007.5516">arXiv:1007.5516</a>,
       <a href="publications/journals/carscore2011.pdf">original bepress version</a>)
</li>
<li>
 
     <strong>Gene ranking and biomarker discovery under correlation</strong>.<br />
        V. Zuber and K. Strimmer. 2009. <br />
       <a href="http://dx.doi.org/10.1093/bioinformatics/btp460">Bioinformatics <strong>25</strong>: 2700-2707</a>.<br />
       (<a href="http://arxiv.org/abs/0902.0751">arXiv:0902.0751</a>)
</li>  
<li>
 
      
      <strong>Accurate ranking of differentially 
      expressed genes by a distribution- free shrinkage
      approach</strong>. <br /> R. Opgen-Rhein  and K. Strimmer. 2007. <br />
      <a href="http://dx.doi.org/10.2202/1544-6115.1252">Statist. Appl. 
      Genet. Mol. Biol. <strong>6</strong>: 9</a>. <br />
      (<a href="publications/journals/shrinkt2007.pdf">original bepress version</a>, <a href="publications/journals/shrinkt2007-typos.txt">typographic corrections</a>)
</li>

</ul>





<p><strong><a id="signal">Signal identification and FDR:</a></strong><br />
 <span class="sidenote">
   <a href="images/research/thresh.png"><img src="images/research/thresh.png" width="150" alt="local FDR tresholds" /></a><br />
Local FDR thresholds and natural class boundary
 (click image for a larger view)<br /> 
    </span>

We have developed statistical approaches for detection of signal
in high-dimension genomic data and for multiple testing using
false discovery rates (FDR):<br />
</p>
<ul>
<li>B. Klaus and K. Strimmer. 2013.
<strong>Signal identification for rare and weak features: higher criticism 
or false discovery rates?</strong><br />
<a href="http://dx.doi.org/10.1093/biostatistics/kxs030">Biostatistics <strong>14</strong>: 129-143</a>.<br />
(<a href="http://arxiv.org/abs/1112.2615">arXiv:1112.2615</a>)
</li>
<li>
<strong>Learning false discovery rates by fitting sigmoidal threshold functions</strong>.<br />
B. Klaus and K. Strimmer. 2011. <br />
<a href="http://smf4.emath.fr/Publications/JSFdS/152_2/pdf/sfds_jsfds_152_2_39-50.pdf">Journal de la Soci&eacute;t&eacute;  Fran&ccedil;aise de Statistique <strong>152</strong>: 39-50</a>.<br />
(<a href="http://arxiv.org/abs/1104.5414">arXiv:1104.5414</a>)
</li>
<li>

     <strong>Feature selection in omics prediction problems using cat scores and false non-discovery rate control</strong>.<br />
        M. Ahdesm&auml;ki and K. Strimmer. 2010. <br />
<a href="http://dx.doi.org/10.1214/09-AOAS277">Ann. Appl. Stat. <strong>4</strong>: 503-519</a>.<br />
       (<a href="http://arxiv.org/abs/0903.2003">arXiv:0903.2003</a>)
</li>  
<li>
  
    <strong>A unified approach to false discovery rate estimation</strong>.<br />
    K. Strimmer. 2008.  <br />
     <a href="http://dx.doi.org/10.1186/1471-2105-9-303">BMC Bioinformatics <strong>9</strong>: 303</a>. <br />
    (<a href="publications/journals/fdr-estimation2008-typos.txt">typographic corrections</a>)
</li>

<li> 
 
       <strong>fdrtool: a versatile R package
       for estimating local and tail area- based false discovery rates</strong>. <br />
       K. Strimmer. 2008. <br />
     <a href="http://dx.doi.org/10.1093/bioinformatics/btn209">Bioinformatics
    <strong>24</strong>: 1461-1462</a>.
</li>


</ul>



<p><strong><a id="networks">Graphical models and biological networks:</a></strong><br />
<span class="sidenote">
   <a href="images/research/entropy-net.png"><img src="images/research/entropy-net.png" width="150" alt="entropy network" /></a><br />
Entropy-based gene association network
 (click image for a larger view)<br /> 
    </span>

In our group we have developed a series of algorithms using graphical
models for learning large-scale gene association networks from high-throughput
data:
</p>
<ul>
<li>
<strong>Introduction to graphical modelling</strong>.<br />
M. Scutari and K. Strimmer. 2011. <br />
Chapter 11 in: M. P. H. Stumpf, D. J. Balding, and M. Girolami (eds.).
      <a href="http://eu.wiley.com/WileyCDA/WileyTitle/productCd-0470710861.html">Handbook of Statistical Systems Biology</a>.  Wiley, Chichester, UK, <a href="http://dx.doi.org/10.1002/9781119970606.ch11">pp. 237-254</a>.<br />(<a href="http://arxiv.org/abs/1005.1036">arXiv:1005.1036</a>)
</li>
<li>
  
     <strong>Entropy inference and the James-Stein estimator, with application
         to nonlinear gene association networks</strong>.<br />
        J. Hausser and K. Strimmer. 2009. <br />
       <a href="http://jmlr.csail.mit.edu/papers/v10/hausser09a.html">J. Mach. Learn. Res. <strong>10</strong>: 1469-1484</a>.<br />
       (<a href="http://arxiv.org/abs/0811.3579">arXiv:0811.3579</a>)
</li>  
<li>
       
      <strong>From correlation to causation networks: a simple approximate
      learning algorithm and its application to high-dimensional plant
      gene expression data</strong>. <br />
      R. Opgen-Rhein and K. Strimmer. 2007.  <br /> <a href=
      "http://dx.doi.org/10.1186/1752-0509-1-37">BMC Syst. Biol.
      <strong>1</strong>: 37</a>. 
</li>

<li>
      
      <strong>Learning causal networks from systems biology time course
      data: an effective model selection procedure for the 
      vector autoregressive process</strong>. <br />
      R. Opgen-Rhein and K. Strimmer. 2007. <br />
      <a href="http://dx.doi.org/10.1186/1471-2105-8-S2-S3">BMC Bioinformatics <strong>8</strong> 
      Suppl. 2: S3</a>.
</li>

<li> 
      <strong>Inferring gene dependency networks from genomic longitudinal data:
      a functional data approach</strong>. <br />
      R. Opgen-Rhein and K. Strimmer. 2006.<br />
      <a href="http://www.ine.pt/revstat/pdf/rs060103.pdf">REVSTAT
      <strong>4</strong>:53-65</a>.
    </li> 

<li>
    <strong>Reverse engineering genetic networks using the GeneNet package</strong>.<br />
    J. Sch&auml;fer, R. Opgen-Rhein,  and K. Strimmer. 2006.  <br />
    <a href="publications/misc/GeneNet-Rnews2006.pdf">R News
    <strong>6/5</strong>:50-53</a>.
      </li>

<li>  
      <strong>A shrinkage approach to large-scale covariance matrix
      estimation and implications for functional genomics</strong>. <br />
      J. Sch&auml;fer and K. Strimmer. 2005. <br /> 
      <a href="http://dx.doi.org/10.2202/1544-6115.1175">Statist. Appl. 
      Genet. Mol. Biol. <strong>4</strong>: 32</a>.<br /> 
      (<a href="publications/journals/shrinkcov2005.pdf">original bepress version</a>)
    </li>    


<li>
      

      <strong>An empirical Bayes approach to inferring large-scale gene association
      networks</strong>. <br /> 
      J. Sch&auml;fer and K. Strimmer. 2005. <br />
      <a href=
      "http://dx.doi.org/10.1093/bioinformatics/bti062">Bioinformatics 
      <strong>21</strong>: 754-764</a>.
     </li>
 

</ul>



<p><strong><a id="molevol">Molecular evolution:</a></strong><br />
 <span class="sidenote">
   <a href="images/research/rjmcmc.png"><img src="images/research/rjmcmc.png" width="150" alt="rjMCMC estimate of population size" /></a><br />
Reversible jump MCMC estimate of population size
 (click image for a larger view)<br /> 
    </span>

One of our first research interests were methods for
phylogenetic analysis and population genetics using sequence data:
</p>
<ul>
 <li>
    
      
      <strong>Inference of demographic history from genealogical trees using reversible
      jump Markov chain Monte Carlo</strong>. <br />
      R. Opgen-Rhein, L. Fahrmeir, and K. Strimmer. 2005. <br />
      <a href=
      "http://dx.doi.org/10.1186/1471-2148-5-6">BMC Evol. Biol.
      <strong>5</strong>: 6</a>. 
      </li>

<li>
      <strong>APE: Analyses of phylogenetics and evolution in R language</strong>. <br />
      E. Paradis, J. Claude, and K. Strimmer. 2004. <br />
      <a href=
      "http://dx.doi.org/10.1093/bioinformatics/btg412">Bioinformatics
      <strong>20</strong>: 289-290</a>.
     </li>

<li><strong>Inferring
      confidence sets of possibly misspecified gene trees</strong>. <br />
      K. Strimmer and A. Rambaut. 2002. <br /> <a href=
      "http://dx.doi.org/10.1098/rspb.2001.1862">Proc. R. Soc.
      Lond. B <strong>269</strong>: 137-142</a>.
</li>

<li>

      <strong>Exploring the
      demographic history of DNA sequences using the generalized skyline plot</strong>.<br />
      K. Strimmer and O. G. Pybus. 2001. <br />
      <a href="http://mbe.oxfordjournals.org/cgi/content/abstract/18/12/2298">Mol. Biol. Evol.
      <strong>18</strong>: 2298-2305</a>.
 </li>

<li>
      
      <strong>Likelihood mapping: a simple method to visualize phylogenetic content of a sequence
      alignment</strong>. <br />
      K. Strimmer  and A. von Haeseler. 1997. <br />
      <a href="http://www.pnas.org/cgi/content/abstract/94/13/6815">Proc. Natl. Acad. Sci. USA
      <strong>94</strong>: 6815-6819</a>.
 </li>

 <li>

      <strong>Quartet puzzling: a quartet maximum likelihood method for reconstructing 
      tree topologies</strong>. <br />
      K. Strimmer and A. von Haeseler. 1996. <br />
      <a href="http://mbe.oxfordjournals.org/content/13/7/964.full.pdf+html">Mol. Biol. Evol.
      <strong>13</strong>: 964-969</a>.<br /> (publisher PDF is defective - instead use this <a href=
      "publications/journals/qp1996.pdf">scan</a>)
</li>
</ul>


 <p><strong><a id="conferences">Conferences:</a></strong><br />
We are hosting the meeting <a href="https://smpgd2017.wordpress.com/">Statistical Methods for Postgenomic Data (SMPGD 2017)</a> at Imperial College London. Previously, we helped organizing the life science session 
at <a href="http://www.gocps-leipzig2010.com/scientific-programme.html">GOCPS 2010</a> and the
<a href="http://web.archive.org/web/20081118073310/http://www.cs.tut.fi/wcsb08/payment.html">Computational Systems Biology (WCSB 2008)</a> conference
at the University of Leipzig. In Munich our group was coorganizer of the workshop
<a href="http://www.stat.uni-muenchen.de/sfb386/workshop/css2004/">Complex Stochastic Systems in Biology and Medicine 2004</a>.</p>

<br />
  
 </div>

<div id="footer">
<em>Last modified:</em>
        December 7, 2015<br />
        <a href="http://validator.w3.org/check?uri=referer">
           <img style="float: right;" src="images/valid-xhtml11.png" alt="Valid XHTML 1.1"
           height="31" width="88" /></a>


 
  </div>
</div>

</body>
</html>
