<?xml version="1.0" encoding="UTF-8" ?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.1//EN" 
   "https://www.w3.org/TR/xhtml11/DTD/xhtml11.dtd">

<html xmlns="https://www.w3.org/1999/xhtml">
<head>
  <meta http-equiv="content-type" content="text/html; charset=utf-8" />
  <meta name="KeyWords" content=
  "Korbinian Strimmer, biostatistics, biomathematics, statistical learning, bionformatics, genetic networks, big data" />
  <meta name="Author" content="Korbinian Strimmer" />
  <meta name="Description" content="Homepage" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />

  <title>Strimmer Lab - Research</title>

  <link rel="icon" href="images/favicon.ico" />
  <link rel="shortcut icon" href="images/favicon.ico" />
  <link rel="stylesheet" type="text/css" href="styles.css" />
</head>

<body>
  <script type="text/javascript">
//<![CDATA[
  <!--  
   if (top.location != self.location) {top.location = self.location}
  //-->
//]]>
</script>

  <div id="page">

   <div id="header"> 

      <a href="https://www.manchester.ac.uk/">
       <img style="border-width:0px;  margin-top: 4px; margin-left: 3px;"  
       src="images/logo-university-of-manchester.png" 
       alt="logo of The University of Manchester"/>
      </a>

       <img class="iconsize" 
        style="border-width:0px; float: right; margin-right: 1px; margin-top: 2px;"
        src="images/icon/microarray.png" alt="microarray logo" /> 
      <br />
      Statistical Learning and Data Science

  </div>


    <div id="nav">
       <a href="korbinian.html">Korbinian Strimmer</a> -
       <a href="https://www.maths.manchester.ac.uk/our-research/">Department of
          Mathematics</a>
       <br />
       <a href="index.html">Home</a> | 
       <a href="people.html">People</a> |
       <a href="courses.html">Teaching</a> |
       <a href="supervision.html">Supervision</a> |
       Research | 
       <a href="korbinian-publications.html">Publications</a>  
    </div>
 <div id="main">


<p><strong>Research interests:</strong></p>

<span class="sidenote">

<a href="images/research/methods.png"><img src="images/research/methods.png" width="140" alt="Overview of methods developed in the group" /></a><br />

Timeline of methods developed in our group
 (click image for a larger view)<br /> </span>

<p>
Our research focuses on statistical and machine learning methods and models   for
high-dimensional data  and to understand the underlying principles of learning from data.</p>


<p>We have developed computationally effective methods for
 high-dimensional regularized estimation using shrinkage methods,
for model and variable selection, for signal identification, for computing false discovery rates and performing dimension reduction,
for learning graphical models and for regularized classification and regression.
</p>
<p>
Biological applications of our methods have included 
the analysis of transcriptomics and proteomics data, gene ranking and biomarker discovery,
construction of biological networks and trees as well as modeling of molecular evolution.
</p>



 <span class="sidenote">
<a href="korbinian.html">Korbinian Strimmer</a> is recognized as 
<a href="https://publons.com/researcher/2863203/korbinian-strimmer/">"Highly Cited Researcher"</a>.
</span>

<p>
Our aim is to make all our work accessible and <a href="https://en.wikipedia.org/wiki/Reproducibility">reproducible</a> by providing corresponding free and open source <a href="#software">software</a> as well as 
analysis scripts and <a href="#datasets">data sets</a>.
</p>






<p><strong><a id="software">Software:</a></strong></p>

   
   
 



    <p><strong>R packages - statistical methods:</strong></p>

<span class="sidenote">
   <a href="images/research/toolbox.png"><img src="images/research/toolbox.png" width="150" alt="big data toolbox" /></a><br />
Statistical toolbox for analyzing high-dimensional data
 (click image for a larger view)<br /> 
    </span>
 
   <ul>
      <li><a href="software/binda/index.html">binda</a>: 
      Multi-class discriminant analysis using binary predictors.</li>

      <li><a href="software/care/index.html">care</a>: 
      High-dimensional regression and CAR score variable selection.</li>

      <li><a href="software/corpcor/index.html">corpcor</a>: 
      Efficient estimation of covariance and (partial) correlation.</li>

      <li><a href="https://cran.r-project.org/package=crossval">crossval</a>: 
      Generic functions for cross validation.</li>

      <li><a href="software/entropy/index.html">entropy</a>:  
      Estimation of entropy, mutual information and other related quantities.</li>

      <li><a href="software/fdrtool/index.html">fdrtool</a>:  
      Estimation of (local) false discovery rates and higher criticism.</li>

      <li><a href="software/longitudinal/index.html">longitudinal</a>: 
      Analysis of multiple time course data.</li>

      <li><a href="software/sda/index.html">sda</a>: 
      Shrinkage discriminant analysis and CAT score variable selection.</li>

     <li><a href="software/st/index.html">st</a>: Shrinkage
      t statistic and CAT score.</li>

     <li><a href="software/whitening/index.html">whitening</a>: Whitening and 
      high-dimensional canonical correlation analysis.</li>
 
   </ul>
 
   
    <p><strong>R packages - computational biology:</strong></p>

   <ul>


       <li><a href="https://cran.r-project.org/package=GeneCycle">GeneCycle</a>: 
      Identification of periodically expressed genes.</li>

      <li><a href="software/genenet/index.html">GeneNet</a>: 
      Modeling and inferring gene networks.</li>

      <li><a href="software/maldiquant/index.html">MALDIquant</a>: 
      Quantitative analysis of mass spectrometry data.</li>

      <li><a href="software/maldiquant/index.html">MALDIquantForeign</a>: 
      Import/export routines for MALDIquant.</li>

      <li><a href="https://cran.r-project.org/package=readBrukerFlexData">readBrukerFlexData</a>: 
      Read MS data in Bruker *flex format.</li>

      <li><a href="https://cran.r-project.org/package=readMzXmlData">readMzXmlData</a>: 
      Read MS data in mzXML format.</li>

 <li> Contributions to
      <a href= "https://cran.r-project.org/package=ape">ape</a> (phylogenetics in R) and
      <a href="https://cran.r-project.org/package=plsgenomics">plsgenomics</a>.
</li> 

 <li>R code for <a href="software/shrink-VAR-R-code.tar.gz">shrinkage estimation of  VAR gene network</a>.</li>
 
   </ul>


     <p><strong>Matlab/Octave code:</strong></p>
 
    <span class="sidenote">
    This code was kindly contributed by <a href="https://www.cs.ubc.ca/~murphyk/">Kevin P. Murphy</a>.
    </span>


     <ul>
     <li>Shrinkage covariance and correlation estimation 
        (<a href="software/m-files/covshrink-kpm/README">README</a>):<br />
 	 <a href="software/m-files/covshrink-kpm/covshrinkKPM.m">covshrinkKPM.m</a> (Matlab function),<br />
	 <a href="software/m-files/covshrink-kpm/testscript-matlab.txt">testscript-matlab.txt</a>,
	 <a href="software/m-files/covshrink-kpm/testscript-R.txt">testscript-R.txt</a> (test scripts),<br />
	 <a href="software/m-files/covshrink-kpm/smalldata.txt">smalldata.txt</a>,
	 <a href="software/m-files/covshrink-kpm/largedata.txt">largedata.txt</a> (example data files).<br />
     </li>
     </ul>
 
     <p><strong>Phylogenetics:</strong></p>
 
     <ul>
 
     <li><a href="http://www.tree-puzzle.de/">TREE-PUZZLE</a>:
      Maximum-likelihood tree reconstruction package.</li>

      <li><a href="https://web.archive.org/web/20041204185511/http://www.cebl.auckland.ac.nz/pal-project/download.html">PAL</a>: Java
      library for methods in molecular sequence analysis.</li>

      <li><a href="software/vanilla/index.html">Vanilla</a>: Java command
      line programs based on PAL.</li>

 
    </ul>
   


<p><strong><a id="datasets">Data sets:</a></strong></p>

    <p>
    Most of the data provided below are packaged as
   part of an
     <a href="https://www.r-project.org">R</a> package.
    For further information concerning the raw data please 
 consult the original data authors and papers. 
    </p>


<p><strong>From <a href="https://doi.org/10.1186/s12859-018-2572-9">Jendoubi and Strimmer, 2019</a> (CCA data integration):</strong></p>  
    <ul>
       <li>
<span class="sidenote">
  See the <a href="software/whitening/index.html">whitening</a> R package.
 </span>
       <a href="https://doi.org/10.1038/nature12634">The Cancer Genome Atlas</a> LUSC data:  see <a href="https://cran.r-project.org/web/packages/whitening/whitening.pdf">lusc</a> data.</li>
      <li> 

<a href="https://doi.org/10.1002/hep.21510">Martin et al. (2007)</a> nutrimouse study:  see <a href="https://cran.r-project.org/web/packages/whitening/whitening.pdf">nutrimouse</a> data.</li>
    </ul>


    <p><strong>From <a href="https://dx.doi.org/10.1093/bioinformatics/btv334">Gibb and Strimmer, 2015</a> (differential proteomics):</strong></p>  
    <ul>
       <li>
<span class="sidenote">
  See the <a href="software/binda/index.html">binda</a> and <a href="software/maldiquant/index.html">MALDIquant</a> R packages.
 </span>
       <a href="https://archive.ics.uci.edu/ml/datasets/Dorothea">Dorothea</a> NIPS 2003 drug discovery data set:   <a href="data/dorothea.rda">dorothea.rda</a>.</li>
      <li> 

<a href="https://dx.doi.org/10.1158/1078-0432.CCR-08-2701">Fiedler et al. (2009)</a> pancreas cancer proteomics data: <a href="https://github.com/sgibb/MALDIquantExamples/blob/master/inst/extdata/fiedler2009/spectra.tar.gz?raw=true">spectra.tar.gz</a> and <a href="https://github.com/sgibb/MALDIquantExamples/blob/master/inst/extdata/fiedler2009/spectra_info.csv?raw=true">spectra_info.csv</a></li>
    </ul>

 
    <p><strong>From <a href="https://dx.doi.org/10.2202/1544-6115.1730">Zuber and Strimmer, 2011</a> (CAR score):</strong></p>  
    <ul>
       <li>
<span class="sidenote">
  See the <a href="software/care/index.html">care</a> R package.
 </span>
       <a href="http://www.nature.com/nature/journal/v429/n6994/abs/nature02661.html">Lu et al. (2004)</a>
      brain aging gene expression data: see <a href="https://cran.r-project.org/web/packages/care/care.pdf">lu2004</a> data.</li>
    </ul>


    <p><strong>From 
       <a href="https://dx.doi.org/10.1093/bioinformatics/btp460">Zuber and Strimmer, 2009</a> (CAT score):</strong></p>  
    <ul>
       <li>
<span class="sidenote">
  See the <a href="software/sda/index.html">sda</a> R package.
 </span>
       <a href="http://www.nature.com/nature/journal/v457/n7231/abs/nature07762.html">Sreekumar et al. (2009)</a>
        Prostate cancer metabolomic data: <a href="data/sreekumar2009.rda.gz">sreekumar2009.rda.gz</a>.</li>
    </ul>



     <p><strong>From <a href="https://dx.doi.org/10.1186/1471-2105-8-S2-S3">Opgen-Rhein and Strimmer, 2007a</a> (VAR network) and 
             <a href="https://dx.doi.org/10.1186/1752-0509-1-37">2007b</a> (approximate causal network):</strong></p>  
    
    <ul>
       <li>
<span class="sidenote">
  See the <a href="software/genenet/index.html">GeneNet</a> R package.
 </span>

<a href="http://www.plantphysiol.org/cgi/content/abstract/136/1/2687/">Smith et al. (2004)</a>
      A. thaliana time series data: <a href="data/arth-smith2004.rda.gz">arth-smith2004.rda.gz</a>.</li>
      <li>See also the <a href="software/genenet/download/arabidopsis-net.R">R script for computing an approximate causal network</a> and the
<a href="software/shrink-VAR-R-code.tar.gz">R code for estimating 
the VAR network</a>.</li>
   
    </ul>
    
     <p><strong>From <a href="https://dx.doi.org/10.2202/1544-6115.1252">Opgen-Rhein and Strimmer, 2007</a> (shrinkage t statistic):</strong></p>  
   
    <ul>
      <li><a href="http://bioinformatics.oxfordjournals.org/cgi/content/abstract/20/3/323">Cope et al. (2004)</a>
      Affymetrix spike-in data: <a href="data/affyspike.rda.gz">affyspike.rda.gz</a>.</li>
    
      <li>
<span class="sidenote">
  See the <a href="software/st/index.html">st</a> R package.
    
   </span>


<a href="http://genomebiology.com/2005/6/2/r16/abstract/">Choe et al. (2005)</a>
      "golden spike" experiment: see <a href="https://cran.r-project.org/web/packages/st/st.pdf">choedata</a>.</li>
      
      <li><a href="http://jvi.asm.org/cgi/content/abstract/77/2/1392">van 't Wout et al. (2003)</a>
      HIV-1 data: <a href="data/vantwout-hiv1.rda.gz">vantwout-hiv1.rda.gz</a>.</li>
      
     </ul>
   
     
     <p><strong>From <a href="https://dx.doi.org/10.1186/1742-4682-2-23">Boulesteix and Strimmer, 2005</a> (PLS transcription factor prediction):</strong></p>  
   
    <ul>
    
       <li><a href="http://www.pnas.org/cgi/content/abstract/101/2/641">Kao et al. (2004)</a> Ecoli experiment:  
          <a href="data/kaoecoli.rda.gz">kaoecoli.rda.gz</a>.
      </li>
      <li><a href="http://www.pnas.org/cgi/content/abstract/100/26/15522">Liao et al. (2003)</a>
      hemoglobin data: <a href="data/hemoglobin.rda.gz">hemoglobin.rda.gz</a>.</li>
      
      <li><a href="http://www.molbiolcell.org/cgi/content/abstract/9/12/3273">Spellmann et al. (1998)</a> and 
      <a href="http://www.molbiolcell.org/cgi/content/abstract/11/12/4241">Gasch et al. (2000)</a> yeast data, with compatible 
      <a href="http://www.sciencemag.org/cgi/content/abstract/298/5594/799">Lee et al. (2002)</a>
      ChIP matrix: <a href="data/yeast-tfa.rda.gz">yeast-tfa.rda.gz</a>.
      </li>
      
      <li>Use this <a href="data/tfa-data.R">R script</a> to check that you retrieved the complete set of data.</li>
     </ul>
    
    <p><strong>From <a href="https://dx.doi.org/10.1093/bioinformatics/btg364">Wichert, Fokianos and Strimmer, 2004</a> (identification of cell cycle genes):</strong></p>  

    <ul>
      <li><a href="http://www.nature.com/ng/journal/v27/n1/abs/ng0101_48.html">Cho et al. (2001)</a>
      human fibrolast data: <a href="data/fibroblasts.rda.gz">fibroblasts.rda.gz</a>.</li>
     
      <li>
<span class="sidenote">
  See the <a href="https://cran.r-project.org/web/packages/GeneCycle/">GeneCycle</a> R package.
    
   </span>


<a href="http://www.sciencemag.org/cgi/content/abstract/290/5499/2144">Laub et al. (2000)</a>
      caulobacter data: <a href="https://cran.r-project.org/web/packages/GeneCycle/GeneCycle.pdf">caulobacter</a>.</li>
      <li><a href="http://www.molbiolcell.org/cgi/content/abstract/9/12/3273">Spellmann et al. (1998)</a>
      yeast data: <a href="data/spellmann-yeast.rda.gz">spellmann-yeast.rda.gz</a>.</li>
      <li><a href="http://www.molbiolcell.org/cgi/content/abstract/13/6/1977">Whitfield et al. (2002)</a>
       human HeLa data: <a href="data/humanhela.rda.gz">humanhela.rda.gz</a>.</li>
      <li>Use this <a href="data/cycle-data.R">R script</a> to check that you retrieved the complete set of data.</li>
      <li>Table 2 of Wichert et al. (2004) can be reproduced by using this <a href="data/wichert-table2.R">R script</a>.</li>
   
    </ul>
   
  
     <p><strong>From <a href="https://dx.doi.org/10.1186/1471-2105-4-10">Strimmer, 2003</a> (quasi-likelihood approach):</strong></p>  
    
    <ul>
       <li><a href="http://www.sciencemag.org/cgi/content/abstract/286/5439/531/">Golub et al. (1999)</a>
      leukemia data: <a href="data/golub.rda.gz">golub.rda.gz</a>.</li>   
    </ul>
    
    
    <p><strong>From diverse other papers:</strong></p>  
    <ul>
 <li>
<span class="sidenote">
 See the <a href="software/longitudinal/index.html">longitudinal</a> R package.
 </span>
<a href="http://bioinformatics.oxfordjournals.org/cgi/content/abstract/20/9/1361">Rangel et al. (2004)</a> 
      human T-cell data:  <a href="https://cran.r-project.org/web/packages/longitudinal/longitudinal.pdf">tcell</a>.</li>
   
     
      <li><a href="http://content.nejm.org/cgi/content/abstract/344/8/539">Hedenfalk et al. (2001)</a>
      breast cancer data: <a href="data/hedenfalk.rda.gz">hedenfalk.rda.gz</a>.</li>
      
      <li><a href="https://doi.org/10.1073/pnas.91.7.2621">Hedges (1994)</a>
      alignment: <a href="data/amniotes-seqs.txt">amniotes-seqs.txt</a> and 
      <a href="data/amniotes-info.txt">amniotes-info.txt</a>.</li>
          
    
      
      <li>
<span class="sidenote">
  See the <a href="software/genenet/index.html">GeneNet</a> R package.
 </span>

<a href="data/schmidt-heck2004.pdf">Schmidt-Heck et al. (2004)</a>
      E. coli data:  <a href="https://cran.r-project.org/web/packages/GeneNet/GeneNet.pdf">ecoli</a>.</li>


      <li><a href="http://www.springerlink.com/link.asp?id=rmfec4eceh4d3f4k">Tanaka et al. (2004)</a> HCV tree: 
      <a href="data/hcv.tre">hcv.tre</a> (Newick tree).</li>


      <li><a href="https://doi.org/10.1073/pnas.201162998">West et al. (2001)</a>
      breast cancer data: <a href="data/westdataclean.rda.gz">westdataclean.rda.gz</a>.</li>

       <li>
<span class="sidenote">
 See the <a href="https://cran.r-project.org/web/packages/ape/">ape</a> R package.
 </span>
<a href="http://www.journals.royalsoc.ac.uk/link.asp?id=dx1w9pepux4dl5lv">Yusim et al. (2001)</a> HIV-1 tree: 
      <a href="https://cran.r-project.org/web/packages/ape/ape.pdf">hivtree</a>.</li>
      
    </ul>
    
    
 <p><strong><a id="conferences">Conferences:</a></strong><br />

<p>Our group was organizer of the meeting <a href="https://smpgd2017.wordpress.com/">Statistical Methods for Postgenomic Data (SMPGD 2017)</a> at Imperial College London. Previously, we helped organizing the life science session 
at <a href="https://web.archive.org/web/20120528103341/http://www.gocps-leipzig2010.com/scientific-programme.html">GOCPS 2010</a> and the
<a href="https://web.archive.org/web/20081119074450/http://www.cs.tut.fi/wcsb08/index.html">Computational Systems Biology (WCSB 2008)</a> conference
at the University of Leipzig. At LMU Munich our group was coorganizer of the workshop
<a href="https://web.archive.org/web/20100121102953/http://www.stat.uni-muenchen.de/sfb386/workshop/css2004/">Complex Stochastic Systems in Biology and Medicine 2004</a>.</p>    

  
 </div>

<div id="footer">
<em>Last modified:</em>
        December 7, 2020<br />
        <a href="https://validator.w3.org/check?uri=https%3A%2F%2Fstrimmerlab.github.io%2Fresearch.html">
           <img style="float: right;" src="images/valid-xhtml11.png" alt="Valid XHTML 1.1"
           height="31" width="88" /></a>


 
  </div>
</div>

</body>
</html>
