<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN"
    "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">

<html xmlns="http://www.w3.org/1999/xhtml">
<head>
  <link rel="stylesheet" type="text/css" href="../../styles.css" />

  <title>MALDIquant: An R Package for Quantitative Analysis of Mass Spectrometry Data</title>
  <meta http-equiv="Content-Type" content=
  "text/html; charset=us-ascii" />
</head>

<body>

  <div id="page">

  <div id="header">
  MALDIquant: Quantitative Analysis of Mass Spectrometry Data
  <img class="toplogo" src="logo.jpg" alt="[R logo]" height="30"/>
  </div>

  <div id="main">

    <span class="sidenote">
    <a href=
    "http://cran.r-project.org/web/packages/MALDIquant/">
    MALDIquant</a> and
<a href=
    "http://cran.r-project.org/web/packages/MALDIquantForeign/">
    MALDIquantForeign</a> pages on CRAN. <br /><br />

<a href="https://github.com/sgibb/MALDIquantExamples/">MALDIquantExamples</a> page on GitHub.

    </span>


  <p>MALDIquant provides a complete analysis pipeline for MALDI-TOF and other mass spectrometry data. Distinctive features include baseline subtraction methods such as TopHat or SNIP, peak alignment using warping functions, handling of replicated measurements as well as allowing spectra with different resolutions.
</p>


  <p>The MALDIquant pipeline consists of two main R packages:  </p>
  <ul>
   <li><a href=
    "http://cran.r-project.org/web/packages/MALDIquant/">
    MALDIquant</a> contains the base functionality for processing mass spectrometry data. </li>
   <li> <a href=
    "http://cran.r-project.org/web/packages/MALDIquantForeign/">
    MALDIquantForeign</a> contains routines for importing and exporting data.</li>

  </ul>

<p>Furthermore,</p>
<ul>
<li> <a href="https://github.com/sgibb/MALDIquantExamples/">MALDIquantExamples</a> provides  examples
of real case studies using MALDIquant.
 </li>

</ul>

 <p>Additionally,</p>

    <span class="sidenote">
Please visit the <a href="../binda/index.html">binda</a> page for corresponding
example R scripts.
   </span>
 <ul>
   <li> <a href="http://strimmerlab.org/software/binda/">binda</a> implements a
binary discriminant approach useful for classification with mass spectrometry data
as described in <a href="http://arxiv.org/abs/1502.07959">Gibb and Strimmer (2015)</a>.
 </li>

</ul>

<p>All softwares are licensed under the <a href="http://www.gnu.org/licenses/gpl-3.0.txt">GNU GPL version 3</a> (or any later version).</p>

  <p><strong>Author:</strong> <a href="http://www.sebastiangibb.de/research.html">Sebastian Gibb</a>.</p>

  <p><strong>Example R Code:</strong></p>

<p>To illustrate the application of MALDIquant for analyzis of mass spectrometry data please find below a number of example R scripts: </p>

<p>Demos contained in <a href=
    "http://cran.r-project.org/web/packages/MALDIquant/">
    MALDIquant</a> package:</p>

  <ul>

<li>Overview over MALDIquant mass spectrometry analysis pipeline -
<a href="http://cran.r-project.org/web/packages/MALDIquant/vignettes/MALDIquant-intro.pdf">vignette (PDF)</a>
.</li>


<li>Example workflow with multiple spectra and alignment - <a href="https://raw.github.com/sgibb/MALDIquant/master/demo/workflow.R">demo R code</a>.</li>

    <li>
Preprocessing and peak detection example for a single
spectrum -
<a href="https://raw.github.com/sgibb/MALDIquant/master/demo/peaks.R">demo R code</a>:<br />
       <a href="peaks.png"><img src="peaks-small.png"  width="98%" alt="Preprocessing and peak detection" /></a> <br /> (click on image to enlarge). </li>

  <li>Comparison of different baseline corrections - <a href="https://raw.github.com/sgibb/MALDIquant/master/demo/baseline.R">demo R code</a>.</li>
  <li>Illustration of peak alignment by warping - <a href="https://raw.github.com/sgibb/MALDIquant/master/demo/warping.R">demo R code</a>.</li>


</ul>

<p>Examples from  <a href="http://cran.r-project.org/web/packages/MALDIquantForeign/">MALDIquantForeign</a>:</p>
<ul>
<li>Overview over functions for data import - <a href="http://cran.r-project.org/web/packages/MALDIquantForeign/vignettes/MALDIquantForeign-intro.pdf">vignette (PDF)</a>.</li>
</ul>

<div class="sidenote">
    To run these analyses without installing  <a href="https://github.com/sgibb/MALDIquantExamples/">MALDIquantExamples</a> use the following R scripts:
  <ol>
  <li><a href="download/nyakas2013-standalone.R">nyakas2013-standalone.R</a>,</li>
  <li><a href="download/fiedler2009-standalone.R">fiedler2009-standalone.R</a>,  </li>
  <li><a href="download/species-standalone.R">species-standalone.R</a>.  </li>
  </ol>

</div>

<p>Examples contained in <a href="https://github.com/sgibb/MALDIquantExamples/">MALDIquantExamples</a>:</p>


<ol>
<li> Mass Spectrometry Imaging (MSI) example - see the <a href="https://raw.githubusercontent.com/sgibb/MALDIquantExamples/master/inst/doc/nyakas2013.pdf">vignette (PDF)</a> and corresponding <a href="https://raw.githubusercontent.com/sgibb/MALDIquantExamples/master/inst/doc/nyakas2013.R">R code</a>.
</li>

<li>Example analysis of <a href="http://dx.doi.org/10.1158/1078-0432.CCR-08-2701">Fiedler et al. (2009)</a> pancreas cancer data (including peak filtering, feature selection and classification using SDA) - see the <a href="https://raw.githubusercontent.com/sgibb/MALDIquantExamples/master/inst/doc/fiedler2009.pdf">vignette (PDF)</a> and corresponding
<a href="https://raw.githubusercontent.com/sgibb/MALDIquantExamples/master/inst/doc/fiedler2009.R">R code</a>.
For an alternative analysis script using binary discriminant analysis see the <a href="http://strimmerlab.org/software/binda/">binda</a> page.
</li>

<li>Example cluster analysis of 96 MALDI-TOF mass spectra of four different bacteria
species (incl. peak selection) - see the <a href="https://raw.githubusercontent.com/sgibb/MALDIquantExamples/master/inst/doc/species.pdf?raw=true">vignette (PDF)</a> and corresponding
<a href="https://raw.githubusercontent.com/sgibb/MALDIquantExamples/master/inst/doc/species.R">R code</a>.
</li>


  </ol>

  <p><strong>Installation:</strong></p>

 <span class="sidenote">
    On <a href="http://debian.org">Debian GNU/Linux</a>  install
<a href="http://packages.debian.org/de/sid/r-cran-maldiquant">r-cran-maldiquant</a>
and <a href="http://packages.debian.org/de/sid/r-cran-maldiquantforeign">r-cran-maldiquantforeign</a>.
</span>

<p>For installing the two main packages simply enter at the R console:</p>
<p style="text-align:center;font-family:monospace">install.packages(c("MALDIquant", "MALDIquantForeign")) </p>

    <span class="sidenote">
   To install <a href="http://cran.r-project.org/web/packages/devtools/">devtools</a> run:<br />
   install.packages("devtools")
</span>


<p>For  <a href="https://github.com/sgibb/MALDIquantExamples/">MALDIquantExamples</a> you need to use the <a href="http://cran.r-project.org/web/packages/devtools/">devtools</a> R package:</p>
<p style="text-align:center;font-family:monospace">
library("devtools")<br />
install_github("sgibb/MALDIquantExamples")</p>

<p>Note that you can run the examples provided in <a href="https://github.com/sgibb/MALDIquantExamples/">MALDIquantExamples</a> also without installing the actual package (see above).</p>



  <p><strong>Documentation:</strong></p>

<p>The <strong>MALDIquant</strong> R package offers the core functions for mass spectrometry analysis, including base classes, plotting, base-line correction, peak identification, alignment etc.</p>  <p>The current version of MALDIquant is 1.12  (published June 2015) :</p>

  <ul><li>
<a href=
    "http://cran.r-project.org/web/packages/MALDIquant/NEWS">Release history</a>.
</li>

    <li><a href="http://cran.r-project.org/web/packages/MALDIquant/MALDIquant.pdf">Manual</a>     and <a href="http://cran.r-project.org/web/packages/MALDIquant/vignettes/MALDIquant-intro.pdf">vignette</a> (pdf files). </li>

    <li>Download of <a href=
    "http://cran.r-project.org/src/contrib/MALDIquant_1.12.tar.gz">MALDIquant version 1.12</a> source
    package.</li>

    <li>Archive of <a href="http://cran.r-project.org/src/contrib/Archive/MALDIquant/">previous versions</a> of MALDIquant.</li>

  </ul>


<p> The <strong>MALDIquantForeign</strong> package provides additional functions for reading (tab, csv, fid, mzXML, mzML, imzML, Analyze 7.5) and writing common file formats for mass spectrometry data into/from MALDIquant objects.</p>

  <p>The current version of MALDIquantForeign is 0.9 (published August 2014):</p>

  <ul>
    <li><a href=
    "http://cran.r-project.org/web/packages/MALDIquantForeign/NEWS">Release history</a>.  </li>

    <li><a href="http://cran.r-project.org/web/packages/MALDIquantForeign/MALDIquantForeign.pdf">Manual</a> and <a href="http://cran.r-project.org/web/packages/MALDIquantForeign/vignettes/MALDIquantForeign-intro.pdf">vignette</a> (pdf files).</li>

    <li>Download of <a href=
    "http://cran.r-project.org/src/contrib/MALDIquantForeign_0.9.tar.gz">MALDIquantForeign version 0.9</a> source
    package.</li>

    <li>Archive of <a href="http://cran.r-project.org/src/contrib/Archive/MALDIquantForeign/">previous versions</a> of MALDIquantForeign.</li>



<li>Associated R packages for data import: <a href="http://cran.r-project.org/web/packages/readBrukerFlexData/">readBrukerFlexData</a> and  <a href="http://cran.r-project.org/web/packages/readMzXmlData/">readMzXmlData</a>. </li>

  </ul>


  <p><strong>Reference Publication:</strong></p>

   <ul>
  <li><em>Please cite this paper if you use MALDIquant:</em> <br /> S. Gibb and K. Strimmer. 2012.
          MALDIquant: a versatile R package for the analysis of
            mass spectrometry data. <a href="http://bioinformatics.oxfordjournals.org/content/28/17/2270.abstract">Bioinformatics <strong>28</strong>: 2270-2271</a> (<a href="http://arxiv.org/abs/1203.5885">arXiv:1203.5885</a>)</li>

    <li><a href="https://raw.githubusercontent.com/sgibb/MALDIquantExamples/master/R/createFigure1.R">R code to reproduce Figure 1</a> (<a href="https://raw.githubusercontent.com/sgibb/MALDIquantExamples/master/R/createFigure1_color.R">color version</a>). <br />

 <a href="figure1.png"><img src="figure1-small.png" width="98%" alt="Figure 1" /></a> <br /> (click on image to enlarge).
</li>
   </ul>

  <p><strong>Further material:</strong></p>


  <ul>

<li>Talk at the <a href="http://ems2013.eu">29th European Meeting of Statisticians</a> (Budapest July 2013): <a href="download/MALDIquantBudapest2013.pdf">Quantitative Analysis of Proteomics Mass Spectrometry Data</a> (size of PDF: 3.5 MB) .</li>

 <li>S. Gibb  and K. Strimmer. 2011.
      <a href="../../publications/conferences/maldiquant-wcsb2011.pdf">Analysis of proteomic data using MALDIquant</a>.
      Proceedings of the <a href="http://www.wcsb2011.ethz.ch/">8th International Workshop on Computational Systems Biology, WCSB 2011</a> (June 6-8, 2011, Zurich, Switzerland), pp. 49-52. </li>


</ul>






  <p><strong>Related Software:</strong></p>

<p>The <a href="http://cran.r-project.org/web/packages/MALDIquant/">MALDIquant</a> and <a href="http://cran.r-project.org/web/packages/MALDIquantForeign/">MALDIquantForeign</a> packages are  part of the Bioconductor <a href="http://www.bioconductor.org/packages/release/data/experiment/html/RforProteomics.html">RforProteomics</a> analysis pipeline developed by <a href="http://proteome.sysbiol.cam.ac.uk/lgatto/">Laurent Gatto</a>.</p>



<p>
The <a href="http://sing.ei.uvigo.es/mass-up/">Mass-Up</a> software developed at the University of Vigo uses some functions of MALDIquant.</p>

<p>
<a href="http://www.lababi.bioprocess.org/index.php/14-sample-data-articles/78-massypup">MASSyPup - The Mass Spectrometry Live Linux</a> developed by <a href="http://www.ira.cinvestav.mx/Investigaci%C3%B3n/Biotecnolog%C3%ADayBioqu%C3%ADmica/ProfesoresInvestigadores/DrRobertWinkler/tabid/626/language/es-MX/Default.aspx">Robert Winkler</a> comes with many useful free software for mass spectrometry analysis
and also includes MALDIquant and MALDIquantForeign.

</p>


 <p>See also our list of <a href="../../notes/mass-spectrometry.html"> open source tools for mass spectrometry
analysis</a> (mostly in R).</p>

</div>

<div id="footer">

  Back to <a href="../../software.html">software</a>
  page.</div>

  </div>
</body>
</html>
