<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html>
  <head>
    <title>
    Statistical Analysis of Gene Expression Data: An
    Introduction</title>
    <meta http-equiv="Content-Type" content=
    "text/html; charset=iso-8859-1">
    <meta name="Author" content="Korbinian Strimmer">
    <meta name="Description" content=
    "course, statistical analysis,  gene expression, microarray, DNA chip">
  </head>

  <body bgcolor="#FFFFFF" link="#0000BF" vlink="#555555" alink="#0000BF">
    <table align="center" width="660">
      <tr>
        <td align="LEFT">
          <h1 align="center"><a href="http://webinfo.campus.lmu.de//view_event.cfm?ev=34834&sort=type">Statistical Analysis of Gene
          Expression Data:<br> An Introduction</a></h1>

          <h3 align="center">Summer 2002</h3>

          <h3 align="center">Seminar course at the Department of
          Statistics, University of Munich</h3>

          <h4 align="center">Organised by the  <a
          href=
          "https://strimmerlab.github.io">Statistics and Computational Biology</a></h4>
          <br>
        </td>
      </tr>

      <tr>
        <td align="LEFT">
          <p>&nbsp;</p>
        </td>
      </tr>

      <tr>
        <td align="LEFT">
          <table cols="2" width="100%">
            <tr valign="top">
              <td width="15%"><u>Contact:</u></td>

              <td><a href=
              "https://strimmerlab.github.io/korbinian.html">Korbinian
              Strimmer</a>  <br>
               <a href=
              "http://www.stat.uni-muenchen.de/~fahrmeir/welcome-e.html">
              Ludwig Fahrmeir</a><br>
              
               <br>
               <em>To register please send email to <a href=
              "mailto:strimmer@stat.uni-muenchen.de">strimmer@stat.uni-muenchen.de</a></em>.
	      <p>
	      </td>
              
	    </tr>

            <tr valign="top">
              <td><u>Time:</u></td>

              <td>Every Monday at 17:15 during summer term (exception: Friday 17.5)
              2002.</td>
            </tr>

            <tr valign="top">
              <td><u>Place:</u></td>

              <td>Seminar room, 1st floor, Ludwigstr. 33.</td>
            </tr>

            <tr valign="top">
              <td><u>"Schein":</u></td>

              <td>Requires full attendance and
              the presentation of a talk.</td>
            </tr>
          </table>
        </td>
      </tr>

      <tr>
        <td align="LEFT">
          <p>&nbsp;</p>
        </td>
      </tr>

      <tr>
        <td align="LEFT">
          <p><em>Overview:</em></p>

          

          <p>The seminar is intended as an introduction to the
	  analysis of gene expression data. We will survey the most
          important methodology and review some influential
          applications of this new biotechnology. The course is
          organized in 12 lectures, each of which deals with a
          specified topic in gene expression analysis.
          Please see below for the reading list - each article has
          a link and can be downloaded from the net (you can also obtain
	  a paper copy from the organizers).
           </p>

          <p>Talks can be in German or English. All participants
          are encouraged to read all the papers relevant for each
          session - we want to discuss the papers together!</p>
        </td>
      </tr>

      <tr>
        <td align="LEFT">
          <p>&nbsp;</p>
        </td>
      </tr>

      <tr>
        <td align="LEFT">
          <p><em>Time Table:</em></p>

          <p>At the first meeting (15.4.2002) we will discuss the
          overall organization and fix volunteer speakers for the
          following 12 sessions. All participants are encouraged to
          present a talk. On
          Monday 3.6.2002 we welcome Dr. Christian Gieger from <a
          href="http://www.lionbioscience.com">IBM Lion Bioscience
          AG</a> as a guest speaker.</p>

          <table cols="3" width="100%">
            <tr bgcolor="#CCCCCC">
              <td width="15%">Day</td>

              <td width="55%">Topic</td>

              <td width="30%">Speaker</td>
            </tr>

            <tr bgcolor="#FFFF00">
              <td>15.4.2002</td>

              <td>Overview and Introduction</td>

              <td>Korbinian Strimmer</td>
            </tr>

            <tr>
              <td>22.4.2002</td>

              <td>Biology and Technology (Lecture 1)</td>

              <td>Florian Burckhardt
              </td>
            </tr>

            <tr>
              <td>29.4.2002</td>

              <td>Expression Indices (Lecture 2)</td>

              <td>Korbinian Strimmer
              </td>
            </tr>

            <tr>
              <td>06.5.2002</td>

              <td>Normalisation (Lecture 3)</td>

              <td>Jan Wolfertz
              </td>
            </tr>

            <tr>
              <td>13.5.2000</td>

              <td>Differential Expression I (Lecture 4)</td>

              <td>Rainer Opgen-Rhein
              </td>
            </tr>

            <tr>
              <td>*17.5.2000</td>

              <td>Differential Expression II (Lecture 5)</td>

              <td>Gangolf Jobb</td>
            </tr>

            <tr>
              <td>27.5.2002</td>

              <td>Differential Expression III (Lecture 6)</td>

              <td>Roland Wolf
              </td>
            </tr>

            <tr bgcolor="#FFFF00">
              <td>03.6.2002</td>

              <td>Activities at IBM LION Bioscience AG</td>

              <td>Christian Gieger<br>
              (invited guest)</td>
            </tr>


            <tr>
              <td>10.6.2002</td>

              <td>ANOVA (Lecture 7)</td>

              <td>Samson Adebayo</td>
            </tr>


            <tr>
              <td>17.6.2002</td>

              <td>Clustering I (Lecture 9)</td>

              <td>Astrid Zierer
	      </td>
            </tr>

            <tr>
              <td>24.6.2002</td>
	        
             <td>Clustering II (Lecture 10)</td>

              <td>Korbinian Strimmer
              </td>


            </tr>

            <tr>
              <td>01.7.2002</td>

              <td>Data and Dimension reduction (Lecture 8)</td>

              <td>
	      Martina Messow
              </td>


             </tr>

            <tr>
              <td>08.7.2002</td>

              <td>Classification I (Lecture 11)</td>

              <td>Anne-Laure Socher-Boulesteix
              </td>
            </tr>

            <tr>
              <td>15.7.2002</td>

              <td>Classification II (Lecture 12)</td>

              <td>Ludwig Fahrmeir
              </td>
            </tr>
          </table>
        </td>
      </tr>

      <tr>
        <td align="LEFT">
          <p>&nbsp;<font size="-1">* on Friday, 10am, same place (Monday, 20
	  May is bank holiday!)</font></p>
        </td>
      </tr>

      <tr>
        <td align="LEFT">
          <p>&nbsp;</p>
        </td>
      </tr>

      <tr>
        <td align="LEFT">
          <p><em>Reading list:</em></p>

          <p>These are the papers we plan to discuss in each session:
          </p>

<ol>
  <li>Biology and Technology
    <ul>
     <li>Review articles in: The Chipping Forecast.  1999.
         <a href="http://www.nature.com/cgi-taf/DynaPage.taf?file=/ng/journal/v21/n1s/index.html">Supplement, Nature Genetics 21:1-60</a>.
     </li>
     <li>Hamadeh, H., and C.A. Afshari. 2000. Gene chips and 
         functional genomics.
	 <a href="http://www.americanscientist.org/articles/00articles/hamadeh.html">American Scientist 88:508-515</a>.
     </li>
     
     <li><a href="http://industry.ebi.ac.uk/~brazma/Biointro/biology.html">A quick introduction
          to elements of biology - cells, molecules, genes,
          functional genomics, microarrays</a><br> (introductory web site at the European
          Bioinformatics Institute)
     </li>

     <li><a href="http://www.dnaftb.org/">DNA from the beginning: an animated
     primer on the basics of DNA, genes and heredity</a><br> (web site with
     an introductory genetics tutorial)
     </li>
     
   </ul>
  </li>
  <li>Expression Indices
    <ul>
     <li>Li, C., and W.H. Wong. 2001.  Model-based analysis of oligonucleotide
         arrays:  expression index computation and outlier detection.
	 <a href="http://www.pnas.org/cgi/content/abstract/98/1/31">PNAS 98:31-36</a>.
     </li>
     <li>Naef, F., D.A. Lim, N. patil, and M.O. Magnasco.  2001.
         From features to expression: high-density oligonucleotide
	 array analysis revisited.
	 <a href="http://xxx.lanl.gov/abs/physics/0102010">Preprint</a>.
     </li>
     <li>Irizarry, R.A., B. Hobbs, F. Collin, Y.S. Beazer-Barclay,
         K.J. Antonellis, U. Scherf, and T.P. Speed.
	 2002. Exploration, normalization, and summaries of high density
	 oligonucleotide array probe level data. 
	 <a href="http://biosun01.biostat.jhsph.edu/~ririzarr/papers/index.html">Preprint</a>.
     </li>
   </ul>
  </li>
  <li>Normalisation 
    <ul>
     <li>Schadt, E.E., C. Li, C. Su, and W. H. Wong.  2000.
         Analyzing high-density oligonucleotide gene expression array data.
	 <a href="http://www3.interscience.wiley.com/cgi-bin/issuetoc?ID=73505449">J. Cellul. Biochem. 80:192-202</a>
	 (<a href="http://www.biostat.harvard.edu/complab/publication/Analyzing%20High-Density%20Oligonucleotide.pdf">reprint</a>).
     </li>
     <li>Yang, Y.-H., S. Dudoit, P. Lu, and T.P. Speed. 2001.
         Normalization for cDNA microarray data.
	 <a href="http://www.spie.org/web/meetings/programs/pw01/confs/4266.html">Proc. Int.
         Symp. Biomedical Optics. 20-26 January, 2001, San Jose</a>.
	 (<a href="http://citeseer.nj.nec.com/406329.html">preprint</a>)
     </li>
     <li>Yang, Y.-H., S. Dudoit, P. Lu, D.M. Lin, V. Peng, J. Ngai,
         and T.P. Speed.  2002.  Normalization for cDNA microarray data:
	 a robust composite method addressing single and multiple slide
	 systematic variation.
	 <a href="http://nar.oupjournals.org/cgi/content/abstract/30/4/e15">Nucl. Acids Res. 30(4):e15</a>.
     </li>
   </ul>
  </li>
  <li>Differential Expression I
    <ul>
     <li>Chen, Y., E.D. Dougherty, and M.L. Bittner.  1997.
         Ratio-based decisions and the quantitative analysis of
	 cDNA microarray images.
	 <a href="http://ojps.aip.org/journal_cgi/dbt?KEY=JBOPFO&Volume=2&Issue=4">J. Biomed. Optics 2:364-374</a>.
     </li>
     <li>Newton, M.A., C.M. Kendziorski, C.S. Richmond, F.R. Blattner,
         and K.W. Tsui.  2001.
	 On differential variability of expression ratios: improving
	 statistical inference about gene expression changes from
	 microarray data.
	 <a href="http://dx.doi.org/10.1089/106652701300099074">J. Comp. Biol. 8:37-52</a>.
     </li>
     <li>Ting Lee, M.L., F.C. Kuo, G.A. Whitmore, and J. Sklar.  2000.
         Importance of replication in microarray gene expression studies:
	 statistical methods and evidence from repetitive cDNA hybridizations.
	 <a href="http://www.pnas.org/cgi/content/abstract/97/18/9834">PNAS 97:9834-9839</a>.
     </li>
   </ul>
  </li>
  <li>Differential Expression II
    <ul>
     <li>Dudoit, S., Y.-H- Yang, M.C. Callow, and T.P. Speed.  2002.
         Statistical methods for identifying differentially expressed
	 genes in replicated cNDA microarray experiments.
	 <a href="http://www.stat.sinica.edu.tw/statistica/">Statistika Sinica</a> 12(1).
	 (<a href="http://citeseer.nj.nec.com/dudoit00statistical.html">preprint</a>)
     </li>
     <li>Goss Tusher, V., R. Tibshirani, and G. Chu.  2001.
         Significance analysis of microarrays applied to the
	 ionizing radiation response.
	 <a href="http://www.pnas.org/cgi/content/abstract/98/9/5116">PNAS 98:5116-5121</a>.
     </li>
     <li>Lönnstedt, I., and T.P. Speed.  2002.
         Replicated microarray data.
	 <a href="http://www.stat.sinica.edu.tw/statistica/">Statistika Sinica</a> 12(1).
	 (<a href="http://citeseer.nj.nec.com/457612.html">preprint</a>)
     </li>
   </ul>
  </li>
  <li>Differential Expression III
    <ul>
     <li>Efron, B., R. Tibshirani, J.D. Storey, and V. Tusher.  2001.
         Empirical Bayes analysis of a microarray experiment.
	 <a href="http://dx.doi.org/10.1198/016214501753382129">JASA 96:1151-1160</a>.
     </li>
     <li>Efron, B., J.D. Storey, and R. Tibshirani. 2001.
         Microarrays empirical Bayes methods, and false discovery rates.
	 <a href="http://www-stat.stanford.edu/research/">Technical Report 2001-23B/217</a>
	 (Dept. of Statistics, Stanford). 
     </li>
     <li>Efron, B.  2001. Robbins, empirical Bayes, and microarrays.
         <a href="http://www-stat.stanford.edu/research/">Technical Report 2001-30B/219</a>
	 (Dept. of Statistics, Stanford).
     </li>
   </ul>
  </li>
  <li>ANOVA
    <ul>
     <li>Kerr, M.K, M. Martin, and G.A. Churchill. 2000.
       Analysis of variance for gene expression microarray data.
       <a href="http://www.catchword.com/titles/10665277.htm">J. Comp. Biol.</a> 7:819-837.
       (<a href="http://www.jax.org/research/churchill/research/expression/kerr-synteni.pdf">preprint</a>)
     </li>
     <li>Kerr, M.K, and G.A. Churchill. 2001.  Statistical design and the analysis
         of gene expression microarray data.
	 <a href="http://www.journals.cup.org/bin/bladerunner?REQUNIQ=1016036271&REQSESS=9979248&117000REQEVENT=&REQINT1=76214&REQAUTH=0">Genet. Res. Camb. 77:123-128</a>.
     </li>
     <li>Kerr, M.K., C.A. Afshari, L. Bennet, P. Bushel, J. Martinez, N.J. Walker,
         and  G.A. Churchill.  2002.
	 <a href="http://www.stat.sinica.edu.tw/statistica/">Statistika Sinica</a> 12(1).
	 (<a href="http://www.jax.org/research/churchill/research/expression/niehs.pdf">preprint</a>)
     </li>
   </ul>
  </li>
  <li>Clustering I
    <ul>
     <li>Tamayo, P., D. Slonim, J. Mesirov, Q. Zhu, S. Kitareewan, E. Dmitrovsky,
          E.S. Lander, and T.R. Golub.  1999.  Interpreting patterns of gene expression with
	  self-organizing maps: methods and application to hematopoietic
	  differentiation.
	  <a href="http://www.pnas.org/cgi/content/abstract/96/6/2907">PNAS 96:2907-2912</a>. 
      </li>
     <li>Yeung, K.Y., C. Fraley, A. Murua, A.E. Raftery, and W.L. Ruzzo. 2001.
         Model-based clustering and data transformations for gene expression
	 data.
	 <a href="http://bioinformatics.oupjournals.org/cgi/content/abstract/17/10/977">Bioinformatics 17:977-987</a>.
     </li>
     <li>Fraley, C., and A.E. Raftery. 1998.  How many clusters? Which
         clustering method? Answers via model-based cluster analysis.
	 <a href="http://www3.oup.co.uk:80/computer_journal/subs/Volume_41/Issue_08/Vol41_08.body.html#AbstractFraley">The Computer Journal 41:578-588.</a>
    </li>
   </ul>
  </li>
  <li>Clustering II
    <ul>
     <li>Eisen, M.B., P.T. Spellman, P.O. Brown, and D. Botstein. 1998.
	 Cluster analysis and  display of genome-wide expression patterns.
	 <a href="http://www.pnas.org/cgi/content/abstract/95/25/14863">PNAS 95:14863-14868</a>.
     </li>
     <li>Herrero, J., A. Valencia, and J. Dopazo. 2001.  A hierarchical
         unsupervised growing neural network for clustering gene
	 expression patterns.
	 <a href="http://bioinformatics.oupjournals.org/cgi/content/abstract/17/2/126">Bioinformatics 17:126-136</a>.
     
     </li>
     <li>Kerr, M.K., and G.A. Churchill.  2001.  Bootstrapping cluster
         analysis: assessing the reliability of conclusions from microarray
	 experiments.
	 <a href="http://www.pnas.org/cgi/content/abstract/98/16/8961">PNAS 98:8961-8965</a>.</li>
   </ul>
  </li>
  <li>Data and Dimension Reduction
    <ul>
     <li>Alter, O., P.O. Brown, and D. Botstein. 2000.  Singular value
         decomposition for genome-wide expression data processing and
	 modelling.
	 <a href="http://www.pnas.org/cgi/content/abstract/97/18/10101">PNAS97:10101-10106</a>
     </li>
     <li>Yeung, K.Y., and W.L. Ruzzo. 2001. Principal component analysis
         for clustering gene expression data.
	 <a href="http://bioinformatics.oupjournals.org/cgi/content/abstract/17/9/763">Bioinformatics 17:763-774</a>.
     </li>
     <li>Hastie, T., R. Tibshirani, M.B. Eisen, A. Alizadeh, R. Levy, L. Staudt,
         W.C. Chan, D. Botstein, and P. Brown. 2000. 
	 'Gene shaving' as a method for
	 identifying distinct sets of genes with similar expression patterns.
	 <a href="http://genomebiology.com/2000/1/2/research/0003">Genome Biology 1(2):research0003.-0003.21</a>
     </li>
   </ul>
  </li>

  
  <li>Classification I
    <ul>
     <li>Golub, T.R., D.K. Slonim, P. Tamayo, C. Huard, M. Gaasenbeek,
         J.P. Mesirov, H. Coller, M.L. Loh, J.R. Downing, M.A. Caligiuri,
	 C.D. Bloomfield, and E.S. Lander.  1999.
	 Molecular classification of cancer: class discovery and class
	 prediction by gene expression monitoring.
	 <a href="http://www.sciencemag.org/cgi/content/abstract/286/5439/531">Science 286:531-537</a>.
     </li>
     <li>Slonim, D.K., P. Tamayo, J.P. Mesirov, T.R. Golub, and E.S. Lander.
         2000.  Class prediction and discovery using gene expression data.  
	 Proceedings RECOMB IV:263-271.
	 (<a href="http://citeseer.nj.nec.com/slonim00class.html">reprint</a>) 
     </li>
     <li>Yeang, C.-H., S. Ramaswamy, P. Tamayo, S. Mukherjee, R.M. Rifkin,
         M. Angelo, M. Reich, E.S. Lander, J. Mesirov, and T. Golub. 2001.
	 Molecular classification of multiple tumor types.
	 <a href="http://bioinformatics.oupjournals.org/cgi/content/abstract/17/suppl_1/S316">Bioinformatics 17:S316-S322</a>.
     </li>
   </ul>
  </li>
  <li>Classification II
    <ul>
     <li>Nguyen, D.V., and D.M. Rocke. 2002. Tumor classification by partial
         least squares using microarray gene expression data.
	 <a href="http://bioinformatics.oupjournals.org/cgi/content/abstract/18/1/39">Bioinformatics 18:39-50</a>.
     </li>
     <li>D. Ghosh. 2002.  Singular value decomposition regression models
         for classication of tumors from microarray experiments.
	 <a href="http://www-smi.stanford.edu/projects/helix/psb02/">Pacific Symposium on Biocomputing 7:18-29</a>.
     </li>
     <li>Eilers, P.H.C., J.M. Boer, G.-J. van Ommen, and H.C. van Houwelingen.
         2001. Classification of microarray data with penalized
         logistic regression.
	 <a href="http://www.spie.org/web/meetings/programs/pw01/confs/4266.html">Proc. Int.
         Symp. Biomedical Optics. 20-26 January, 2001, San Jose</a>.
     </li>
   </ul>
  </li>
  
  
  </ol>


          <p>Most of these papers are downloadable from within the network of the
	   University of Munich. (You may need to configure your browser properly
           - see the <a href="http://www.lrz-muenchen.de/services/netzdienste/proxy/browser-config/">proxy guide</a>
	   at the <a href="http://www.lrz-muenchen.de">LRZ</a>).  If you have
	   problems or prefer to copy an article please contact us.</p>
        </td>
      </tr>



      <tr>
        <td align="LEFT">
          <p>&nbsp;</p>
        </td>
      </tr>

      <tr>
        <td align="LEFT">
        
 
          <p><em>More references and links:</em></p>
          <p>We have only time to discuss a few selected papers - other interesting articles dealing with microarray analysis
          are listed at the following places: </p>

  <ul>
     <li><a href="http://www.biostat.umn.edu/~weip/course/ge/syl1.html">Wei Pan's reference list</a> (Minnesota)</li>
     <li><a href="http://www.stat.wisc.edu/~yandell/statgen/reference/array.html">Brian Yandell's reference list</a> (Wisconsin)</li>
     <li><a href="http://linkage.rockefeller.edu/wli/microarray/">Wentian Li's reference list</a> (Rockefeller)</li>
     <li><a href="http://www.molgen.mpg.de/~heydebre/explit.html">Anja von Heydebreck's reference list</a> (Berlin)</li>
   </ul>
 
    <p>
    <a href="http://www.bioconductor.org/workshop.html">Gene expression analysis workshops (with Bioconductor)</a>
    
    
    </p>
 
 

        </td>
      </tr>



      <tr>
        <td align="LEFT">
          <p>&nbsp;</p>
        </td>
      </tr>

      <tr>
        <td align="LEFT">
          <p><em>Software:</em></p>

           <p>
	     An overview over the large number of available software
	     for gene expression analysis is attempted,
	    e.g., <a href="http://genome-www5.stanford.edu/MicroArray/SMD/restech.html">here</a>
	    (Stanford) and
	     <a href="http://ihome.cuhk.edu.hk/~b400559/arraysoft.html">here</a> (Y.F. Leung).
	     Note that many of these programs are commercial.</p>
	    
	    <p>
	    Fortunately, most of the state-of-the-art methods for gene expression
	    analysis 
	    are also available (and sometimes even exclusively so!)  in the form of
	    packages for the
	    <a href="http://www.r-project.org">R system</a>.  R is an
	    open-source clone of S-Plus, and is a widely used and very versatile
	    statistics package. R is freely available
	    for Windows, Macintosh and
	    Linux/Unix systems. 
	  </p>
	  
	  <p>
	    Hence, it is strongly recommended to analyze gene expression
	    data with R.
	    We have compiled a fairly exhaustive list of
	    <a href="../../notes/rexpress.html">R packages for gene
	     expression analysis</a>.
	    One of these packages deserves particular mention: 
	    the <a href="http://www.bioconductor.org/">BioConductor project</a>
	    (Harvard) unifies and merges several previously independent R packages
	    and provides tools for the analysis of 
	    both cDNA and Affymetrix arrays. 
	  </p>
	  
	  
        </td>
      </tr>

      <tr>
        <td align="LEFT">
          <p align="right"><font size="-2"><em>Last
          modified:</em><br>
           May 28, 2002</font></p>
        </td>
      </tr>
    </table>
  </body>
</html>

