<?xml version="1.0" encoding="UTF-8" ?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.1//EN" 
   "http://www.w3.org/TR/xhtml11/DTD/xhtml11.dtd">

<html xmlns="http://www.w3.org/1999/xhtml">
<head>
  <meta http-equiv="content-type" content="text/html; charset=utf-8" />
  <meta name="KeyWords" content=
  "current topics in biostatistics, Korbinian Strimmer" />
  <meta name="Author" content="Korbinian Strimmer" />
  <meta name="Description" content="Course Description" />
  <meta name="viewport" content="width=device-width, initial-scale=1" />

  <title>Current Topics in Biostatistics</title>
  <link rel="stylesheet" type="text/css" href="../../../styles.css" />
</head>

<body>
  <script type="text/javascript">
//<![CDATA[
  <!--  
   if (top.location != self.location) {top.location = self.location}
  //-->
//]]>
</script>

  <div id="page">

  <div id="header">Current Topics in Biostatistics</div>


 <div id="main">

 
  <span class="sidenote">
  Back to <a href="https://strimmerlab.github.io/courses.html">Teaching</a> page


    </span>

  <p><strong>Contact and Information:</strong></p>


    <p><a href="https://strimmerlab.github.io/korbinian.html">Korbinian Strimmer</a> and <a href="http://www.brunel.ac.uk/people/alex-lewin">Alex Lewin</a><br />
       
 Imperial College London <br />
 Academic Year 2015-16</p>

  <span class="sidenote">
<b>Begin:</b> 30 September 2015 <br />
<b>End:</b> July 2016 <br />
<b>Time:</b> Tuesday 13:30-15:00 <br />
<b>Place:</b> EBS Meeting Room (162), 1st Floor, St. Mary's<br />

    </span>


 
   <p><strong>Synopsis:</strong></p>


<p>
In this reading seminar we aim to discuss in an informal way statistical and machine learning papers presenting methods
relevant to biostatistical analysis.   We read across books, journals and preprints, as we see fit.
</p>


 

    <p><strong>Schedule:</strong></p>

<span class="sidenote">See here for the <a href="../../2014-15/current-topics/index.html">papers discussed in previous sessions</a> in the academic year 2014-15</span>

<p>We start the journal / book club on <strong>30 September 2015</strong> and 
then meet regulary about twice a month, now usually on a Tuesday afternoon. For precise dates and links to the papers see below!</p>

<p>Announcements concerning this seminar are also sent via the
<a href="https://mailman.ic.ac.uk/mailman/listinfo/statistics-stmarys">statistics-stmarys</a>
mailing list. Please join this list if you would like to receive regular updates by email.</p>


<p>Sessions:</p>
<table border="1"  summary="dates and contents of each session">
  <tr>
    <th>#</th>
    <th>Date</th>
    <th>Topic</th>
  </tr>


<tr>
    <td>28</td>
    <td>26 Jul 2016</td>
    <td>
J. W. Miller and M. T. Harrison. 2015. 
<a href="http://arxiv.org/abs/1502.06241">Mixture models with a prior on the number of
components</a>. arXiv:1502.06241
    </td>
 </tr>
<tr>
    <td>27</td>
    <td>12 Jul 2016</td>
    <td>
J. G. Scott et al. 2015. <a href="http://dx.doi.org/10.1080/01621459.2014.990973">False discovery rate regression: An application to neural synchrony detection  in primary visual cortex</a>. JASA 110: 459-471; <br />
W. Tansey, O. Koyejo, R. A. Poldrack, J. G. Scott.
2014. <a href="http://arxiv.org/abs/1411.6144">False discovery rate smoothing</a>. 
arXiv:1411.6144.
    </td>
 </tr>

<tr>
    <td>&nbsp;</td>
    <td></td>
    <td> 
    </td>
 </tr>


<tr>
    <td>26</td>
    <td>7 Jun 2016</td>
    <td>G. J. Szekely and M. L. Rizzo. 2009. 
<a href="http://dx.doi.org/10.1214/09-AOAS312">Brownian distance correlation</a>. Ann. Applied Statist. 3: 1236â€“1265; <br />
 D. N. Reshef et al. 2011. <a href="http://dx.doi.org/10.1126/science.1205438">Detecting novel associations in large data sets</a> 
Science 334: 1518-1524; <br />
N. Simon and R. Tibshirani. 2014.
<a href="http://arxiv.org/abs/1401.7645">Comment on "Detecting novel associations in large data sets"</a>.
arXiv:1401.7645; <br />
S. de Siqueira Santos et al. 2014.
<a href="http://dx.doi.org/10.1093/bib/bbt051">A comparative study of statistical methods used to identify dependencies between gene expression signals</a>. 
Brief. Bioinf. 15: 906-918. 
    </td>
 </tr>

<tr>
    <td>&nbsp;</td>
    <td></td>
    <td> 
    </td>
 </tr>


<tr>
    <td>25</td>
    <td>17 May 2016</td>
    <td>J. Wang, Q. Zhao, T. Hastie, and A. B. Owen. 2016. 
        <a href="http://arxiv.org/abs/1508.04178">Confounder adjustment in 
        multiple hypothesis testing.</a> arXiv:1508.04178.
    </td>
 </tr>


<tr>
    <td>24</td>
    <td>5 May 2016</td>
    <td>R. Guhaniyogi and D. B. Dunson 2015. <a href="http://dx.doi.org/10.1080/01621459.2014.969425">Bayesian compressed regression.</a>
         JASA 110: 1500-1514.
    </td>
 </tr>

<tr>
    <td>&nbsp;</td>
    <td></td>
    <td> 
    </td>
 </tr>



<tr>
    <td>23</td>
    <td>17 Mar 2016</td>
    <td>
 R. Zhang, C. Czado, and K. Sigloch. 2016.
<a href="http://dx.doi.org/10.1111/rssc.12118">Bayesian spatial modelling for high
dimensional seismic inverse problems</a>. JRSS C 65: 187-213.
    </td>
 </tr>

<tr>
    <td>22</td>
    <td>Thu 3 Mar 2016</td>
    <td>Q. Song and F. Liang. 2015.  <a href="http://dx.doi.org/10.1111/rssb.12095">A split-and-merge Bayesian variable selection approach for ultrahigh dimensional regression</a>. JRSS B 77: 947-972.   
    </td>
 </tr>

<tr>
    <td>&nbsp;</td>
    <td></td>
    <td> 
    </td>
 </tr>

<tr>
    <td>21</td>
    <td>Thu 18 Feb 2016</td>
    <td>  
       B. Efron. 2014. <a href="http://dx.doi.org/10.1214/13-STS455">Two modeling strategies for empirical Bayes estimation.</a>
        Statistical Science 29:285-301. B. Efron. 2015. <a href="http://statweb.stanford.edu/~ckirby/brad/papers/2015BayesDeconvolutionProblem.pdf">The Bayes deconvolution problem.</a> Preprint.
    </td>
 </tr>


<tr>
    <td>20</td>
    <td>Thu 4 Feb 2016</td>
    <td>N. Fusi et al. 2012.
<a href="http://dx.doi.org/10.1371/journal.pcbi.1002330">Joint modelling of confounding factors and prominent genetic regulators provides increased accuracy in genetical genomics studies.</a> PLOS Comp. Biol. 8: e1002330 and 
O. Stegle et al. 2010. <a href="http://dx.doi.org/10.1371/journal.pcbi.1000770">A Bayesian framework to account for complex non-genetic factors.</a> PLOS Comp. Biol. 6: e1000770
    </td>
 </tr>

<tr>
    <td>&nbsp;</td>
    <td></td>
    <td>  
    </td>
 </tr>


<tr>
    <td>19</td>
    <td>Thu 21 Jan 2016</td>
    <td> M. Taddy et al. 2015. 
       <a href="http://jmlr.org/proceedings/papers/v37/matthew15.pdf">Bayesian and empirical Bayesian forests.</a>  32nd International Conference on Machine Learning (ICML). JMLR Workshop Proceeedings Vol. 37.

    </td>
 </tr>

<tr>
    <td>&nbsp;</td>
    <td></td>
    <td>  
    </td>
 </tr>


<tr>
    <td>18</td>
    <td>Mon 14 Dec 2015</td>
    <td>J. Taylor and R. J. Tibshirani. 2015. <a href="http://dx.doi.org/10.1073/pnas.1507583112">Statistical learning and selective inference.</a>
        PNAS 25:7629-7634.


    </td>
 </tr>
 

<tr>
    <td>17</td>
    <td>Thu 3 Dec 2015</td>
    <td>A. P. Dawid. 2010. <a href="http://jmlr.csail.mit.edu/proceedings/papers/v6/dawid10a/dawid10a.pdf">Beware of the DAG!</a>
       JMLR Workshop and Conference Proceedings 6:59-86 (NIPS 2008 Workshop on Causality).
    </td>
 </tr>


<tr>
    <td>&nbsp;</td>
    <td></td>
    <td>  
    </td>
 </tr>


<tr>
    <td>16</td>
    <td>Thu 19 Nov 2015</td>
    <td>J. Piironen and A. Vehtari. 2015. 
        <a href="http://arxiv.org/abs/1503.08650">Comparison of Bayesian predictive 
        methods for model selection</a>. arXiv:1503.08650. <br />
J. Piironen and A. Vehtari. 2015. <a href="http://arxiv.org/abs/1508.02502">Projection predictive variable selection using
Stan+R</a>. arXiv:1508.02502.
        
    </td>
 </tr>



<tr>
    <td>&nbsp;</td>
    <td></td>
    <td>
    </td>
 </tr>



<tr>
    <td>15</td>
    <td>Thu 29 Oct 2015</td>
    <td> 
 D. P. Simpson, H. Rue, T. G. Martins, A. Riebler, and S. H. S&oslash;rbye. 2014. <a href="http://arxiv.org/abs/1403.4630">Penalising model component complexity: A principled, practical approach to constructing priors</a>. arXiv:1403.4630.
    </td>
 </tr>



<tr>
    <td>14</td>
    <td>Wed 14 Oct 2015</td>
    <td>
         B. Lakshminarayanan, D. M. Roy, Y. W. Teh. 
         2014. <a href="http://arxiv.org/abs/1406.2673">Mondrian forests: efficient online random forests</a>.
         Advances in Neural Information Processing Systems 27:3140-3148.<br />
B. Lakshminarayanan, D. M. Roy, Y. W. Teh. 
         2015. 
         <a href="http://arxiv.org/abs/1506.03805">Mondrian forests for 
         large-scale regression when uncertainty matters</a>. arXiv:1506.03805.
         


 
    </td>
 </tr>



<tr>
    <td>&nbsp;</td>
    <td></td>
    <td>  
    </td>
 </tr>



<tr>
    <td>13</td>
    <td>Wed 30 Sep 2015</td>
    <td>B. Letham, C. Rudin, T.H. McCormick and D. Madigan. 2015. <a href="http://www.e-publications.org/ims/submission/AOAS/user/submissionFile/17235?confirm=20d145e7">Interpretable classifiers using rules and

         Bayesian analysis: building a better stroke prediction model.</a> AOAS in press.<br />
         See also the <a href="https://www.aaai.org/ocs/index.php/WS/AAAIW13/paper/download/7088/6696">short note</a> presented at the 2013 AAAI Conference on Artificial Intelligence.
    </td>
 </tr>


</table>  


</div>


<div id="footer">
<em>Last modified:</em>
        July 15, 2016<br />
        <a href="http://validator.w3.org/check?uri=referer">
           <img style="float: right;" src="../../../images/valid-xhtml11.png" alt="Valid XHTML 1.1"
           height="31" width="88" /></a>

  </div>

  </div>
</body>
</html>
