<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">

<html>
<head>
   <meta name="viewport" content="width=device-width, initial-scale=1" >
   <meta http-equiv="Content-Type" content="text/html; charset=utf-8" >

  <title>GeneNet</title>
</head>

<body bgcolor="#FFFFFF">
  <script type="text/javascript" language="JavaScript">
  //<![CDATA[
  <!--  
   if (top.location != self.location) {top.location = self.location}
  //-->

  //]]>
  </script>

  <table  border="0">
    <tr>
      <td>
        <h1 align="center">
           <img  align="bottom" height="32px" src="logo.jpg" alt="R logo"> GeneNet
        </h1>

        <p align="center"><font size="+1">Modeling and Inferring Gene Networks</font></p>
        <hr noshade="noshade" size="1">

        <table width="100%" border="0">
          <tbody>
            <tr>
              <td width="20%" valign="top" align="left">
              <b><font size="-1" face=
              "Arial, Verdana, sans-serif">What is
              it?</font></b></td>

              <td>
                <p><a href=
                  "http://cran.r-project.org/package=GeneNet">GeneNet</a> is an <a href=
                "http://www.r-project.org">R</a> package for
                learning high-dimensional dependency networks from genomic
                data (e.g. <a href="../../notes/ggm.html">gene association networks</a>).
                The current version of GeneNet also allows to assign putative
                directions to edges in the network (see <a href="download/arabidopsis-net.R">arabidopsis example</a>).</p>

                <p>This is a screenshot of GeneNet running in R on GNU/Linux:</p>

                <p align="center"><img alt="Screenshot" border="0"
                src="genenet-screenshot.jpg"></p>

                <p align="left">The picture shows a sparse
                graphical Gaussian model for 102 genes inferred from
                a microarray data set with 9 data points (<a href="ecoli-net.pdf">detailed view</a>).</p>
		
		<p>See also an <a href="../../publications/misc/GeneNet-Rnews2006.pdf">R News article 
		about GeneNet</a>.</p>
		
		
              </td>
            </tr>
          </tbody>
        </table><br>


        <table width="100%" border="0">
          <tbody>
            <tr>
              <td width="20%" valign="top" align="left">
              <b><font size="-1" face=
              "Arial, Verdana, sans-serif">Authors:</font></b></td>

              <td>
                <p>The GeneNet package is maintained by <a href=
                "../../korbinian.html">
                Korbinian Strimmer</a> (<a
		href="mailto:strimmer@uni-leipzig.de">email</a>).</p>
		
		<p>The methods and the code was jointly
                developed by</p>

                <ul>

                <li><a href=
                  "http://www.bice.ch/index.php?id=303&amp;tx_wfqbe_pi1[uid]=24">Juliane
                  Sch&auml;fer</a>, and</li>

                  <li><a href=
                  "http://www.simon-kucher.com/">Rainer Opgen-Rhein</a>,</li>
                 
                  <li><a href=
                  "http://strimmerlab.org/korbinian.html">
                  Korbinian Strimmer</a>.</li>
                </ul>
              </td>
            </tr>
          </tbody>
        </table><br>


        <table width="100%" border="0">
          <tbody>
            <tr>
              <td width="20%" valign="top" align="left">
              <b><font size="-1" face=
              "Arial, Verdana, sans-serif">Examples and Documentation:</font></b></td>

              <td valign="top">
                
		
		<p>The following example scripts illustrate some of the functionality of GeneNet:</p>
		
		<ul>
<li>
A. thaliana data (Opgen-Rhein and Strimmer 2007) - <a href="download/arabidopsis-net.R">R script with Markdown</a> / <a href="download/arabidopsis-net.pdf">resulting PDF</a> 
</li>
<li>
E. coli data (Sch&auml;fer and Strimmer 2005) - <a href="download/ecoli-net.R">R script with Markdown</a> / <a href="download/ecoli-net.pdf">resulting PDF</a> 
</li>
<li>
T-cell data (Opgen-Rhein and Strimmer 2006) - <a href="download/tcell-net.R">R script  with Markdown</a> / <a href="download/tcell-net.pdf">resulting PDF</a> 
</li>

		</ul>
		
		
		<p>All methods available in GeneNet are described
                with examples in their respective help pages -
                please browse the online
                <a href="http://cran.r-project.org/web/packages/GeneNet/GeneNet.pdf">Manual</a>  (pdf file)
                for an overview.</p>
	
		
              </td>
            </tr>
          </tbody>
        </table><br>



       <table width="100%" border="0">
          <tbody>
            <tr>
              <td width="20%" valign="top" align="left">
              <b><font size="-1" face=
              "Arial, Verdana, sans-serif">Papers:</font></b></td>

              <td valign="top">
                <p>There are a number of papers describing the methods available
		 in GeneNet:</p>
		 		
                <ul>
 
                 <li>Opgen-Rhein, R., and K. Strimmer. 2007.  From correlation to causation 
                 networks: a simple approximate learning algorithm and its application 
                 to high-dimensional plant gene expression data. 
                  <a href="http://dx.doi.org/10.1186/1752-0509-1-37">BMC Syst.
                 Biol. <strong>1</strong>: 37</a>.   
                </li>

                  <li>Sch&auml;fer, J., R. Opgen-Rhein,  and K. Strimmer. 2006. 
                 Reverse engineering genetic networks using the GeneNet package.
                 <a href="../../publications/misc/GeneNet-Rnews2006.pdf">R News</a>
                 <strong>6/5</strong>:50-53. </li>
                 
                  <li>Opgen-Rhein, R.,  and K. Strimmer. 2006. 
                  Inferring gene dependency networks from genomic longitudinal data:
                  a functional data approach. <a href="http://www.ine.pt/revstat/pdf/rs060103.pdf">REVSTAT</a>
                  <strong>4</strong>:53-65. </li>    

                 <li>Opgen-Rhein, R., and K. Strimmer. 2006. 
                  Using regularized dynamic correlation to infer gene dependency
                  networks from time-series microarray data.
                  Proceedings of the <a href="../../publications/conferences/dyncorshrink2006.pdf">4th International 
                  Workshop on Computational Systems 
                  Biology, WCSB 2006</a> (June 12-13, 2006, Tampere, Finland), pp. 73-76. 
                  </li>

		  <li>Sch&auml;fer, J., and K. Strimmer. 2005. A shrinkage approach
	          to large-scale covariance matrix estimation and implications for
		  functional genomics. <a href="http://dx.doi.org/10.2202/1544-6115.1175">Statist.
		  Appl.  Genet. Mol. Biol. <strong>4</strong>: 32</a>.
                 </li>
	  
                  <li>Sch&auml;fer, J., and K. Strimmer. 2005. An empirical Bayes
                  approach to inferring large-scale gene association
                  networks. <a href="http://dx.doi.org/10.1093/bioinformatics/bti062">Bioinformatics 
                  <strong>21</strong>: 754-764</a>.</li>

                </ul>
                
              </td>
            </tr>
          </tbody>
        </table><br>




        <table width="100%" border="0">
          <tbody>
            <tr>
              <td width="20%" valign="top" align="left">
              <b><font size="-1" face=
              "Arial, Verdana, sans-serif">License:</font></b></td>

              <td valign="top">
                <p>GeneNet is distributed under the terms of the
                <a href="http://www.gnu.org/licenses/gpl-3.0.txt">GNU GPL version 3</a>
                 (or any later version).</p>
              </td>
            </tr>
          </tbody>
        </table><br>


        <table width="100%" border="0">
          <tbody>
            <tr>
              <td width="20%" valign="top" align="left">
                <p align="left"><b><font size="-1" face=
                "Arial, Verdana, sans-serif">Requirements:</font></b></p>
              </td>

              <td width="80%" valign="top">
                <p>GeneNet requires a recent version of the <a href=
                "http://www.r-project.org">R</a> program (at least
                version 3.0.2).</p>
                
		<p>The current version 1.2.13 of GeneNet  requires
		installation of three other packages available from
		CRAN:
		<a href= "http://cran.r-project.org/web/packages/corpcor/">corpcor</a>,
		<a href= "http://cran.r-project.org/web/packages/longitudinal/">longitudinal</a>,
		<a href="http://cran.r-project.org/web/packages/fdrtool/">fdrtool</a>.
		 </p>


                <p>For plotting network structures GeneNet uses the <a href="http://www.bioconductor.org/packages/release/bioc/html/graph.html">graph</a> and <a href="http://www.bioconductor.org/packages/release/bioc/html/Rgraphviz.html">Rgraphviz</a> R packages from Bioconductor.
In addition, the original <a href="http://www.graphviz.org">graphviz</a> software may also
be used (via <a href="https://en.wikipedia.org/wiki/DOT_%28graph_description_language%29">dot files</a>).</p>
              </td>
            </tr>
          </tbody>
        </table><br>

        <table width="100%" border="0">
          <tbody>
            <tr>
              <td width="20%" valign="top" align="left">
                <p align="left"><b><font size="-1" face=
                "Arial, Verdana, sans-serif">Download and
                Installation:</font></b></p>
              </td>

              <td width="80%" valign="top">
                <p>The current release is GeneNet version <strong>1.2.13</strong>. It
                can be downloaded from <a href=
                "http://cran.r-project.org/">CRAN</a>:</p>

                <ul>
                  
                  <li>GeneNet on CRAN: <a href=
                  "http://cran.r-project.org/web/packages/GeneNet/">
                  Unix source package as well as Windows and MacOS X binaries</a>.</li>
		  
		  
		  
                </ul>

                <p>For installation proceed as described in the
                documentation that accompanied your version of R.
                For example, under Linux type in the command shell:
                <em>R CMD INSTALL GeneNet_1.2.13.tar.gz</em></p>

                <p>If your computer is connected to the internet
                you may also install the GeneNet package by
                typing at the R prompt:
                <em>install.packages("GeneNet", dependencies = TRUE)</em></p>
              </td>
            </tr>
          </tbody>
        </table><br>




 
        <table width="100%" border="0">
          <tbody>
            <tr>
              <td width="20%" valign="top" align="left">
              <b><font size="-1" face=
              "Arial, Verdana, sans-serif">Release
              History:</font></b></td>

              <td valign="top">
                <p>The changes between subsequent releases of
                GeneNet are documented in the <a href=
                "http://cran.r-project.org/web/packages/GeneNet/NEWS">release history</a>.</p>

                <p>The current version is <a href="http://cran.r-project.org/src/contrib/GeneNet_1.2.13.tar.gz">GeneNet 1.2.13</a>. 
		Previous releases of GeneNet can also be downloaded from the <a href="http://cran.r-project.org/src/contrib/Archive/GeneNet/">GeneNet archive</a>. Note however
		that these versions are obsolete and only provided for
                reference purposes.</p>

 

                <p>Prior to its release as a standalone package, GeneNet was part of
		the GeneTS package.  Hence, for older versions please check the 
		<a href="../genets/index.html">GeneTS</a> page.</p>
		

              </td>
            </tr>
          </tbody>
        </table><br>



        <table width="100%" border="0">
          <tbody>
            <tr>
              <td width="20%" valign="top" align="left">
              <b><font size="-1" face=
              "Arial, Verdana, sans-serif">Acknowledgments:</font></b></td>

              <td valign="top">
                <p>Development of GeneNet was supported by an Emmy
                Noether research grant of the <a href=
                "http://www.dfg.de/en/">DFG</a>.</p>
              </td>
            </tr>
          </tbody>
        </table><br>
        <hr size="1" noshade="noshade">

        <div align="center">
          <font size="-1"><em>Last modified:</em> 2 August 2015</font>
        </div>
      </td>
    </tr>
  </table>
</body>
</html>
