<?xml version="1.0" encoding="UTF-8" ?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.1//EN" "http://www.w3.org/TR/xhtml11/DTD/xhtml11.dtd">

<html xmlns="http://www.w3.org/1999/xhtml"><head>

  <meta http-equiv="content-type" content="text/html; charset=utf-8" />
  <meta name="KeyWords" content="open source mass spectrometry tools" />
  <meta name="Author" content="Sebastian Gibb and Korbinian Strimmer" />
  <meta name="Description" content="open source mass spectrometry tools" />
  <meta name="viewport" content="width=device-width, initial-scale=1" />

  <title>Open Source Tools for Mass Spectrometry Analysis</title>
  <link rel="stylesheet" type="text/css" href="../styles.css" />

</head><body>
  <script type="text/javascript">
//<![CDATA[
  <!--
   if (top.location != self.location) {top.location = self.location}
  //-->
//]]>
</script>

  <div id="page">


    <div id="header">Open Source Tools for Mass Spectrometry Analysis</div>

    <div id="main">

    <span class="sidenote">Back to <a href="http://strimmerlab.org/notes/index.html">index</a></span>

    <p><strong>Overview:</strong></p>

    <p>This is a list intended to facilitate comparison of
open source software for analyzing mass spectrometry data.
The list comprises R packages and some other software and contains
 links to the home pages and a short description of the respective features.
    </p>

    <p><strong>R packages</strong></p>

    <p><strong>CRAN (<a href="http://cran.r-project.org/">http://cran.r-project.org</a>)</strong></p>

    <table style="table-layout: fixed;" summary="CRAN packages">
      <thead><tr style="background-color: rgb(204, 204, 204);">
        <td><b>Package</b></td>
        <td><b>License</b></td>
        <td><b>Version</b></td>

        <td><b>Input Data</b></td>
        <td><b>Baseline Correction</b></td>
        <td><b>Peak Detection</b></td>
        <td><b>Normalization</b></td>
        <td><b>Peak Alignment</b></td>
        <td><b>Miscellaneous</b></td>
        <td><b>Authors</b></td>
      </tr>
      </thead><tbody>
      <tr class="row1" valign="top">
        <td>
          <a href="http://strimmerlab.org/software/maldiquant/index.html" id="MALDIquant">MALDIquant</a>
        </td>

        <td>
          GPL (&ge;3)
        </td>

        <td>
          1.19
        </td>

        <td>
            raw data (mass and intensity);
            tab, csv, Bruker Daltonics *flex-series format, Ciphergen XML, mzXML, mzML, msd, imzML, Analyze 7.5 and CDF (via <a href="http://cran.r-project.org/web/packages/MALDIquantForeign/">MALDIquantForeign</a>)
        </td>

        <td>
          SNIP, TopHat, Convex Hull, Moving Median
        </td>

        <td>
          local maxima over SNR*noise (noise estimation by MAD or Friedman's Super Smoother)
        </td>

        <td>
          intensity transformation and smoothing; total-ion-current/probability-quotient-normalization/median calibration
        </td>

        <td>
          first landmark peaks are identified that occur in most spectra and subsequently,
          a warping function is computed for each spectrum by fitting a local
          regression to the matched reference peaks
        </td>

        <td>
         trim spectra, monoisotopic peak detection, peak labeling, diverse plots for calibrated mass spectra and peaks,
         merge technical replicates, peak filtering, intensity matrix creation, create and plot MSI slices
        </td>

        <td>
          Sebastian Gibb
        </td>
     </tr>

     <tr class="row2" valign="top">
        <td>
          <a href="http://cran.r-project.org/web/packages/baseline/index.html">baseline</a>
        </td>

        <td>
          GPLv2
        </td>

        <td>
          1.2
        </td>

        <td>
          raw data (2-column matrix)
        </td>

        <td>
          Asymmetric Least Squares, Fill peaks, Iterative Restricted Least Squares,
          Low-pass FFT filter, Median window, Modified polynomial fitting, Simultaneous Peak Detection and Baseline Correction,
          Robust Baseline Estimation, Rolling ball
        </td>

        <td>
          local maxima
        </td>

        <td>
          NA
        </td>

        <td>
          NA
        </td>

        <td>
          GUI available
        </td>

        <td>
          Kristian Hovde Liland and Bj&oslash;rn-Helge Mevik
        </td>
     </tr>

     <tr class="row1" valign="top">
        <td>
          <a href="http://cran.r-project.org/web/packages/enviPick/index.html">enviPick</a>
        </td>

        <td>
          GPLv3
        </td>

        <td>
          1.3
        </td>

        <td>
          mzXML
        </td>

        <td>
          NA
        </td>

        <td>
          Data partitioning, EIC clustering, peak extraction by penalizing intensity
reversions
        </td>

        <td>
          NA
        </td>

        <td>
          NA
        </td>

        <td>
        </td>

        <td>
          Martin Loos
        </td>
     </tr>

     <tr class="row2" valign="top">
        <td>
          <a href="http://cran.r-project.org/web/packages/MSeasy/index.html">MSeasy</a>
        </td>

        <td>
          GPLv2
        </td>

        <td>
          5.3.3
        </td>

        <td>
            ASCII, Agilent *.D; mzXML and CDF via <a href="http://bioconductor.org/packages/release/bioc/html/xcms.html">xcms</a>
        </td>

        <td>
          NA
        </td>

        <td>
          local maxima (highest point in a moving window of size 7)
        </td>

        <td>
          NA
        </td>

        <td>
          unsupervised clustering (partitional and hierarchical algorithms with different distance metrics and link methods)
        </td>

        <td>
          focus on GC/MS; <a href="http://cran.r-project.org/web/packages/MSeasyTkGUI/index.html">GUI available</a>; file output for NIST/ARISTO search
        </td>

        <td>
          Elodie Courtois, Yann Guitton, Florence Nicole
        </td>
     </tr>

      <tr class="row1" valign="top">
        <td>
          <a href="https://spectrino.com" id="Spectrino">Spectrino</a>
        </td>

        <td>
          GPL (&ge;2)
        </td>

        <td>
          2.0
        </td>

        <td>
            tab
        </td>

        <td>
          removal of constant threshold and replacing negative intensities with zero
        </td>

        <td>
          NA
        </td>

        <td>
          normalization to 1e6
        </td>

        <td>
          binning by rounding to the nearest integer or split the intensity to its proportion to it the two closest integers
        </td>

        <td>
            trim spectra, average spectra, spectra grouping/rearranging, visualisation, GUI but <i>windows-only</i>, connections to Java, python, ... possible
        </td>

        <td>
          Teodor Krastev
        </td>
      </tr>

      <tr class="row2" valign="top">
        <td>
          <a href="http://cran.r-project.org/web/packages/Peaks/">Peaks</a>
        </td>

        <td>
          LGPL
        </td>

        <td>
          0.2
        </td>

        <td>
          vector of intensities
        </td>

        <td>
          SNIP
        </td>

        <td>
          gaussian deconvolution
        </td>

        <td>
          NA
        </td>

        <td>
          NA
        </td>

        <td>
        </td>

        <td>
          Miroslav Morhac
        </td>
      </tr>

      </tbody></table>

    <p><strong>Bioconductor mass spectrometry packages (<a href="http://bioconductor.org/packages/release/BiocViews.html#MassSpectrometry">http://bioconductor.org/packages/release/BiocViews.html#MassSpectrometry</a>)</strong></p>

    <table style="table-layout: fixed;" summary="BIOC packages">
      <thead><tr style="background-color: rgb(204, 204, 204);">
        <td><b>Package</b></td>
        <td><b>License</b></td>
        <td><b>Version</b></td>

        <td><b>Input Data</b></td>
        <td><b>Baseline Correction</b></td>
        <td><b>Peak Detection</b></td>
        <td><b>Normalization</b></td>
        <td><b>Peak Alignment</b></td>
        <td><b>Miscellaneous</b></td>
        <td><b>Authors</b></td>
      </tr>
      </thead><tbody>
      <tr class="row1" valign="top">
        <td>
          <a href="http://bioconductor.org/packages/release/bioc/html/MassSpecWavelet.html" id="MassSpecWavelet">MassSpecWavelet</a>
        </td>

        <td>
          LGPL (&ge;2)
        </td>

        <td>
          1.34.0
        </td>

        <td>
          vector of intensities
        </td>

        <td>
          NN; additional Savitzky-Golay Algorithm
        </td>

        <td>
          continuous wavelet
        </td>

        <td>
          NA
        </td>

        <td>
          NA
        </td>

        <td>
        </td>

        <td>
          Pan Du, Warren Kibbe, Simon Lin
        </td>
      </tr>

      <tr class="row2" valign="top">
        <td>
          <a href="http://bioconductor.org/packages/release/bioc/html/MSnbase.html">MSnbase</a>
        </td>

        <td>
            Artistic-2.0
        </td>

        <td>
          1.18.0
        </td>

        <td>
            mgf, netCDF, mzData, mzML, mzXML (mostly via <a href="http://bioconductor.org/packages/release/bioc/html/mzR.html">mzR</a>)
        </td>

        <td>
          NA
        </td>

        <td>
          local maxima (see <a href="#MALDIquant">MALDIquant</a>)
        </td>

        <td>
            sum, max, quantiles, <a href="http://bioconductor.org/packages/release/bioc/html/vsn.html">vsn</a>
        </td>

        <td>
          NA
        </td>

        <td>
            a lot of annotations are possible; methods for cleaning spectra; quantitation (labeled and labelfree), integration of identification data
        </td>

        <td>
            Laurent Gatto
        </td>
      </tr>


      <tr class="row1" valign="top">
        <td>
          <a href="http://bioconductor.org/packages/release/bioc/html/PROcess.html">PROcess</a>
        </td>

        <td>
          Artistic-2.0
        </td>

        <td>
          1.46.0
        </td>

        <td>
          2-column matrix (cols: 1st: m/z 2nd: intensities)
        </td>

        <td>
          local minimum (or user-defined quantile) + LOESS
        </td>

        <td>
          local maximum (optional smoothing (moving average))
        </td>

        <td>
          median of all TIC
        </td>

        <td>
           intersection graphs
        </td>

        <td>
        </td>

        <td>
          Xiaochun Li
        </td>
      </tr>

      <tr class="row2" valign="top">
        <td>
          <a href="http://bioconductor.org/packages/release/bioc/html/TargetSearch.html">TargetSearch</a>
        </td>

        <td>
          GPL (&ge; 2)
        </td>

        <td>
          1.26.0
        </td>

        <td>
          NetCDF
        </td>

        <td>
          divide spectrum in subparts, calculate standard deviation, a
          user-definied percentage above the standard deviations become true signal
        </td>

        <td>
          smooth spectrum, determine sign changes;
          using PPC
        </td>

        <td>
          equalize max peak intensity
        </td>

        <td>
          NA
        </td>

        <td>
          GC/MS
        </td>

        <td>
          Alvaro Cuadros-Inostroza, Jan Lisec, Henning Redestig , Matt Hannah
        </td>
      </tr>

      <tr class="row1" valign="top">
        <td>
          <a href="http://bioconductor.org/packages/release/bioc/html/xcms.html">xcms</a>
        </td>

        <td>
          GPL (&ge; 2)
        </td>

        <td>
          1.46.0
        </td>

        <td>
          NetCDF/mzXML/mzData/mzML files
        </td>

        <td>
          constant threshold; Savitzky-Golay Algorithm or no one (depending on PD method)
        </td>

        <td>
          centroid base wavelet (for LC/MS);
          continuous wavelet (using MassSpecWavelet, for MS)
        </td>

        <td>
          NA
        </td>

        <td>
          construct a master peak list and align by best match;
          heuristically clustering; nearest peak
        </td>

        <td>
          database search possible;
          write support for mzData and NetCDF;
          a lot of functions for LC/MS
        </td>

        <td>
          Colin A. Smith, Ralf Tautenhahn , Steffen Neumann , Paul Benton
        </td>
      </tr>
    </tbody></table>

    <p><strong>Other R packages (not on CRAN/Bioconductor)</strong></p>

    <table style="table-layout: fixed;" summary="other R packages">
      <thead><tr style="background-color: rgb(204, 204, 204);">
        <td><b>Package</b></td>
        <td><b>License</b></td>
        <td><b>Version</b></td>

        <td><b>Input Data</b></td>
        <td><b>Baseline Correction</b></td>
        <td><b>Peak Detection</b></td>
        <td><b>Normalization</b></td>
        <td><b>Peak Alignment</b></td> <td><b>Classification</b></td>
        <td><b>Miscellaneous</b></td>
        <td><b>Authors</b></td>
      </tr>
      </thead><tbody>

      <tr class="row1" valign="top">
        <td><a href="http://omics.pnl.gov/software/DanteR.php">DanteR</a></td>
        <td>GPL (&ge;2)</td>
        <td>0.2</td>
        <td>Excel, SQLite DB, Access, CSV, tab-delimited text</td>
        <td>NA</td>

        <td>NA</td>
        <td>eigenvalues, linear regression, LOESS, quantile</td>

        <td>NA</td>
        <td>NA</td>
        <td>impute data (k-nearest-neighbor, row means, ANOVA, ...); interactive plots, histogram, QQ, boxplots, 3D plots, Venn diagram, PCA plots;
          please see full feature list at the <a href="http://omics.pnl.gov/software/DanteR.php">DanteR website</a></td>
        <td>Tom Taverner and Ashoka Polpitiya</td>
      </tr>

      <tr class="row2" valign="top">
        <td><a href="http://bioinformatics.nki.nl/software.php">MASDA</a></td>
        <td>GPL (&ge;2)</td>
        <td>0.6</td>
        <td>CSV</td>
        <td>PROcess (local minimum + LOESS), LOWESS, Friedman's super smoother, cubic smooth spline</td>
        <td>sign change in first derivation</td>
        <td>(intensities - offset)/scale [different combinations of mean, median, sd, mad, range etc. for offset and scale]</td>

        <td>hierarchical clustering</td>
        <td>ANOVA, Kruskal-Wallis</td>
        <td></td>
        <td>Wouter Meuleman</td>
      </tr>

      <tr class="row1" valign="top">
        <td><a href="http://www-stat.stanford.edu/%7Etibs/PPC/">PPC</a></td>
        <td>GPLv2</td>
        <td>1.02</td>
        <td>CSV</td>
        <td>NN</td>

        <td>local maximum above noise estimated by Friedman's super smoother</td>
        <td>log-transformation + linear transformation (10th percentile becomes 0; 90th, 1)</td>

        <td>hierarchical clustering</td>
        <td>nearest shrunken centroids (PPC)</td>
        <td></td>
        <td>Balasubramanian Narasimhan, R. Tibshirani, T. Hastie</td>
      </tr>

      <tr class="row2" valign="top">
        <td><a href="http://www.meb.ki.se/~yudpaw/">ProSpect</a></td>
        <td>KI</td>
        <td>0.3.6</td>
        <td>CSV</td>
        <td>LOESS, <a href="http://www.meb.ki.se/~yudpaw/">rsmooth</a></td>

        <td>finding regions of interest (significance)</td>
        <td>NA</td>

        <td>NA</td>
        <td>NA</td>
        <td></td>
        <td>Andreas Quandt, Tan Chuen Seng, Alexander Ploner, Stefano Calza, Yudi Pawitan</td>
      </tr>

    </tbody></table>

    <p><strong>Obsolete R packages</strong></p>

    <table style="table-layout: fixed;" summary="obsolete packages">
      <thead><tr style="background-color: rgb(204, 204, 204);">
        <td><b>Package</b></td>
        <td><b>License</b></td>
        <td><b>Version</b></td>

        <td><b>Input Data</b></td>
        <td><b>Baseline Correction</b></td>
        <td><b>Peak Detection</b></td>
        <td><b>Normalization</b></td>
        <td><b>Peak Alignment</b></td>
        <td><b>Classification</b></td>
        <td><b>Miscellaneous</b></td>
        <td><b>Authors</b></td>
      </tr>
      </thead><tbody>
      <tr class="row1" valign="top">
        <td>
          <a href="http://cran.r-project.org/web/packages/caMassClass/">caMassClass</a>
        </td>

        <td>
          GPLv3
        </td>

        <td>
          1.9
        </td>

        <td>
          CSV, mzXML
        </td>

        <td>
          see PROcess
        </td>

        <td>
          see PROcess, additional: faster (uses C), different use of SNR, no AUC
        </td>

        <td>
          adjust peak high to
          min-max: min=0, max=1 for each spectrum;
          avr-std: mean=0, unit variance;
          med-mad: median=0, unit median absolute deviation
        </td>

        <td>
          based on Peakminer algorithm
          (<a href="http://www.evms.edu/evms-virginia-prostate-center/seldi-protein-profiling-studies.html">Virginia Prostate Center</a>),
          bins peaks with similarly mass (±constant value)
        </td>

        <td>
          LogitBoost from <a href="http://cran.r-project.org/web/packages/caTools">caTools</a>;
          lda and qda from <a href="http://cran.r-project.org/web/packages/MASS">MASS</a>;
          rpart from <a href="http://cran.r-project.org/web/packages/rpart">rpart</a>
        </td>

        <td>
        </td>

        <td>
          Jarek Tuszynski
        </td>
      </tr>

      <tr class="row2" valign="top">
        <td>
          <a href="http://cran.r-project.org/web/packages/msProcess/">msProcess</a>
        </td>

        <td>
          GPLv2
        </td>

        <td>
          1.0.6
        </td>

        <td>
          Ciphergen XML, vectors of m/z and intensity values
        </td>

        <td>
          determine local minima and apply one of the following R functions:
          loess.smooth (default), spline, supsmu, approx, cummin, msSmoothMRD
          (msProcess, wavelet based)
        </td>

        <td>
          local maxima; local maxima higher than estimated background
          (msPeaksSearch); continous wavelets; discrete wavelets
        </td>

        <td>
          TIC; Standard Normal Variate (SNV) transformation; max intensity or
          count
          quantification
        </td>

        <td>
          hierarchical clustering; cluster by distance (smaller than threshold);
          vote; mrd (histogram smoothing) (for details see msProcess documentation)
        </td>

        <td>
          NA
        </td>

        <td>
          in silico spectrometer; a lot of denoising functions; additional data
          packages: <a href="http://cran.r-project.org/web/packages/msBreast/">msBreast</a>,
          <a href="http://cran.r-project.org/web/packages/msDilution">msDilution</a>,
          <a href="http://cran.r-project.org/web/packages/msProstate">msProstate</a>
        </td>

        <td>
          Lixin Gong, William Constantine, Yu Alex Chen
        </td>
      </tr>

      <tr class="row1" valign="top">
        <td>
          <a href="http://cran.r-project.org/web/packages/pkDACLASS/">pkDACLASS</a>
        </td>

        <td>
          LGPL
        </td>

        <td>
          1.0
        </td>

        <td>
          2-column dataframe
        </td>

        <td>
          see PROcess
        </td>

        <td>
          monoisotopic peak detection (poisson-distribution+EM-algorithm)
        </td>

        <td>
          NA
        </td>

        <td>
          round non-integers mass to integer and using decimal fraction to weight their intensity
        </td>

        <td>
          using <a href="http://cran.r-project.org/web/packages/randomForest/">randomForest</a>
        </td>

        <td>
          contains some datasets
        </td>

        <td>
          Juliet Ndukum, Mourad Atlas, Susmita Datta
        </td>
      </tr>

      <tr class="row2" valign="top">
        <td>
          <a href="http://cran.r-project.org/web/packages/rTOFsPRO/">rTOFsPRO</a>
        </td>

        <td>
          GPL (&ge;2)
        </td>

        <td>
          1.4.1
        </td>

        <td>
          lists generated by <a href="http://cran.r-project.org/web/packages/WMBrukerParser/">WMBrukerParser</a>
        </td>

        <td>
          estimate baseline by a linear, exponential or gaussian model;
          substract a constant value
        </td>

        <td>
          peak detection on the average spectrum (to use high-precision peak
          detection you have to contact the authors)
        </td>

        <td>
          smoothing (moving average)
        </td>

        <td>
          align peaks against peak list of the average spectrum
          (to use a global align+binning you have to contact the authors)
        </td>

        <td>
          NA
        </td>

        <td>
          everything is controlled by text files =&gt; very difficult interface
        </td>

        <td>
          Dariya Malyarenko, Maureen Tracy, William Cooke
        </td>
      </tr>
    </tbody></table>

    <table style="table-layout: fixed;" summary="R packages for file import">
      <thead><tr style="background-color: rgb(204, 204, 204);">
        <td><b>Package</b></td>
        <td><b>License</b></td>
        <td><b>Version</b></td>
        <td><b>Authors</b></td>
        <td><b>Miscellaneous</b></td>
      </tr>
      </thead><tbody>
      <tr class="row1" valign="top">
        <td><a href="http://cran.r-project.org/web/packages/isopat/">isopat</a></td>
        <td>GPLv2</td>
        <td>1.0</td>
        <td>Martin Loos</td>
        <td>superseeded by <a href="http://cran.r-project.org/web/packages/enviPat/">enviPat</a></td>
      </tr>
    </tbody></table>

    <p><strong>R packages for importing mass spectrometry data files</strong></p>

    <table style="table-layout: fixed;" summary="R packages for file import">
      <thead><tr style="background-color: rgb(204, 204, 204);">
        <td><b>Package</b></td>
        <td><b>License</b></td>
        <td><b>Version</b></td>

        <td><b>File Formats</b></td>
        <td><b>Miscellaneous</b></td>
        <td><b>Authors</b></td>
      </tr>
      </thead><tbody>

      <tr class="row1" valign="top">
        <td><a href="http://cran.r-project.org/web/packages/MALDIquantForeign/">MALDIquantForeign</a></td>
        <td>GPLv3</td>
        <td>0.12</td>
        <td>tab, csv, Bruker Daltonics *flex series format, Ciphergen XML, mzXML, mzML, msd, imzML, Analyze 7.5, CDF</td>
        <td></td>
        <td>Sebastian Gibb</td>
      </tr>

      <tr class="row2" valign="top">
        <td><a href="http://bioconductor.org/packages/release/bioc/html/mzID.html">mzID</a></td>
        <td>GPLv2</td>
        <td>1.8.0</td>
        <td>mzIdentML</td>
        <td></td>
        <td>Thomas Lin Pedersen</td>
      </tr>

      <tr class="row1" valign="top">
        <td><a href="http://bioconductor.org/packages/release/bioc/html/mzR.html">mzR</a></td>
        <td>Artistic-2.0</td>
        <td>2.4</td>
        <td>mzXML, mzData, mzML, mzIdentML, NetCDF</td>
        <td></td>
        <td>Bernd Fischer, Steffen Neumann, Laurent Gatto</td>
      </tr>

      <tr class="row2" valign="top">
        <td><a href="http://cran.r-project.org/web/packages/readBrukerFlexData/">readBrukerFlexData</a></td>
        <td>GPLv3</td>
        <td>1.8.2</td>
        <td>fid files of Bruker Datlonics *flex series</td>
        <td></td>
        <td>Sebastian Gibb</td>
      </tr>

      <tr class="row1" valign="top">
        <td><a href="http://cran.r-project.org/web/packages/readMzXmlData/">readMzXmlData</a></td>
        <td>GPLv3</td>
        <td>2.8.1</td>
        <td><a href="http://tools.proteomecenter.org/wiki/index.php?title=Formats:mzXML">mzXML</a>
        </td>
        <td></td>
        <td>Sebastian Gibb</td>
      </tr>
    </tbody></table>

    <br />

    <p><strong>R packages for Mass Spectrometry Imaging</strong></p>

    <table style="table-layout: fixed;" summary="R packages for file import">
      <thead><tr style="background-color: rgb(204, 204, 204);">
        <td><b>Package</b></td>
        <td><b>License</b></td>
        <td><b>Version</b></td>
        <td><b>Miscellaneous</b></td>
        <td><b>Authors</b></td>
      </tr>
      </thead><tbody>
      <tr class="row1" valign="top">
        <td><a href="http://cran.r-project.org/web/packages/MALDIquant/">MALDIquant</a></td>
        <td>GPL (&ge;3)</td>
        <td>1.19</td>
        <td></td>
        <td>Sebastian Gibb</td>
      </tr>

      <tr class="row2" valign="top">
        <td><a href="http://bioconductor.org/packages/release/bioc/html/Cardinal.html">Cardinal</a></td>
        <td>Artistic-2.0</td>
        <td>1.2.0</td>
        <td></td>
        <td>Kyle D. Bemis</td>
      </tr>

    </tbody></table>

    <br />

    <p><strong>R packages for calculation of isotopic pattern/distribution</strong></p>

    <table style="table-layout: fixed;" summary="R packages for file import">
      <thead><tr style="background-color: rgb(204, 204, 204);">
        <td><b>Package</b></td>
        <td><b>License</b></td>
        <td><b>Version</b></td>

        <td><b>Authors</b></td>
      </tr>
      </thead><tbody>
      <tr class="row1" valign="top">
        <td><a href="http://cran.r-project.org/web/packages/enviPat/">enviPat</a></td>
        <td>GPLv2</td>
        <td>2.0</td>
        <td>Martin Loos, Christian Gerber</td>
      </tr>

      <tr class="row2" valign="top">
        <td><a href="http://bioconductor.org/packages/release/bioc/html/BRAIN.html">BRAIN</a></td>
        <td>GPLv2</td>
        <td>1.16</td>
        <td>Piotr Dittwald</td>
      </tr>

      <tr class="row1" valign="top">
        <td><a href="http://bioconductor.org/packages/release/bioc/html/Rdisop.html">Rdisop</a></td>
        <td>GPL</td>
        <td>1.30</td>
        <td>Anton Pervukhin, Steffen Neumann</td>
      </tr>

    </tbody></table>

    <br />

    <p><strong>Obsolete R packages for calculation of isotopic pattern/distribution</strong></p>

    <table style="table-layout: fixed;" summary="R packages for file import">
      <thead><tr style="background-color: rgb(204, 204, 204);">
        <td><b>Package</b></td>
        <td><b>License</b></td>
        <td><b>Version</b></td>
        <td><b>Authors</b></td>
        <td><b>Miscellaneous</b></td>
      </tr>
      </thead><tbody>
      <tr class="row1" valign="top">
        <td><a href="http://cran.r-project.org/web/packages/isopat/">isopat</a></td>
        <td>GPLv2</td>
        <td>1.0</td>
        <td>Martin Loos</td>
        <td>superseeded by <a href="http://cran.r-project.org/web/packages/enviPat/">enviPat</a></td>
      </tr>
    </tbody></table>

    <br />

    <p><strong>R packages to detect isotopic patterns</strong></p>

    <table style="table-layout: fixed;" summary="R packages for file import">
      <thead><tr style="background-color: rgb(204, 204, 204);">
        <td><b>Package</b></td>
        <td><b>License</b></td>
        <td><b>Version</b></td>
        <td><b>Miscellaneous</b></td>
        <td><b>Authors</b></td>
      </tr>
      </thead><tbody>
      <tr class="row1" valign="top">
        <td><a href="http://cran.r-project.org/web/packages/nontarget/">nontarget</a></td>
        <td>GPLv2</td>
        <td>1.7</td>
        <td>also screening for peaks related by different adducts and/or homologue series</td>
        <td>Martin Loos</td>
      </tr>

      <tr class="row2" valign="top">
        <td><a href="http://bioconductor.org/packages/release/bioc/html/IPPD.html">IPPD</a></td>
        <td>GPL (&ge;2)</td>
        <td>1.18.0</td>
        <td></td>
        <td>Martin Slawski</td>
      </tr>

    </tbody></table>

    <br />

    <p><strong>R packages for annotation of mass spectrometry data</strong></p>

    <table style="table-layout: fixed;" summary="R packages for annotation">
      <thead><tr style="background-color: rgb(204, 204, 204);">
        <td><b>Package</b></td>
        <td><b>License</b></td>
        <td><b>Version</b></td>
        <td><b>Miscellaneous</b></td>
        <td><b>Authors</b></td>
      </tr>
      </thead><tbody>
      <tr class="row1" valign="top">
        <td><a href="http://www.bioconductor.org/packages/release/bioc/html/CAMERA.html">CAMERA</a></td>
        <td>GPLv2</td>
        <td>1.26</td>
        <td>annotation of peaklist generated by <a href="http://bioconductor.org/packages/release/bioc/html/xcms.html">xcms</a></td>
        <td>Carsten Kuhl, Ralf Tautenhahn, Steffen Neumann</td>
      </tr>

    </tbody></table>

    <br />

    <p><strong>R packages to handle mass spectrometry libraries</strong></p>

    <table style="table-layout: fixed;" summary="R packages for annotation">
      <thead><tr style="background-color: rgb(204, 204, 204);">
        <td><b>Package</b></td>
        <td><b>License</b></td>
        <td><b>Version</b></td>
        <td><b>Miscellaneous</b></td>
        <td><b>Authors</b></td>
      </tr>
      </thead><tbody>
      <tr class="row1" valign="top">
        <td><a href="http://www.bioconductor.org/packages/release/bioc/html/RMassBank.html">RMassBank</a></td>
        <td>Artistic-2.0</td>
        <td>1.12.0</td>
        <td>preparation of MS/MS spectra for a MassBank submission</td>
        <td>Michael Stravs, Emma Schymanski, Steffen Neumann, Erik Mueller</td>
      </tr>

    </tbody></table>

    <br />

    <p><strong>Non-R tools</strong></p>

    <table style="table-layout: fixed;" summary="non-R tools">
      <thead><tr style="background-color: rgb(204, 204, 204);">
        <td><b>Application</b></td>
        <td><b>Programming Language</b></td>
        <td><b>Operating Systems</b></td>
        <td><b>License</b></td>
        <td><b>Version</b></td>

        <td><b>Input Data</b></td>
        <td><b>Baseline Correction</b></td>
        <td><b>Peak Detection</b></td>
        <td><b>Normalization</b></td>
        <td><b>Peak Alignment</b></td>
        <td><b>Classification</b></td>
        <td><b>Miscellaneous</b></td>
        <td><b>Authors</b></td>
      </tr>
      </thead><tbody>
      <tr class="row1" valign="top">
        <td><a href="http://emzed.ethz.ch/">eMZed</a></td>
        <td>Python</td>
        <td>L, M, W</td>
        <td>GPLv3</td>
        <td>2.22.2</td>
        <td></td>
        <td></td>
        <td></td>
        <td></td>
        <td></td>
        <td></td>
        <td>LC-MS development framework based on <a href="http://open-ms.sourceforge.net/">OpenMS/TOPP</a> and <a href="http://bioconductor.org/packages/release/bioc/html/xcms.html">xcms</a></td>
        <td>Patrick Kiefer and Uwe Schmitt</td>
      </tr>

      <tr class="row2" valign="top">
        <td><a href="http://sing.ei.uvigo.es/mass-up/">Mass-Up</a></td>
        <td>Java/R</td>
        <td>L, W</td>
        <td>GPL</td>
        <td>1.0.7</td>
        <td>CSV, mzML, mzXML</td>
        <td>see <a href="#MALDIquant">MALDIquant</a></td>
        <td>see <a href="#MALDIquant">MALDIquant</a> and see <a href="#MassSpecWavelet">MassSpecWavelet</a></td>
        <td>see <a href="#MALDIquant">MALDIquant</a></td>
        <td>Forward algorithm or see <a href="#MALDIquant">MALDIquant</a></td>
        <td>PCA, SVM</td>
        <td>various tools and plots for Quality Control, Biomarker Discovery; Hierachical Clustering, Biclustering</td>
        <td>Florentino Fdez-Riverola, Daniel Glez-Pe&ntilde;a, Miguel Reboiro-Jato, Jos&eacute; Lu&iacute;s Capelo-Mart&iacute;nez, Hugo L&oacute;pez-Fern&aacute;ndez</td>
      </tr>

      <tr class="row1" valign="top">
        <td><a href="http://msxpertsuite.org/wiki/pmwiki.php/Main/Massxpert">massXpert</a></td>
        <td>C++</td>
        <td>L, M, W</td>
        <td>GPLv3</td>
        <td>6.0.2</td>
        <td></td>
        <td></td>
        <td></td>
        <td></td>
        <td></td>
        <td></td>
        <td>various tools for polymer editing/simulation/calculation;
          please visit: <a href="http://msxpertsuite.org/wiki/pmwiki.php/Main/Massxpert">http://msxpertsuite.org/wiki/pmwiki.php/Main/Massxpert</a></td>
        <td>Filippo Rusconi</td>
      </tr>

      <tr class="row2" valign="top">
        <td><a href="http://msxpertsuite.org/wiki/pmwiki.php/Main/Minexpert2">mineXpert2</a></td>
        <td>C++</td>
        <td>L, M, W</td>
        <td>GPLv3</td>
        <td>6.0.2</td>
        <td></td>
        <td></td>
        <td></td>
        <td></td>
        <td></td>
        <td></td>
        <td>full-depth data visualization and mining of
MS^n mass spectrometric data;
          please visit: <a href="http://msxpertsuite.org/wiki/pmwiki.php/Main/Minexpert2">http://msxpertsuite.org/wiki/pmwiki.php/Main/Minexpert2</a></td>
        <td>Filippo Rusconi</td>
      </tr>

      <tr class="row1" valign="top">
        <td><a href="http://mmass.org/">mMass</a></td>
        <td>Python</td>
        <td>L, M, W</td>
        <td>GPLv3</td>
        <td>5.5.0</td>
        <td>
          mzData, mzXML, mzML, ASCII, CSV, fid (Bruker
          Daltonics' compassXport has to be installed (W only))
        </td>
        <td>
           median of all intensities minus median of absolute deviations
           (additional you can add a relative offset and smooth the baseline);
           gaussian smoothing
        </td>
        <td>local maximum above (relative and absolute) intensity threshold</td>
        <td>intensity*1/max_intensity, (range: 0-1)</td>
        <td>NA</td>
        <td>NA</td>
        <td>deisotoping function, connections to a lot of protein databases,
            batch processing, please see also: <a href="http://mmass.org/features/">complete feature list</a></td>
        <td>Martin Strohalm</td>
      </tr>

      <tr class="row2" valign="top">
        <td><a href="http://mzmine.sourceforge.net/">MZmine2</a></td>
        <td>Java</td>
        <td>L, M, W</td>
        <td>GPLv2</td>
        <td></td>
        <td></td>
        <td></td>
        <td></td>
        <td></td>
        <td></td>
        <td></td>
        <td></td>
        <td>Matej Ore&#353;i&#269; et al (<a href="http://mzmine.sourceforge.net/development.shtml">full
            list</a>)</td>
      </tr>

      <tr class="row1" valign="top">
        <td><a href="http://www.openchrom.net/">OpenChrom</a></td>
        <td>Java</td>
        <td>L, M, W</td>
        <td>EPL</td>
        <td>1.1.0</td>
        <td>NetCDF, mzXML, CSV, D (Agilent Technologies), Bruker Daltonics *flex-series format, own file format *.chrom</td>
        <td>moving minimum, SNIP</td>
        <td>zero of first derivation of TIC signal</td>
        <td>NA</td>
        <td>NA</td>
        <td>NA</td>
        <td>batch processing; smoothing filter: Savitzky-Golay; extendable by
          <a href="https://marketplace.openchrom.net/overview">plugins</a>; database based identification possible (as plugin, NIST-DB);</td>
        <td>Philip Wenig</td>
      </tr>

      <tr class="row2" valign="top">
        <td><a href="http://open-ms.sourceforge.net/">OpenMS/TOPP</a></td>
        <td>C++</td>
        <td>L, M, W</td>
        <td>LGPL</td>
        <td></td>
        <td></td>
        <td></td>
        <td></td>
        <td></td>
        <td></td>
        <td></td>
        <td></td>
        <td>Knut Reinert, Oliver Kohlbacher, Andreas Hildebrandt and many others
          (<a href="http://open-ms.sourceforge.net/team/">full list</a>)</td>
      </tr>

      <tr class="row1" valign="top">
        <td><a href="http://proteowizard.sourceforge.net/">ProteoWizard</a></td>
        <td>C++</td>
        <td>L, M, W</td>
        <td>Apache v2</td>
        <td>3.0</td>
        <td>mzXML, mzML, mzIdentML, ... (only on Windows a lot of different vendor specific raw formats: AB Sciex, Aglient, Bruker, Thermo, Waters)</td>
        <td></td>
        <td></td>
        <td></td>
        <td></td>
        <td></td>
        <td>framework for rapid development of data analysis tools; supports various methods for accessing metadata, plotting,smoothing, peak peaking, etc.</td>
        <td>Robert Burke, Matt Chambers, Brendan MacLean and many others (<a href="http://proteowizard.sourceforge.net/team.shtml">full list</a>)</td>
      </tr>

    </tbody></table>

    <br />


    <p><strong>Non-R tools for importing mass spectrometry data files</strong></p>

    <table style="table-layout: fixed;" summary="non-R tools for importing mass spectrometry data files">
      <thead><tr style="background-color: rgb(204, 204, 204);">
        <td><b>Application</b></td>
        <td><b>License</b></td>
        <td><b>Version</b></td>

        <td><b>File Formats</b></td>
        <td><b>Miscellaneous</b></td>
        <td><b>Authors</b></td>
      </tr>
      </thead><tbody>
      <tr class="row1" valign="top">
        <td><a href="http://pymzml.github.com">pymzML</a></td>
        <td>LGPL</td>
        <td>0.7.6</td>
        <td>mzML</td>
        <td>Python 2.6.5/Python 3</td>
        <td>Till Bald, Johannes Barth, Anna Niehues, Michael Specht, Michael Hippler, Christian Fufezan</td>
      </tr>
    </tbody></table>

    <br />


    <p><strong>Abbreviations:</strong></p>

    <table summary="abbreviations">
    <tbody><tr><td><strong>AUC</strong></td><td>area under the curve</td></tr>
    <tr><td><strong>BC</strong></td><td>baseline correction</td></tr>
    <tr><td><strong>DN</strong></td><td>denoising</td></tr>
    <tr><td><strong>PA</strong></td><td>peak alignment</td></tr>
    <tr><td><strong>PD</strong></td><td>peak detection</td></tr>
    <tr><td><strong>SNR</strong></td><td>signal to noise ratio</td></tr>
    <tr><td><strong>TIC</strong></td><td>total ion current/total ion count</td></tr>
    <tr><td></td></tr>
    <tr><td><strong>GC/MS</strong></td><td>gas chromatography/mass
        spectrometry</td></tr>
    <tr><td><strong>LC/MS</strong></td><td>liquid chromatography/mass
        spectrometry</td></tr>
    <tr><td><strong>MS</strong></td><td>mass spectrometry</td></tr>
    <tr><td><strong>IMS</strong></td><td>imaging mass spectrometry</td></tr>
    <tr><td></td></tr>
    <tr><td><strong>NA</strong></td><td>not available</td></tr>
    <tr><td><strong>NN</strong></td><td>not needed</td></tr>
    <tr><td></td></tr>
    <tr><td><strong>L</strong></td><td>Linux</td></tr>
    <tr><td><strong>M</strong></td><td>Mac OS X</td></tr>
    <tr><td><strong>W</strong></td><td>Microsoft Windows</td></tr>
    </tbody></table>


    </div>

    <div id="footer">
<em>Last modified:</em>2020-09-12<br />
<a href="http://validator.w3.org/check?uri=referer">
        <img src="http://strimmerlab.org/images/valid-xhtml11.png" alt="Valid XHTML 1.1" height="31" width="88" />
      </a>
    </div>


  </div>

</body></html>
