<?xml version="1.0" encoding="ISO-8859-1" ?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.1//EN" 
   "http://www.w3.org/TR/xhtml11/DTD/xhtml11.dtd">

<html xmlns="http://www.w3.org/1999/xhtml">
<head>
  <meta http-equiv="content-type" content="text/html; charset=ISO-8859-1" />
  <meta name="KeyWords" content=
  "gene expression analysis, microarray, DNA chip, R system, Korbinian Strimmer" />
  <meta name="Author" content="Korbinian Strimmer" />
  <meta name="Description" content="R Links for gene expression analysis" />
  <title>R Packages For Gene Expression Analysis</title>
  <link rel="stylesheet" type="text/css" href="../styles.css" />
</head>

<body>
  <script type="text/javascript">
//<![CDATA[
  <!--  
   if (top.location != self.location) {top.location = self.location}
  //-->
//]]>
</script>

<div id="page">

<div id="header">R Packages For Gene Expression Analysis</div>

<div id="main">

    <span class="sidenote">Back to <a href="index.html">index</a></span>

    <p>&nbsp;</p>

    
    <p>I started this list in 2001. Since then, most R packages
    for genomic analysis have now come together under the roof
    of the <a href="http://www.bioconductor.org">Bioconductor</a> project.  
    For that reason, and because
    of time constraints on my side, I will not update this list 
    any more (<strong>last change July 2005</strong>). For a well-organized list of the packages
    available in Bioconductor
    see the 
    <a href="http://bioconductor.org/packages/1.8/Software.html">
    BioC task list</a> (version 1.8).  Also helpful should be the
    CRAN 
    <a href="http://cran.r-project.org/src/contrib/Views/Multivariate.html">
    Multivariate Statistics</a> task view.
     </p>



    <p class="c1">Overview:</p>

    <p>This list contains R packages, and software based on the R system, to
    analyze gene expression data from DNA array experiments, both for
    olignonucleotide chips and cDNA microarrays. Please drop <a href=
    "../korbinian.html">me</a> a line (<a href=
    "mailto:strimmer@uni-leipzig.de">strimmer@uni-leipzig.de</a>) if there
    are any inaccuracies or to suggest other packages that should be listed
    here.</p>

    <p>&nbsp;</p>

    <p class="c1">General-purpose Packages:</p>

    <table style="table-layout:fixed">
      <tr style="background-color:#CCCCCC">
        <td style="width:10%"><b>Package</b></td>

        <td style="width:50%"><b>Description</b></td>

        <td style="width:15%"><b>Author</b></td>

        <td style="width:15%"><b>Contact</b></td>

        <td style="width:5%"><b>Version</b></td>

        <td style="width:5%"><b>License</b></td>
      </tr>

      <tr valign="top">
        <td><a href="http://www.bioconductor.org">BioConductor</a> <br />
	(<a href="http://www.bioconductor.org/packages/bioc/stable/src/contrib/html/">Stable</a> and <br />
	<a href="http://www.bioconductor.org/packages/bioc/devel/src/contrib/html/">development</a> packages)
	
	
	</td>

        <td>Collaborative open-source project to develop a modular general
        framework for the analysis of cDNA arrays and gene chips. Includes and
        unifies some of the packages below.</td>

        <td>R. Gentleman<br />
        (and many others)</td>

        <td>rgentlem@<br />
        hsph.harvard.edu</td>

        <td>1.6</td>

        <td>GNU GPL</td>
      </tr>

      

      <tr valign="top">
        <td><a href="http://mips.gsf.de/proj/express/R/">FunDaMiner</a></td>

        <td>All-purpose package for gene expression analysis. Collection of
        various methods from preprocesssing over differential expression and
        multiple testing to clustering and classification.</td>

        <td>Michael T. Mader</td>

        <td>mtmader@gmx.de</td>

        <td>1.0.1</td>

        <td>GNU GPL</td>
      </tr>

      <tr valign="top">
        <td><a href=
        "http://bioinformatics.mdanderson.org/oomal.html">OOMAL</a></td>

        <td>Object-Oriented Microarray Analysis Library implemented in
        S-Plus.</td>

        <td>?</td>

        <td>?</td>

        <td>4.0</td>

        <td>No commercial use</td>
      </tr>

      <tr valign="top">
        <td><a href="http://www.rci.rutgers.edu/~cabrera/DNAMR/">DNAMR</a></td>

        <td>Companion R package to "Exploration and Analysis of DNA Microarray
        and Protein Array Data" book.</td>

        <td>J. Cabrera</td>

        <td>cabrera@<br />
        rci.rutgers.edu</td>

        <td>0.1</td>

        <td>GPL</td>
      </tr>

      <tr valign="top">
        <td><a href=
        "http://www.stats.gla.ac.uk/~microarray/book/smida.html">SMIDA</a></td>

        <td>Companion R package to "Statistics for Microarrays" book.</td>

        <td>J. McClure and E. Wit</td>

        <td>johndm@<br />
        stats.gla.ac.uk</td>

        <td>0.1</td>

        <td>GPL</td>
      </tr>
    </table>

    <p>&nbsp;</p>

    <p class="c1">Specialized Packages: Differential Expression (Error
    Models)</p>

    <table style="table-layout:fixed">
      <tr style="background-color:#CCCCCC">
        <td style="width:10%"><b>Package</b></td>

        <td style="width:50%"><b>Description</b></td>

        <td style="width:15%"><b>Author</b></td>

        <td style="width:15%"><b>Contact</b></td>

        <td style="width:5%"><b>Version</b></td>

        <td style="width:5%"><b>License</b></td>
      </tr>

      <tr valign="top">
        <td><a href=
        "http://www.healthsystem.virginia.edu/internet/hes/biostat/bioinformatics/Software/LPE/">
        LPE</a></td>

        <td>Local pooled error (LPE) test for gene expression data with a small
        number of replicates (2-3).</td>

        <td>N. Jain</td>

        <td>nitin.jain@<br />
        pfizer.com</td>

        <td>1.1.5</td>

        <td>GNU GPL</td>
      </tr>

      <tr valign="top">
        <td><a href=
        "http://www.healthsystem.virginia.edu/internet/hes/biostat/bioinformatics/Software/HEM/">
        HEM</a></td>

        <td>Heterogeneous Error Model for Analysis of Microarray Data.</td>

        <td>HJ. Cho</td>

        <td>hcho@virginia.edu</td>

        <td>1.0.2</td>

        <td>GNU GPL</td>
      </tr>

      <tr valign="top">
        <td><a href=
        "http://www.stat.berkeley.edu/users/terry/zarray/Software/smacode.html">
        SMA</a></td>

        <td>Data import from GenePIX and SPOT image programs, data management
        utilities, normalization and differential expression, diagonal
        discriminant analysis, various plot functions etc. Includes some mouse
        data.</td>

        <td>S. Dudoit<br />
        (and others)</td>

        <td>sandrine@<br />
        stat.berkeley.edu</td>

        <td>0.5.13</td>

        <td>GNU GPL</td>
      </tr>

      <tr valign="top">
        <td><a href=
        "http://www.bioconductor.org/repository/devel/package/html/vsn.html">vsn</a></td>

        <td>Variance stabilization and calibration for microarray data.</td>

        <td>W. Huber<br /></td>

        <td>w.huber@dkfz.de</td>

        <td>1.5</td>

        <td>GNU GPL</td>
      </tr>

      <tr valign="top">
        <td><a href=
        "http://www.cryst.bbk.ac.uk/~wernisch/yasma.html">YASMA</a></td>

        <td>ANOVA analysis, filtering and interpolation functions. Includes
        some tuberculosis data.</td>

        <td>L. Wernisch<br />
        (and others)</td>

        <td>l.wernisch@<br />
        cryst.bbk.ac.uk</td>

        <td>0.20</td>

        <td>GNU GPL</td>
      </tr>

      <tr valign="top">
        <td><a href=
        "http://www.jax.org/staff/churchill/labsite/software/anova/rmaanova/index.html">
        maanova</a></td>

        <td>Analysis of two-dye Micro Array experiment (using ANOVA,
        permutation and bootstrap, cluster and consensus tree).</td>

        <td>Hao Wu</td>

        <td>hao@jax.org</td>

        <td>0.91-3</td>

        <td>GPL 2</td>
      </tr>

      <tr valign="top">
        <td><a href="http://bioinf.wehi.edu.au/limma/">LIMMA</a></td>

        <td>Linear models for microarray data.</td>

        <td>G. Smyth</td>

        <td>smyth@wehi.edu.au</td>

        <td>1.8.9</td>

        <td>GNU GPL</td>
      </tr>
      
      <tr valign="top">
        <td><a href="http://www.biostat.ucsf.edu/jean/DEDS.htm">DEDS</a></td>

        <td>Various statistics of differential expression for microarray data,
        including t statistics, fold change, F statistics, SAM,
	moderated t and F statistics and B statistics.</td>

        <td>Y. Xiao and J. Yang</td>

        <td>jean@biostat.<br />ucsf.edu</td>

        <td>1.03</td>

        <td>GNU GPL</td>
      </tr>
     
      
      
    </table>

    <p>&nbsp;</p>

    <p class="c1">Specialized Packages: Differential Expression (Empirical
    Bayes and FDR)</p>

    <table style="table-layout:fixed">
      <tr style="background-color:#CCCCCC">
        <td style="width:10%"><b>Package</b></td>

        <td style="width:50%"><b>Description</b></td>

        <td style="width:15%"><b>Author</b></td>

        <td style="width:15%"><b>Contact</b></td>

        <td style="width:5%"><b>Version</b></td>

        <td style="width:5%"><b>License</b></td>
      </tr>

      <tr valign="top">
        <td><a href=
        "http://www.bioconductor.org/repository/devel/package/html/siggenes.html">
        siggenes</a></td>

        <td>Significance Analysis of Microarrays (SAM) and Empirical Bayes
        Analyses of Microarrays (EBAM).</td>

        <td>H. Schwender</td>

        <td>holger.schw@<br />
        gmx.de</td>

        <td>1.2.11</td>

        <td>Free for non-commercial use</td>
      </tr>

      <tr valign="top">
        <td><a href=
        "http://www.bioconductor.org/repository/devel/package/html/EBarrays.html">
        EBarrays</a></td>

        <td>Empirical Bayes methods for microarray analysis.</td>

        <td>C. Kendziorski<br />
        and M. Newton</td>

        <td>kendzior@biostat.<br />
        wisc.edu</td>

        <td>1.0-19</td>

        <td>GNU GPL</td>
      </tr>

      <tr valign="top">
        <td><a href=
        "http://www.bioconductor.org/repository/devel/package/html/qvalue.html">
        qvalue</a></td>

        <td>Several approaches to estimate the false discovery rate (FDR).</td>

        <td>J. Storey (and others)<br /></td>

        <td>jstorey@<br />
        u.washington.edu</td>

        <td>1.1</td>

        <td>GNU GPL</td>
      </tr>



      <tr valign="top">
        <td><a href=
        "../software/fdrtool/index.html">
        fdrtool</a></td>

        <td>Estimation and Control of (Local) False Discovery Rates.</td>

        <td>K. Strimmer<br /></td>

        <td>strimmer@<br />
        uni-leipzig.de</td>

        <td>1.1.0</td>

        <td>GNU GPL</td>
      </tr>




      <tr valign="top">
        <td><a href="http://www.stjuderesearch.org/depts/biostats/">BUM/<br />
        SPLOSH</a></td>

        <td>Improved estimates of local FDR to detect differentially expressed
        genes.</td>

        <td>S. Pounds</td>

        <td>stanley.pounds@<br />
        stjude.org</td>

        <td>?</td>

        <td>?</td>
      </tr>

      <tr valign="top">
        <td><a href=
        "http://cran.r-project.org/src/contrib/Descriptions/HighProbability.html">
        High<br />
        Probability</a></td>

        <td>Decision-theoretic improvement of FDR (=dFDR) to detect
        differentially expressed genes.</td>

        <td>D. Bickel</td>

        <td>bickel@prueba.info</td>

        <td>1.0-2</td>

        <td>GPL</td>
      </tr>

      <tr valign="top">
        <td><a href=
        "http://cran.r-project.org/src/contrib/Descriptions/locfdr.html">locfdr</a></td>

        <td>Computation of local false discovery rates.</td>

        <td>B. Efron and B. Narasimhan</td>

        <td>brad@stat.<br />
        stanford.edu</td>

        <td>1.0-3</td>

        <td>GPL</td>
      </tr>

      <tr valign="top">
        <td><a href=
        "http://www.bioconductor.org/repository/devel/package/html/twilight.html">
        twilight</a></td>

        <td>Improved estimate of local FDR to detect differentially expressed
        genes.</td>

        <td>S. Scheid</td>

        <td>stefanie.scheid@<br />
        molgen.mpg.de</td>

        <td>1.0.1</td>

        <td>GNU GPL</td>
      </tr>

      <tr valign="top">
        <td><a href="http://ifr69.vjf.inserm.fr/lbe/">LBE</a></td>

        <td>Improved estimate of local FDR to detect differentially expressed
        genes.</td>

        <td>C. Dalmasso (and others)</td>

        <td>dalmasso@<br />
        vjf.inserm.fr</td>

        <td>?</td>

        <td>?</td>
      </tr>

      <tr valign="top">
        <td><a href=
        "http://web.inapg.fr/ens_rech/mathinfo/recherche/mathematique/outil_A.html">
        LocalFDR/<br />
        Varmixt</a></td>

        <td>Improved estimate of local FDR from p-values to detect differentially expressed
        genes.</td>

        <td>J. Aubert (and others)</td>

        <td>aubert@inapg.fr</td>

        <td>?</td>

        <td>?</td>
      </tr>
      
      
      
      <tr valign="top">
        <td><a href=
        "http://www.geocities.com/jg_liao/software/localFDR.txt">
        localFDR</a></td>

        <td>Estimation of local FDR from p-values using stochastic order models.</td>

        <td>J.G. Liao</td>

        <td>liaojg@umdnj.edu</td>

        <td>?</td>

        <td>?</td>
      </tr>
     
      
      
      <tr valign="top">
        <td><a href=
        "http://www-stat.stanford.edu/~owen/reports/fdvar.R">FDVAR</a></td>

        <td>Variance estimation of FDR.</td>

        <td>A.B. Owen</td>

        <td>owen@stat.stanford.edu</td>

        <td>?</td>

        <td>?</td>
      </tr>
      
      
      <tr valign="top">
        <td><a href=
        "http://www.mep.ki.se/~yudpaw/">OCplus</a></td>

        <td>Computes theoretical and empirical FDR, sensitivity, 
   	false positive rate and sample size requirements when selecting 
	differentially expressed genes in simple microarray experiments. </td>

        <td>Y. Pawitan and A. Ploner</td>

        <td>alexander.ploner@meb.<br />ki.se</td>

        <td>1.2.0</td>

        <td>GNU GPL</td>
      </tr>
     
 
 
       <tr valign="top">
        <td><a href=
        "http://cran.r-project.org/src/contrib/Descriptions/SAGx.html">SAGx</a></td>

        <td>Statistical Analysis of the GeneChip. 
	 Includes methods for identifying differentially expressed genes (pava FDR).</td>

        <td>P. Broberg</td>

        <td>per.broberg@<br />
        astrazeneca.com</td>

        <td>1.5.2</td>

        <td>GNU GPL</td>
      </tr>     
      
    </table>

    <p>&nbsp;</p>

    <p class="c1">Specialized Packages: Networks</p>

    <table style="table-layout:fixed">
      <tr style="background-color:#CCCCCC">
        <td style="width:10%"><b>Package</b></td>

        <td style="width:50%"><b>Description</b></td>

        <td style="width:15%"><b>Author</b></td>

        <td style="width:15%"><b>Contact</b></td>

        <td style="width:5%"><b>Version</b></td>

        <td style="width:5%"><b>License</b></td>
      </tr>

      <tr valign="top">
        <td><a href="../software/genenet/index.html">GeneNet</a></td>

        <td>Modeling and inferring gene networks.</td>

        <td>K. Strimmer<br />
        (and others)</td>

        <td>strimmer@<br />uni-leipzig.de</td>

        <td>1.1.0</td>

        <td>GNU GPL</td>
      </tr>

      <tr valign="top">
        <td><a href=
        "http://www-personal.umich.edu/~zhud/genent.htm">GeneNT</a></td>

        <td>Testing of edges in gene networks with two-stage screening
        algorithm.</td>

        <td>D. Zhu<br />
        (and others)</td>

        <td>zhud@umich.edu</td>

        <td>1.0</td>

        <td>GNU GPL</td>
      </tr>
    </table>

    <p>&nbsp;</p>

    <p class="c1">Specialized Packages: Other</p>

    <table style="table-layout:fixed">
      <tr style="background-color:#CCCCCC">
        <td style="width:10%"><b>Package</b></td>

        <td style="width:50%"><b>Description</b></td>

        <td style="width:15%"><b>Author</b></td>

        <td style="width:15%"><b>Contact</b></td>

        <td style="width:5%"><b>Version</b></td>

        <td style="width:5%"><b>License</b></td>
      </tr>

      <tr valign="top">
        <td><a href="http://www.maths.lth.se/help/R/aroma/">aroma</a></td>

        <td>Object-oriented microarray analysis package.</td>

        <td>H. Bengtsson</td>

        <td>hb@maths.lth.se</td>

        <td>0.75</td>

        <td>GNU GPL</td>
      </tr>

      <tr valign="top">
        <td><a href=
        "http://www.bioconductor.org/repository/devel/package/html/pickgene.html">
        pickgene</a></td>

        <td>Normalization, differential expression, simulation, etc.</td>

        <td>B. S. Yandell</td>

        <td>yandell@<br />
        stat.wisc.edu</td>

        <td>1.0.0</td>

        <td>GNU GPL</td>
      </tr>

      <tr valign="top">
        <td><a href=
        "http://cran.r-project.org/src/contrib/Descriptions/som.html">som</a></td>

        <td>Clustering using self-organizing maps. Also provides simple
        filtering and normalization functions. Includes yeast cell cycle
        data.</td>

        <td>J. Yan</td>

        <td>jyan@<br />
        stat.wisc.edu</td>

        <td>0.3-4</td>

        <td>GNU GPL</td>
      </tr>

      <tr valign="top">
        <td><a href=
        "http://cran.r-project.org/src/contrib/Descriptions/permax.html">permax</a></td>

        <td>Permutation tests for microarray data (2-sample t-test, correlation
        test, etc.).</td>

        <td>R. J. Gray</td>

        <td>gray@<br />
        hsph.harvard.edu</td>

        <td>2.2</td>

        <td>GNU GPL</td>
      </tr>

      <tr valign="top">
        <td><a href=
        "http://visitor.ics.uci.edu/genex/cybert/hdarray">hdarray</a></td>

        <td>Bayesian t-tests for expression change. Part of <a href=
        "http://visitor.ics.uci.edu/genex/cybert/">GeneX/Cyber T</a>.</td>

        <td>A. D. Long</td>

        <td>tdlong@uci.edu</td>

        <td>3.70</td>

        <td>No commercial use</td>
      </tr>

      <tr valign="top">
        <td><a href=
        "http://www.molgen.mpg.de/~heydebre/software.html">ISIS</a></td>

        <td>Class discovery based on maximizing a discriminant score.</td>

        <td>A. von Heydebreck</td>

        <td>heydebre@<br />
        molgen.mpg.de</td>

        <td>2.0</td>

        <td>No commercial use</td>
      </tr>

      <tr valign="top">
        <td><a href=
        "http://odin.mdacc.tmc.edu/~kim/geneclust/">GeneClust</a></td>

        <td>Exploratory analysis of gene expression microarray data (S-Plus).
        Implements Gene Shaving.</td>

        <td>K.-A. Do<br />
        (and others)</td>

        <td>kim@<br />
        mdanderson.org</td>

        <td>1.0b11</td>

        <td>No commercial use</td>
      </tr>

      <tr valign="top">
        <td><a href=
        "http://www.cbs.dtu.dk/laurent/download/affyR/">affyR</a></td>

        <td>Analysis of data from Affymetrix oligonucleotide arrays.</td>

        <td>L. Gautier</td>

        <td>laurent@<br />
        cbs.dtu.dk</td>

        <td>0.3.3</td>

        <td>No commercial use</td>
      </tr>

      <tr valign="top">
        <td><a href=
        "http://www.math.chalmers.se/~magnusaa/maffy_software.html">maffy</a></td>

        <td>Routines for normalizing Affymetrix Oligonucleotide Arrays.</td>

        <td>M. &Aring;strand</td>

        <td>magnus.astrand@<br />
        astrazeneca.com</td>

        <td>0.2</td>

        <td>GNU GPL</td>
      </tr>



      <tr valign="top">
        <td><a href="http://www.pi.csiro.au/gena/tRMA/">tRMA</a></td>

        <td>Tools for microarray analysis: normalisation, differential
        expression, visualisation etc.</td>

        <td>P. Baker<br />
        (and others)</td>

        <td>trma@<br />
        cmis.csiro.au</td>

        <td>1.7.0</td>

        <td>?</td>
      </tr>

      <tr valign="top">
        <td><a href="http://astor.som.jhmi.edu/poe/">POE</a></td>

        <td>Probability of expression (POE). An approach to the analysis of
        gene expression microarrays using three-component mixtures.</td>

        <td>G. Parmigiani<br />
        and E. Garrett</td>

        <td>gp@jhu.edu<br />
        esg@jhu.edu</td>

        <td>?</td>

        <td>GNU GPL</td>
      </tr>

      <tr valign="top">
        <td><a href="http://www.bios.unc.edu/~fwright/fbss/#splus">Li-Wong
        Model</a></td>

        <td>S-Plus Scripts for Li-Wong full and reduced model estimates.</td>

        <td>F. A. Wright</td>

        <td>fwright@bios.<br />
        unc.edu</td>

        <td>?</td>

        <td>?</td>
      </tr>

      <tr valign="top">
        <td><a href=
        "http://www-stat.stanford.edu/~tibs/PAM/Rdist/">PAM</a></td>

        <td>Sample classification from gene expression data, by the method of
        nearest shrunken centroids.</td>

        <td>R. Tibshirani<br />
        (and others)</td>

        <td>tibs@stat.<br />
        stanford.edu</td>

        <td>1.24</td>

        <td>?</td>
      </tr>

      <tr valign="top">
        <td><a href="http://www.dawningrealm.org/stats/statomics/">statomics</a></td>

        <td>Statistical analysis of genomic and proteomic data</td>

        <td>D. Bickel</td>

        <td>bickel@prueba.info</td>

        <td>0.2</td>

        <td>GNU GPL</td>
      </tr>
      
      <tr valign="top">
        <td><a href="http://cran.r-project.org/src/contrib/Descriptions/plsgenomics.html">plsgenomics</a></td>

        <td>PLS analyses for genomics.</td>

        <td>A.-L. Bourlesteix</td>

        <td>boulesteix@stat.<br />uni-muenchen.de</td>

        <td>1.0</td>

        <td>GNU GPL</td>
      </tr>
      
      
    </table>

    <p>&nbsp;</p>

    <p class="c1">Frontends Using R (Web-based or Windows-based):</p>

    <table style="table-layout:fixed">
      <tr style="background-color:#CCCCCC">
        <td style="width:10%"><b>Package</b></td>

        <td style="width:50%"><b>Description</b></td>

        <td style="width:15%"><b>Author</b></td>

        <td style="width:15%"><b>Contact</b></td>

        <td style="width:5%"><b>Version</b></td>

        <td style="width:5%"><b>License</b></td>
      </tr>

      <tr valign="top">
        <td><a href="http://www.ncgr.org/genex/">GeneX</a></td>

        <td>General platform for the analysis and comparison of gene expression
        data.</td>

        <td>NCGR and the Computational Genomics Group at the University of
        California, Irvine</td>

        <td>genex@ncgr.org</td>

        <td>?</td>

        <td>GNU LGPL</td>
      </tr>

      <tr valign="top">
        <td><a href="http://www.iobion.com">GeneTraffic</a></td>

        <td>General platform for the data management and the analysis of
        two-colour microarray experiments (requires Windows).</td>

        <td>IOBION Informatics LCC</td>

        <td>help@iobion.com</td>

        <td>2.5</td>

        <td>Commercial</td>
      </tr>

      <tr valign="top">
        <td><a href=
        "http://pevsnerlab.kennedykrieger.org/snomadinput.html">SNOMAD</a></td>

        <td>Various procedures for standardization and normalization of
        microarray data.</td>

        <td>C. Colantuoni<br />
        (and others)</td>

        <td>ccolantu@jhmi.edu</td>

        <td>?</td>

        <td>?</td>
      </tr>

      <tr valign="top">
        <td><a href=
        "http://www.pitt.edu/~ctseng/research/cDNA_download.html">cDNA<br />
        microarray<br />
        analysis</a></td>

        <td>Analysis of two-color microarray data (requires Windows).</td>

        <td>G. C. Tseng</td>

        <td>ctseng@pitt.edu</td>

        <td>?</td>

        <td>?</td>
      </tr>
    </table>

    <p>&nbsp;</p>

    <p class="c1">General Packages:</p>

    <table style="table-layout:fixed">
      <tr style="background-color:#CCCCCC">
        <td style="width:10%"><b>Package</b></td>

        <td style="width:50%"><b>Description</b></td>

        <td style="width:15%"><b>Author</b></td>

        <td style="width:15%"><b>Contact</b></td>

        <td style="width:5%"><b>Version</b></td>

        <td style="width:5%"><b>License</b></td>
      </tr>

      <tr valign="top">
        <td><a href=
        "http://cran.r-project.org/src/contrib/PACKAGES.html">mva</a></td>

        <td>Classical multivariate analysis: principal component analysis,
        K-means, hierarchical clustering, factor analysis, etc.<br />
        <b>R base package</b>.</td>

        <td>R Core Team</td>

        <td>R-core@<br />
        r-project.org</td>

        <td>1.5.0</td>

        <td>GNU GPL</td>
      </tr>

      <tr valign="top">
        <td><a href=
        "http://cran.r-project.org/src/contrib/PACKAGES.html">modreg</a></td>

        <td>Modern regression analysis: Smoothing and Local Methods.<br />
        <b>R base package</b>.</td>

        <td>R Core Team</td>

        <td>R-core@<br />
        r-project.org</td>

        <td>1.5.0</td>

        <td>GNU GPL</td>
      </tr>

      <tr valign="top">
        <td><a href=
        "http://cran.r-project.org/src/contrib/PACKAGES.html#multiv">multiv</a></td>

        <td>Multivariate data analysis routines: hierarchical clustering,
        principal component analysis, Sammon's nonlinear mapping,
        correspondence analysis, K-means, etc.</td>

        <td>F. Murtagh</td>

        <td>fmurtagh@eso.org</td>

        <td>1.1-4</td>

        <td>No commercial use</td>
      </tr>

      <tr valign="top">
        <td><a href=
        "http://cran.r-project.org/src/contrib/PACKAGES.html#fastICA">fastICA</a></td>

        <td>Implementation of FastICA algorithm to perform Independent
        Component Analysis (ICA) and Projection Pursuit.</td>

        <td>J. L.Marchini</td>

        <td>marchini@<br />
        stats.ox.ac.uk</td>

        <td>1.1-4</td>

        <td>GNU GPL</td>
      </tr>

      <tr valign="top">
        <td><a href=
        "http://cran.r-project.org/src/contrib/PACKAGES.html#cluster">cluster</a></td>

        <td>Functions for (hierarchical) cluster analysis.</td>

        <td>P.&nbsp;Rousseeuw<br />
        (and others)</td>

        <td>rousse@uia.ua.ac.be</td>

        <td>1.6-4</td>

        <td>GNU GPL</td>
      </tr>

      <tr valign="top">
        <td><a href=
        "http://cran.r-project.org/src/contrib/PACKAGES.html#cclust">cclust</a></td>

        <td>Convex clustering methods, including K-means algorithm and
        calculation of several indexes for finding the number of clusters in a
        data set.</td>

        <td>E.&nbsp;Dimitriadou</td>

        <td>dimi@ci.tuwien.ac.at</td>

        <td>0.6-9</td>

        <td>GNU GPL</td>
      </tr>

      <tr valign="top">
        <td><a href=
        "http://cran.r-project.org/src/contrib/PACKAGES.html#tree">tree</a></td>

        <td>Classification and regression trees.</td>

        <td>B. Ripley</td>

        <td>ripley@stats.ox.ac.uk</td>

        <td>1.0-12</td>

        <td>GNU GPL</td>
      </tr>

      <tr valign="top">
        <td><a href=
        "http://cran.r-project.org/src/contrib/PACKAGES.html">class</a></td>

        <td>Various functions for classification.</td>

        <td>B. Ripley</td>

        <td>ripley@stats.ox.ac.uk</td>

        <td>6.2-6</td>

        <td>GNU GPL</td>
      </tr>

      <tr valign="top">
        <td><a href=
        "http://cran.r-project.org/src/contrib/PACKAGES.html">nnet</a></td>

        <td>Feed-forward neural networks and multinomial log-linear
        models.</td>

        <td>B. Ripley</td>

        <td>ripley@stats.ox.ac.uk</td>

        <td>6.2-6</td>

        <td>GNU GPL</td>
      </tr>

      <tr valign="top">
        <td><a href=
        "http://www.stat.washington.edu/fraley/mclust/soft.shtml">mclust</a></td>

        <td>Model-based clustering and discriminant analysis, including
        hierarchical clustering and EM for parameterized Gaussian mixtures and
        Poisson noise.</td>

        <td>C. Fraley<br />
        (and others)</td>

        <td>fraley@<br />
        stat.washington.edu</td>

        <td>1.1-4</td>

        <td>No commercial use</td>
      </tr>

      <tr valign="top">
        <td><a href=
        "http://cran.r-project.org/src/contrib/PACKAGES.html#e1071">e1071</a></td>

        <td>Functions for latent class analysis, support vector machines, fuzzy
        clustering, bagged clustering, etc.</td>

        <td>F. Leisch<br />
        (and others)</td>

        <td>Friedrich.Leisch@<br />
        ci.tuwien.ac.at</td>

        <td>1.3-11</td>

        <td>GNU GPL</td>
      </tr>

      <tr valign="top">
        <td><a href=
        "http://cran.r-project.org/src/contrib/PACKAGES.html#dr">dr</a></td>

        <td>Dimension reduction regression, incl. sliced inverse regression
        (SIR)</td>

        <td>S. Weisberg<br /></td>

        <td>sandy@stat.umn.edu</td>

        <td>1.0-3</td>

        <td>GNU GPL</td>
      </tr>

      <tr valign="top">
        <td><a href=
        "http://cran.r-project.org/src/contrib/PACKAGES.html#randomForest">randomForest</a></td>

        <td>Classification based on a forest of classification trees using
        random inputs.</td>

        <td>A. Liaw</td>

        <td>andy_liaw@<br />
        merck.com</td>

        <td>3.4-4</td>

        <td>GNU GPL</td>
      </tr>

      <tr valign="top">
        <td><a href=
        "http://www.stat.math.ethz.ch/~dettling/boosting.html">LogitBoost</a></td>

        <td>Classification with LogitBoost.</td>

        <td>M. Dettling</td>

        <td>dettling@stat.<br />
        math.ethz.ch</td>

        <td>1.0</td>

        <td>GNU GPL</td>
      </tr>

      <tr valign="top">
        <td><a href=
        "http://user.cs.tu-berlin.de/~astro/kmethods/">kmethods</a></td>

        <td>Kernel based dimensionality reduction and clustering methods
        (ISOMAP etc).</td>

        <td>M. Kuss</td>

        <td>astro@cs.tu-berlin.de</td>

        <td>0.1-1</td>

        <td>GNU GPL</td>
      </tr>
    </table>

    <p>&nbsp;</p>

</div>	

<div id="footer">
    <em>Last modified:</em>
    July 1, 2005
</div>
    

  </div>
</body>
</html>
