<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html>
  <head>
    <title>
    Practical Analysis of Gene Expression Data</title>
    <meta http-equiv="Content-Type" content=
    "text/html; charset=iso-8859-1">
    <meta name="Author" content="Korbinian Strimmer">
   </head>

  <body bgcolor="#FFFFFF" link="#0000BF" vlink="#555555" alink="#0000BF">

 <table align="center" width="660">
      <tr>
        <td align="left">   
	
   <p>Winter Term 2003/2004</p>
   
   <h3>Workshop</h3>
   
   <h1><a href="http://webinfo.campus.lmu.de/view_event.cfm?ev=66561&amp;sort=type">
              Practical Analysis of Gene Expression Data</a> </h1>

   
              <p><em>Jointly taught by:</em>
	      <ul>
	      <li>Korbinian Strimmer, Juliane Sch&auml;fer, Anne-Laure Boulesteix
	       (<a href="http://www.stat.uni-muenchen.de">Statistics/LMU</a>), and</li>
	       <li>Alexander Yassouridis, Klaus Hechenbichler, Bertram M&uuml;ller-Myhsok,
              Markus Panhuysen, Benno P&uuml;tz, Daria Salyakina, Klaus Schliep, Shaun Seaman
	      (<a href="http://www.mpipsykl.mpg.de/pages/english/welcome_en.html">Max Planck
	      Institute for Psychiatry</a>).</li></p>
              </ul>

             <h3>Registration:</h3>
	      
	      <p>For participation please register
              (until 31 January 2004) with either
	      <a href="https://strimmerlab.github.io/korbinian.html">Korbinian Strimmer</a>
	      (<a href="mailto:strimmer@stat.uni-muenchen.de">email)</a> or
	       <a href="http://www.mpipsykl.mpg.de/pages/forschung/yass_mitarb.htm">Alexander
              Yassouridis</a> (<a href="mailto:yassou@mpipsykl.mpg.de">email)</a>.</p>



 

              <h3>Overview:</h3>

              <p>This microarray workshop   provides a practical introduction to
              statistical and computational aspects of gene expression
              analysis. Emphasis is put both on the theoretical background,
              such as normalization procedures, differential expression, ANOVA,
              and clustering and classification procedures, as well as on
              practical exercises using <a
              href="https://web.archive.org/web/20061026041319/http://strimmerlab.org/notes/rexpress.html">R
              packages for gene expression analysis</a> (in particular the <a
              href="http://www.bioconductor.org">Bioconductor</a>
              packages).</p>


              <p>The course runs full four days (10:00-17:00) from <strong>16 to 19 February
              2004</strong> and takes place in the computer room of
              the Department of Statistics. </p>


              <h3>Agenda:</h3>

              <ul>
	        <li>Introduction to microarray technology</li>
	      
                <li>Normalization procedures</li>

                <li>Differential expression</li>

                <li>Clustering gene expression data</li>

		
		<li>Classification of tissue samples</li>

                <li>Inference of genetic networks</li>
              </ul>
              
              <p>On each topic there will be both lectures and practical excercises.</p>

	      
	      <p>For further details please download the course flyer
	       (<a href="flyer.pdf">pdf</a>) and the
	      timetable and program (<a href="program.pdf">pdf</a>).</p>




            <h3>Course Materials:</h3>


<table cols="3" width="100%">
            <tr bgcolor="#CCCCCC">
              <td width="15%">Day</td>

              <td width="55%">Talk/Tutorial</td>

              <td width="30%">Speaker/Tutor</td>
            </tr>

            <tr bgcolor="#FFFFAA">
              <td>16.2.2004</td>

              <td>
	      <a href="d1/talk-arrayintro.pdf">Introduction to microarray technology</a>
	      </td>

              <td>Marcus Panhuysen</td>
            </tr>

            <tr bgcolor="#FFFFAA">
              <td> </td>

              <td><a href="d1/talk-normalization.pdf">Preprocessing and normalization</a></td>

              <td>Benno P&uuml;tz
              </td>
            </tr>

            <tr  bgcolor="#FFFFAA">
              <td> </td>

              <td><a href="d1/talk-Rintro.pdf">Introduction to R</a></td>

              <td>Klaus Schliep
              </td>
            </tr>

            <tr  bgcolor="#EEEEEE">
              <td> </td>

              <td>Introduction to R
	          (<a href="d1/exercise-Rintro.pdf">exercise</a> and
	          <a href="d1/code-Rintro.txt">solution</a>)
	      </td>

              <td>Klaus Schliep
              </td>
            </tr>

            <tr  bgcolor="#EEEEEE">
              <td> </td>

              <td>
	        Normalization 
		(<a href="d1/exercise-normalization.pdf">exercise</a>
	        and <a href="d1/code-normalization.txt">solution</a>)
	      </td>

              <td>Korbinian Strimmer
              </td>
            </tr>

 
 
 
 
 
 

            <tr bgcolor="#FFFFAA">
              <td>17.2.2004</td>

              <td>
	      <a href="d2/talk-ttest.pdf">t-test and other related methods</a>
	      </td>

              <td>Shaun Seaman</td>
            </tr>

            <tr bgcolor="#FFFFAA">
              <td> </td>

              <td><a href="d2/talk-multtest.pdf">Multiple testing</a></td>

              <td>Daria Salyakina 
              </td>
            </tr>

            <tr  bgcolor="#FFFFAA">
              <td> </td>

              <td><a href="d2/talk-diffexp.pdf">Differential expression</a></td>

              <td>Bertram M&uuml;ller-Myhsok
              </td>
            </tr>

            <tr  bgcolor="#EEEEEE">
              <td> </td>

              <td>Multiple testing and differential expression
	          (<a href="d2/exercise-multtest.pdf">exercises)
	      </td>

              <td>Benno P&uuml;tz
              </td>
            </tr>

           








 
            <tr bgcolor="#FFFFAA">
              <td>18.2.2004</td>

              <td>
	      <a href="d3/talk-expdesign.pdf">Experimental design/MANOVA</a>
	      </td>

              <td>Alexander Yassouridis</td>
            </tr>

            <tr bgcolor="#FFFFAA">
              <td> </td>

              <td><a href="d3/talk-mva.pdf">Clustering and classification methods</a></td>

              <td>Klaus Hechenbichler
              </td>
            </tr>


            <tr   bgcolor="#EEEEEE">
              <td> </td>

              <td>Clustering and classification
	          (<a href="d3/exercise-mva.pdf">exercises</a></a>)
	      </td>

              <td>Klaus Schliep and <br>Klaus Hechenbichler
              </td>
            </tr>

            
	    
	    
	    
 
            <tr bgcolor="#FFFFAA">
              <td>19.2.2004</td>

              <td>
	      <a href="d4/talk-beyond.pdf">Microarray analysis beyond differential expression</a>
	      </td>

              <td>Korbinian Strimmer</td>
            </tr>

            <tr bgcolor="#FFFFAA">
              <td> </td>

              <td><a href="d4/talk-cellcycle.pdf">Identifying periodically expressed genes</a></td>

              <td>Korbinian Strimmer
              </td>
            </tr>

            <tr bgcolor="#FFFFAA">
              <td> </td>

              <td><a href="d4/talk-eps.pdf">Emerging patterns and gene interactions</a></td>

              <td>Anne-Laure Boulesteix
              </td>
            </tr>

            <tr bgcolor="#FFFFAA">
              <td> </td>

              <td><a href="d4/talk-ggm.pdf">Inferring large graphical models from microarray data</a></td>

              <td>Juliane Sch&auml;fer
              </td>
            </tr>


            <tr  bgcolor="#EEEEEE">
              <td> </td>

              <td>Special topics 
	          (<a href="d4/exercise-special.pdf">exercise</a> and
	          <a href="d4/code-special.txt">solution</a>)
	      </td>

              <td>Juliane Sch&auml;fer and Anne-Laure Boulesteix
              </td>
            </tr>

            

          </table>

<p>
The data sets required for the exercises on clustering and classification (day 3)
are also available (compressed with gzip):
<a href="d3/carci.rda.gz">carci.rda</a>,
<a href="d3/nci.rda.gz">nci.rda</a>,
<a href="d3/leukemia.rdata.gz">leukemia.rdata</a>,
<a href="d3/srbct.rdata.gz">srbct.rdata</a>,
<a href="d3/colon.rdata.gz">colon.rdata</a>.
</p>

<P>The <a href="http://www.r-project.org">R</a> packages mainly used are the
 <a href="http://www.bioconductor.org">Bioconductor</a> packages
and the <a href="https://strimmerlab.github.io/software/genets/">GeneTS</a> package.</p>
</p>

<p>We gratefully acknowledge that parts of the above material is taken from
the <a href="http://www.bioconductor.org">Bioconductor</a>
web site and from the <a href="http://compdiag.molgen.mpg.de/ngfn/">NGFN Microarray data analysis
resource</a>.

</p>
 
<div align="right">

            <i><font size="-2">Last modified:</font></i><br>
             <font size="-2">January 20, 2004</font>
          </div>
	       
         </td>
	 </tr>
	 </table>     
              
  </body>
</html>

